<!DOCTYPE html>
<html lang="en-us">
<head>

  <meta charset="utf-8">
  <meta name="viewport" content="width=device-width, initial-scale=1">
  <meta http-equiv="X-UA-Compatible" content="IE=edge">
  <meta name="theme" content="hugo-academic">
  <meta name="generator" content="Hugo 0.35" />
  <meta name="author" content="David McGaughey">

  
  
  
  
    
      
    
  
  <meta name="description" content="Day 3 Rafael Irizarry Keynote Session 4: Transcriptomics, Alternative Splicing and Gene Predictions Mark Robinson (Robinson) Hagen Tilgner (Tilgner) Nikka Keivanfar (Church, 10X) Koen Van den Berge (Clement) Barbara Englehardt (Englehardt) Jeff Gaither (White) Fiona Dick (Tzoulis)  Epigenetics and non-coding genome Jordana Bell (Bell) Wouter Meuleman (Stamatoyannopoulos) Maša Roller (Flicek) Alexander Suh (Suh) Raquel Garcia-Perez (Juan)    Day 3 Very sparse and poorly written notes covering #GI2018.">

  
  <link rel="alternate" hreflang="en-us" href="/./post/gi2018-day-three/">

  


  

  
  
  <meta name="theme-color" content="#0095eb">
  
  
  
  
    
  
  
    
    
      
        <link rel="stylesheet" href="//cdnjs.cloudflare.com/ajax/libs/highlight.js/9.12.0/styles/github.min.css">
      
    
  
  
  <link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.3.7/css/bootstrap.min.css" integrity="sha512-6MXa8B6uaO18Hid6blRMetEIoPqHf7Ux1tnyIQdpt9qI5OACx7C+O3IVTr98vwGnlcg0LOLa02i9Y1HpVhlfiw==" crossorigin="anonymous">
  <link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/academicons/1.8.6/css/academicons.min.css" integrity="sha256-uFVgMKfistnJAfoCUQigIl+JfUaP47GrRKjf6CTPVmw=" crossorigin="anonymous">
  <link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/4.7.0/css/font-awesome.min.css" integrity="sha512-SfTiTlX6kk+qitfevl/7LibUOeJWlt9rbyDn92a1DqWOw9vWG2MFoays0sgObmWazO5BQPiFucnnEAjpAB+/Sw==" crossorigin="anonymous">
  <link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/fancybox/3.2.5/jquery.fancybox.min.css" integrity="sha256-ygkqlh3CYSUri3LhQxzdcm0n1EQvH2Y+U5S2idbLtxs=" crossorigin="anonymous">
  
  
  
  
  <link rel="stylesheet" href="//fonts.googleapis.com/css?family=Montserrat:400,700%7cRoboto:400,400italic,700%7cRoboto&#43;Mono">
  
  <link rel="stylesheet" href="/styles.css">
  

  
    <script>
      window.ga=window.ga||function(){(ga.q=ga.q||[]).push(arguments)};ga.l=+new Date;
      ga('create', 'UA-113461893-1', 'auto');
      
      ga('require', 'eventTracker');
      ga('require', 'outboundLinkTracker');
      ga('require', 'urlChangeTracker');
      ga('send', 'pageview');
    </script>
    <script async src="//www.google-analytics.com/analytics.js"></script>
    
    <script async src="https://cdnjs.cloudflare.com/ajax/libs/autotrack/2.4.1/autotrack.js" integrity="sha512-HUmooslVKj4m6OBu0OgzjXXr+QuFYy/k7eLI5jdeEy/F4RSgMn6XRWRGkFi5IFaFgy7uFTkegp3Z0XnJf3Jq+g==" crossorigin="anonymous"></script>
    
  

  
  <link rel="alternate" href="/./index.xml" type="application/rss+xml" title="eye Bioinformatician">
  <link rel="feed" href="/./index.xml" type="application/rss+xml" title="eye Bioinformatician">
  

  <link rel="manifest" href="/site.webmanifest">
  <link rel="icon" type="image/png" href="/img/icon.png">
  <link rel="apple-touch-icon" type="image/png" href="/img/icon-192.png">

  <link rel="canonical" href="/./post/gi2018-day-three/">

  <meta property="twitter:card" content="summary_large_image">
  
  <meta property="twitter:site" content="@https://twitter.com/David_McGaughey">
  <meta property="twitter:creator" content="@https://twitter.com/David_McGaughey">
  
  <meta property="og:site_name" content="eye Bioinformatician">
  <meta property="og:url" content="/./post/gi2018-day-three/">
  <meta property="og:title" content="#GI2018 - Day Three | eye Bioinformatician">
  <meta property="og:description" content="Day 3 Rafael Irizarry Keynote Session 4: Transcriptomics, Alternative Splicing and Gene Predictions Mark Robinson (Robinson) Hagen Tilgner (Tilgner) Nikka Keivanfar (Church, 10X) Koen Van den Berge (Clement) Barbara Englehardt (Englehardt) Jeff Gaither (White) Fiona Dick (Tzoulis)  Epigenetics and non-coding genome Jordana Bell (Bell) Wouter Meuleman (Stamatoyannopoulos) Maša Roller (Flicek) Alexander Suh (Suh) Raquel Garcia-Perez (Juan)    Day 3 Very sparse and poorly written notes covering #GI2018.">
  <meta property="og:locale" content="en-us">
  
  <meta property="article:published_time" content="2018-09-19T00:00:00&#43;00:00">
  
  <meta property="article:modified_time" content="2018-09-19T00:00:00&#43;00:00">
  

  
  

  <title>#GI2018 - Day Three | eye Bioinformatician</title>

</head>
<body id="top" data-spy="scroll" data-target="#toc" data-offset="71" >

<nav class="navbar navbar-default navbar-fixed-top" id="navbar-main">
  <div class="container">

    
    <div class="navbar-header">
      
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse"
              data-target=".navbar-collapse" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      
      <a class="navbar-brand" href="/">eye Bioinformatician</a>
    </div>

    
    <div class="collapse navbar-collapse">

      
      
      <ul class="nav navbar-nav navbar-right">
        

        
        
        
        
        
          
        

        <li class="nav-item">
          <a href="/#about">
            
            <span>Home</span>
            
          </a>
        </li>

        
        

        
        <li class="dropdown">
          <a href="#" class="dropdown-toggle" data-toggle="dropdown" aria-haspopup="true">
            
            <span>Posts</span>
            
            <span class="caret"></span>
          </a>
          <ul class="dropdown-menu">
            
            <li class="nav-item">
              <a href="/#posts">
                
                <span>Recent Posts</span>
                
              </a>
            </li>
            
            <li class="nav-item">
              <a href="/post">
                
                <span>All Posts</span>
                
              </a>
            </li>
            
          </ul>
        </li>

        
        

        
        
        
        
        
          
        

        <li class="nav-item">
          <a href="/#projects">
            
            <span>Projects</span>
            
          </a>
        </li>

        
        

        
        
        
        
        
          
        

        <li class="nav-item">
          <a href="/#culture">
            
            <span>Culture</span>
            
          </a>
        </li>

        
        

        
        
        
        
        
          
        

        <li class="nav-item">
          <a href="/#publications_selected">
            
            <span>Publications</span>
            
          </a>
        </li>

        
        
      

      
      </ul>

    </div>
  </div>
</nav>


<article class="article" itemscope itemtype="http://schema.org/Article">

  


  <div class="article-container">
    <h1 itemprop="name">#GI2018 - Day Three</h1>

    

<div class="article-metadata">

  <span class="article-date">
    
    <time datetime="2018-09-19 00:00:00 &#43;0000 UTC" itemprop="datePublished dateModified">
      Sep 19, 2018
    </time>
  </span>
  <span itemscope itemprop="author publisher" itemtype="http://schema.org/Person">
    <meta itemprop="name" content="David McGaughey">
  </span>

  
  <span class="middot-divider"></span>
  <span class="article-reading-time">
    9 min read
  </span>
  

  
  
  <span class="middot-divider"></span>
  <a href="/./post/gi2018-day-three/#disqus_thread"></a>
  

  
  
  
  <span class="middot-divider"></span>
  <span class="article-categories">
    <i class="fa fa-folder"></i>
    
    <a href="/./categories/gi2018/">GI2018</a
    >, 
    
    <a href="/./categories/bioinformatics/">bioinformatics</a
    >
    
  </span>
  
  

  
  
<div class="share-box" aria-hidden="true">
  <ul class="share">
    <li>
      <a class="twitter"
         href="https://twitter.com/intent/tweet?text=%23GI2018%20-%20Day%20Three&amp;url=%2f.%2fpost%2fgi2018-day-three%2f"
         target="_blank" rel="noopener">
        <i class="fa fa-twitter"></i>
      </a>
    </li>
    <li>
      <a class="facebook"
         href="https://www.facebook.com/sharer.php?u=%2f.%2fpost%2fgi2018-day-three%2f"
         target="_blank" rel="noopener">
        <i class="fa fa-facebook"></i>
      </a>
    </li>
    <li>
      <a class="linkedin"
         href="https://www.linkedin.com/shareArticle?mini=true&amp;url=%2f.%2fpost%2fgi2018-day-three%2f&amp;title=%23GI2018%20-%20Day%20Three"
         target="_blank" rel="noopener">
        <i class="fa fa-linkedin"></i>
      </a>
    </li>
    <li>
      <a class="weibo"
         href="http://service.weibo.com/share/share.php?url=%2f.%2fpost%2fgi2018-day-three%2f&amp;title=%23GI2018%20-%20Day%20Three"
         target="_blank" rel="noopener">
        <i class="fa fa-weibo"></i>
      </a>
    </li>
    <li>
      <a class="email"
         href="mailto:?subject=%23GI2018%20-%20Day%20Three&amp;body=%2f.%2fpost%2fgi2018-day-three%2f">
        <i class="fa fa-envelope"></i>
      </a>
    </li>
  </ul>
</div>


  

</div>


    <div class="article-style" itemprop="articleBody">
      <div id="TOC">
<ul>
<li><a href="#day-3">Day 3</a></li>
<li><a href="#rafael-irizarry-keynote">Rafael Irizarry Keynote</a></li>
<li><a href="#session-4-transcriptomics-alternative-splicing-and-gene-predictions">Session 4: Transcriptomics, Alternative Splicing and Gene Predictions</a><ul>
<li><a href="#mark-robinson-robinson">Mark Robinson (Robinson)</a></li>
<li><a href="#hagen-tilgner-tilgner">Hagen Tilgner (Tilgner)</a></li>
<li><a href="#nikka-keivanfar-church-10x">Nikka Keivanfar (Church, 10X)</a></li>
<li><a href="#koen-van-den-berge-clement">Koen Van den Berge (Clement)</a></li>
<li><a href="#barbara-englehardt-englehardt">Barbara Englehardt (Englehardt)</a></li>
<li><a href="#jeff-gaither-white">Jeff Gaither (White)</a></li>
<li><a href="#fiona-dick-tzoulis">Fiona Dick (Tzoulis)</a></li>
</ul></li>
<li><a href="#epigenetics-and-non-coding-genome">Epigenetics and non-coding genome</a><ul>
<li><a href="#jordana-bell-bell">Jordana Bell (Bell)</a></li>
<li><a href="#wouter-meuleman-stamatoyannopoulos">Wouter Meuleman (Stamatoyannopoulos)</a></li>
<li><a href="#masa-roller-flicek">Maša Roller (Flicek)</a></li>
<li><a href="#alexander-suh-suh">Alexander Suh (Suh)</a></li>
<li><a href="#raquel-garcia-perez-juan">Raquel Garcia-Perez (Juan)</a></li>
</ul></li>
</ul>
</div>

<div id="day-3" class="section level1">
<h1>Day 3</h1>
<p>Very sparse and poorly written notes covering <a href="https://twitter.com/hashtag/GI2018?src=hash">#GI2018</a>.</p>
<p>Typos everywhere. Things may change dramatically over time as I scan back through notes.</p>
<p>I’ve tried to respect #notwitter. Will be updated periodically.</p>
<p>Speaker (Lab | Group)</p>
</div>
<div id="rafael-irizarry-keynote" class="section level1">
<h1>Rafael Irizarry Keynote</h1>
<p>Understanding variabliity in high throughput data</p>
<p>Typical workflow raw data -&gt; pre-processing -&gt; analysis -&gt; discovery (false?????)</p>
<p>“When you find an <strong>unexpected result</strong>, be skeptical, check for systematic errors”</p>
<p>“Always, look at the data”</p>
<p>“Dynamite plots needs to die” (barplot + SE/SD)</p>
<ul>
<li>Plug for a blog post by me: <a href="http://davemcg.github.io/post/let-s-plot-4-r-vs-excel/" class="uri">http://davemcg.github.io/post/let-s-plot-4-r-vs-excel/</a></li>
<li>Show the data (boxplot + individual points)</li>
</ul>
<p>Showing batch effect example where huge diff in gene expression between groups is actually derived from technical issues (two groups were effectively done at different times)</p>
<p>Lin et al. 2014 (notorious paper comparing tx between human and mouse the claimed dif tissues within species more common to each other)</p>
<p>Yoav Gilad quickly pointed out that the mouse and human were done on different sequencers</p>
<p>So they re-did on same sequencer machine….got similar result</p>
<p>Not a new “result” (yanai 2004 had similar conclusion with microarray)</p>
<p>Rafael pointing out that high genes are high and low are low…always, so correlation will always be high.</p>
<p>Once you fix probe bias (microarray), then tissues cluster together.</p>
<p>But what about Lin RNA-seq?</p>
<p>A lot is explained by:</p>
<ul>
<li>number of transcripts</li>
<li>GC content</li>
</ul>
<p>Single Cell RNA-seq</p>
<ul>
<li>trying to disentangle batch effect from tumor type</li>
<li>actually proportion of zeros is driving variability
<ul>
<li>this is probably not biology</li>
</ul></li>
</ul>
<p>Do DNAm(ethylation) changes drive gene expression changes?</p>
<p>Discussing Ford et al. 2017 <em>bioRxiv</em></p>
<p>“Distinguishing consistent differences from random ones is challenging”</p>
<p>Talking about how technical variability and biological variability are different and must be considered when designing experiment (moar replicates!)</p>
<p>Do we trust individual measurements?</p>
<ul>
<li>when you do lots of tests, individual tests can be wrong</li>
<li>region test was done for CpG methylation (DMR test Jaffe 2012)</li>
<li>can also shuffle data (shuffle samples, not points) and see how often a diff can occur by chance</li>
<li>region + bootstrapping powerful approach (plus looking at replicates individually!!!)</li>
</ul>
</div>
<div id="session-4-transcriptomics-alternative-splicing-and-gene-predictions" class="section level1">
<h1>Session 4: Transcriptomics, Alternative Splicing and Gene Predictions</h1>
<div id="mark-robinson-robinson" class="section level2">
<h2>Mark Robinson (Robinson)</h2>
<p>On the analysis of long-read sequencing data for gene expression</p>
<p>Illumina &lt;-&gt; ONT (Oxford Nanopore) cheap &lt;-&gt; expensive short fragments &lt;-&gt; full length lower error rates &lt;-&gt; higher error rates</p>
<p>DataSet</p>
<ul>
<li>WT vs Srpk1-KO</li>
<li>ONT (various preps) and Illumina</li>
<li>1M reads / sample (ONT); 30M reads / sample (Illumina)</li>
</ul>
<p>25% mismatch with ONT vs &lt;1% for Illumina (but they still map)</p>
<p>ONT doesn’t look useful for variation in sequence (way too noisy)</p>
<p>Do we actually get full length transcripts?</p>
<ul>
<li>for longer ones…not really</li>
<li>showing how a longer transcript has worse coverage with ONT</li>
</ul>
<p>Quantification ONT &lt;-&gt; Illumina</p>
<ul>
<li>looks pretty good for protein coding genes</li>
<li>more discrepancy at the tx level (still OK)</li>
</ul>
<p>ONT</p>
<ul>
<li>minimap2 + salmon current choice for tooling</li>
</ul>
<p>ONT direct RNA or ONT cDNA???</p>
<ul>
<li>look pretty comparable</li>
<li>direct RNA requires crazy input amounts, probably not worth it unless you are looking at RNA base modifications</li>
<li>ONT has new stranded cDNA protocol / kit</li>
</ul>
<p>Diff expression analysis</p>
<ul>
<li>once you account for way lower depth (again 1M vs 30M) for ONT, roughly comparable</li>
</ul>
<p>Definitely not time to jump over for more standard DE type RNAseq tests</p>
</div>
<div id="hagen-tilgner-tilgner" class="section level2">
<h2>Hagen Tilgner (Tilgner)</h2>
<p>Single-cell isoform RNA sequencing (ScISOr-Seq) across thousands of cells reveals isoforms of cerebellar cell types.</p>
<p>Need long-read RNA seq to confidently ID isoforms</p>
<p>ScISOr-Seq</p>
<ul>
<li>10X uses 3’ sequencing</li>
<li>Add PacBio ISO-Seq and find the 10X barcode</li>
<li>blend 10X and PacBio data</li>
</ul>
<p>Barcode ID (long read data can be noisy…)</p>
<ul>
<li>99.99% specificity with PacBio</li>
<li>Only 58% of molecules have barcode (74% in Illumina)
<ul>
<li>not catastrophic, more annoying</li>
</ul></li>
</ul>
<p>It works</p>
<ul>
<li>validate some junctions with MS/MS</li>
</ul>
<p>Plug for <a href="isoformAtlas.com" class="uri">isoformAtlas.com</a></p>
<p>See some compelling cell type specific isoform usage</p>
</div>
<div id="nikka-keivanfar-church-10x" class="section level2">
<h2>Nikka Keivanfar (Church, 10X)</h2>
<p>Bootstrapping Biology: Quick and easy de novo genome assembly to enable single cell gene expression analysis</p>
<p>Linked-Reads + Supernova + Cactus/CAT</p>
<p>Supernova does assembly (with phase)</p>
<p>Supernova v2 can get N50 over 100+ kb for many species (HUman, Hummingbird, Drosophila)</p>
<ul>
<li>Zebrafish is tougher (only around 20kb)</li>
</ul>
<p>CAT</p>
<ul>
<li>Comparative Annotation Toolkit</li>
<li>“provides a flexible way to simultaneously annotate entire clades and identify orthology relationships”</li>
<li>doi.org/10.1101/231118</li>
<li><a href="https://genome.cshlp.org/content/28/7/1029" class="uri">https://genome.cshlp.org/content/28/7/1029</a></li>
</ul>
<p>“Blackjack” (Donkey) blood sequenced with 10X</p>
<ul>
<li>made assembly with Supernova</li>
<li>better metrics than other donkey genomes</li>
</ul>
<p>Used assembly (+CAT to get gene info) for scRNA-seq of the blood</p>
<ul>
<li>labels major blood types (T cells, B cells, NK, monocytes, plasma, etc)</li>
</ul>
</div>
<div id="koen-van-den-berge-clement" class="section level2">
<h2>Koen Van den Berge (Clement)</h2>
<p>Discrete and continuous differential expression analysis for single-cell RNA-seq data</p>
<p>scRNA-seq data is WAY noiser than bulk RNA-seq (at least cell to cell)</p>
<p>Bulk RNAseq generally use negative binomial models (edgeR, DESeq2)</p>
<p>Sort of surprisingly, these bulk RNAseq methods seem to work well in scRNA</p>
<p>But zeros…..showing simulations where adding zeros getting weird dispersion patterns</p>
<ul>
<li><strong>Why not ZINB-WAaE??</strong></li>
</ul>
<p>In real also see some odd dispersion</p>
<p>Propose zero inflated negative binomal (ZINB) distribution</p>
<ul>
<li><strong>Oh, this is the ZINB-WaVE author</strong></li>
</ul>
<p>Downweighting excess zeros helps reduce variance and improve power</p>
<p><a href="https://genomebiology.biomedcentral.com/articles/10.1186/s13059-018-1406-4" class="uri">https://genomebiology.biomedcentral.com/articles/10.1186/s13059-018-1406-4</a></p>
<p>Continuous DE</p>
<ul>
<li>following SC across developmental trajectory</li>
<li>slingshot tool (Street et al. 2018)
<ul>
<li><a href="https://github.com/kstreet13/slingshot" class="uri">https://github.com/kstreet13/slingshot</a></li>
</ul></li>
</ul>
<p>What do you do downstream of the trajectory analyses???</p>
<ul>
<li>most do DE between clusters</li>
<li>which he argues is sub-optimal
<ul>
<li>no fixed biological meaning</li>
<li>hetergeneous</li>
<li>inflate FDR with so many comparisons required</li>
</ul></li>
</ul>
<p>Propose NB-GAM</p>
<ul>
<li>so smooth out pseudotime trajectory with offset of sequencing depth (or other covariates?)</li>
<li>cubic spline smoother right now</li>
<li><strong>this is pretty slick</strong></li>
<li>“very preliminary”</li>
</ul>
</div>
<div id="barbara-englehardt-englehardt" class="section level2">
<h2>Barbara Englehardt (Englehardt)</h2>
<p>A generative model for single-cell RNA-sequencing</p>
<p>Goal: develop low-dimensional generative model to address confounders</p>
<p>What is useful about this?</p>
<ul>
<li>normalization</li>
<li>batch correction</li>
<li>imputation</li>
<li>visualization</li>
</ul>
<p>Student’s t Gaussian process laternt varaible model (tGPLVM)</p>
<p>Compare viz results of tGPLVM to t-SNE, PCA, ZIFA</p>
<p>Many many examples … hard to explain. Looks roughly comparable to ZIFA, t-SNE for cell type / gene expression</p>
<p>Nice trajectory tree plotting</p>
<ul>
<li>absolutely no idea how the plot was made. Some kind of branching tree thing</li>
</ul>
</div>
<div id="jeff-gaither-white" class="section level2">
<h2>Jeff Gaither (White)</h2>
<p>Constraint for mRNA structure in human synonymous mutations</p>
<p>Synonymous mutations that alter mRNA stability have lower incidence rate in humas</p>
<p>AT -&gt; CG stabilizing CG -&gt; AT de-stabilizing</p>
<p>Bigger shift in stability (their own metric, no idea what it is) are less common (gnomAD)</p>
<p>mRNA sturcture previously implicated in</p>
<ul>
<li>translation speed</li>
<li>mRNA half life</li>
<li>miRNA/protein binding</li>
</ul>
<p>Measure ‘deform structure’ with <a href="https://www.tbi.univie.ac.at/RNA/">ViennaRNA</a></p>
<p>dMFE is delta minimum free energy</p>
<ul>
<li>low is stabilizing</li>
<li>high is de-stabilizing</li>
</ul>
<p>CpG transitions that are destabilizing are selected against</p>
<p>But is this causative???</p>
<ul>
<li>CpG content is partially responsible</li>
<li>as is GC content</li>
</ul>
</div>
<div id="fiona-dick-tzoulis" class="section level2">
<h2>Fiona Dick (Tzoulis)</h2>
<p>Differential isoform usage in Parkinson’s disease</p>
<p>Discussing DTE vs DTU E is for expression U is for usage DT is for Differential Transcript</p>
<p>U is more for proportion of transcript</p>
<p>Ribo-Zero capture (reduce 5’ 3’ bias) -&gt; Salmon -&gt; Tximport -&gt; EVERYTHING*</p>
<ul>
<li></li>
<li>DRIMSeq –&gt; StageR</li>
<li>DEXSeq –&gt; StageR</li>
<li>ISAR + DRIMSEq</li>
<li>ISAR (IsoformSwitchAnalyzeR)</li>
</ul>
<p>Intersected all results -&gt; OI ANOTHER CRAZY VENN</p>
<p>ISAR returned far fewer results</p>
<ul>
<li>it filters out a ton of transcripts</li>
</ul>
<p>Of the 6 genes that all tools agree -&gt; 5 switch from protein coding to not protein coding</p>
</div>
</div>
<div id="epigenetics-and-non-coding-genome" class="section level1">
<h1>Epigenetics and non-coding genome</h1>
<div id="jordana-bell-bell" class="section level2">
<h2>Jordana Bell (Bell)</h2>
<p>Interpreting variation in the human methylome</p>
<p>DNA methylation heritability in TwinSUK</p>
<ul>
<li>400 female twins</li>
<li>Imputed genotypes</li>
<li>Infinium MethylationEPIC (772k CpGs after QC)</li>
</ul>
<p>Heritability of MZ vs DZ twins</p>
<ul>
<li>MZ have higher cor, than DZ, which is higher than unrelated</li>
<li>calculated heritability (h2)&gt; 40% in 10.5% of methylome</li>
<li>enhancer more heritable, promoter less so (really?)</li>
</ul>
<p>meQTLs</p>
<ul>
<li>found a bunch (200k)</li>
<li>more in CpG islands, promoters</li>
</ul>
<p>Extending/validating in new cohort</p>
<ul>
<li>another 2k people</li>
</ul>
<p>GoDMC consortium</p>
<ul>
<li><a href="godmc.org.uk" class="uri">godmc.org.uk</a></li>
<li>blood, Illumina 450k, 1000G imputation</li>
<li>27k samples</li>
<li>many many covariates</li>
<li>190k (45%) of CpGs have &gt;=1 mQTL
<ul>
<li>90% in cis</li>
</ul></li>
<li>Grouped CpGs with many cis and trans links (“community”)</li>
</ul>
</div>
<div id="wouter-meuleman-stamatoyannopoulos" class="section level2">
<h2>Wouter Meuleman (Stamatoyannopoulos)</h2>
<p>Delineation and annotation of the human regulatory landscape across 400+ cell types and states</p>
<p><span class="citation">@nameluem</span></p>
<p>Quick overview of DNaseI HS assay</p>
<p>400+ cell types and states</p>
<ul>
<li>733 DNase-seq</li>
<li>439 unique cellular combinations</li>
</ul>
<p>Convert continuous signal into consensus summits</p>
<ul>
<li>3.5 million+ DNase1 HS sites</li>
<li>21% of genome</li>
<li>primarily distal to TSS</li>
<li>minority of DHs are constitutive
<ul>
<li>most are unique to sample / cell</li>
</ul></li>
</ul>
<p>Decompose DHS x cell matrix into Non-negative matrix factorization (NMF)</p>
<ul>
<li>picked k of 16</li>
<li><strong>k is kind of like cluster</strong></li>
<li>which they hand assign to cell type / role</li>
<li>Can even further drop down and assign genes to predominant k / color / assignment
<ul>
<li>20% of protein coding genes can get a label (Bonferonni FWER &lt; 0.05)</li>
</ul></li>
</ul>
<p>Interpret genetic variation</p>
<ul>
<li>match against GWAS catalogs</li>
<li>find matches for k / color against GWAS diseases</li>
<li>regulatory terms informative in GWAS</li>
</ul>
<p><strong>Very cool work, I’ve found in my <a href="https://github.com/davemcg/distill">Distill</a> project that the k/cluster assignment across the genome provided value in assignment of pathogenicity for DNA mutation</strong></p>
</div>
<div id="masa-roller-flicek" class="section level2">
<h2>Maša Roller (Flicek)</h2>
<p>Tissue-specific enhancer and promoter evolution in mammals</p>
<p>Gene expression / promoters more conserved than enhancers across great evolutionary distance</p>
<p>4 tissues, 10 mammals, 3 ChIP-seq, RNA-seq, 3 biological replicates</p>
<ul>
<li>H3K4me3, H3K27ac, H3K4me1 used to delineate active promoters, active enhnancers, primed enhancers</li>
<li>fairly stable n of promoters, enhancers across evolution</li>
<li>brain has largest num of active enhnacers</li>
</ul>
<p><strong>Maša IS USING AN <a href="http://caleydo.org/tools/upset/">UpSET</a> I AM SO HAPPY</strong></p>
<ul>
<li>showing counts of tissue shared and tissue specific enhnacers
<ul>
<li>active are more tissue specific, poised much less so</li>
<li>this would be a nightmare with a venn diagram</li>
</ul></li>
</ul>
<p>Primed enhancers evolving more rapidly than active enhancers</p>
<ul>
<li>or are less conserved</li>
<li>tissue specific even less conserved</li>
</ul>
<p>Regulatory element re-use</p>
<ul>
<li>enhancers used over and over again across tissues through evolution
<ul>
<li><strong>implication is that sets of enhancers crucial for activity?</strong></li>
</ul></li>
</ul>
</div>
<div id="alexander-suh-suh" class="section level2">
<h2>Alexander Suh (Suh)</h2>
<p>Mind the gap – interrogating the non-coding genome with single-molecule technologies</p>
<p>Discussion of genome assembly gaps with short reads with the puzzle analogy</p>
<p>Chicken genome is well assembled (big $ in chickens)</p>
<p>Look at bird of paradise</p>
<p>Paradise grove bird used as reference</p>
<ul>
<li>PACBIO</li>
<li>Dovetail</li>
<li>Illumina</li>
<li>10X linked-reads</li>
<li>Hi-C PHASE genomics</li>
</ul>
<p>Birds full of repeats</p>
<ul>
<li>especially sex chr</li>
<li>make assembly tricky</li>
<li>crazy collinearity plots (<strong>not sure if this is the right name for the plots that lay out chr level genome vs genome alignment</strong>)</li>
<li>SO MANY LTRs in the sex chr for the paradise grove bird
<ul>
<li>hundreds vs tens for the autosomes</li>
</ul></li>
</ul>
<p>How to align over complex / long repeats?</p>
<ul>
<li>optical mapping helps get longer (20kb -&gt; 500kb or so)</li>
</ul>
</div>
<div id="raquel-garcia-perez-juan" class="section level2">
<h2>Raquel Garcia-Perez (Juan)</h2>
<p>Recent evolution of the epigenetic regulatory landscape in human and other primates</p>
<p>Comparative epigenomics</p>
<p>WGBS ChIP-seq (main ones) ATAC-Seq RNA-Seq</p>
<p>Across primates</p>
<p><strong>What tissue(s)???</strong></p>
<p>chromHMM -&gt; 16 state model, orthologous regulatory elements</p>
<p>Gene annotation + 3C chromatin interactions = match enhancers with promoters or enhancers</p>
<p>trying to model gene expression differences with histone marks</p>
<ul>
<li>semi-succesful (explain 40% of variance)</li>
</ul>
</div>
</div>

    </div>

    


<div class="article-tags">
  
  <a class="btn btn-primary btn-outline" href="/./tags/conference/">conference</a>
  
  <a class="btn btn-primary btn-outline" href="/./tags/gi2018/">GI2018</a>
  
  <a class="btn btn-primary btn-outline" href="/./tags/bioinformatics/">bioinformatics</a>
  
  <a class="btn btn-primary btn-outline" href="/./tags/talks/">talks</a>
  
</div>




    
    
    <div class="article-widget">
      <div class="hr-light"></div>
      <h3>Related</h3>
      <ul>
        
        <li><a href="/post/gi2018-day-two/">#GI2018 - Day Two</a></li>
        
        <li><a href="/post/gi2018-day-one/">#GI2018 - Day One</a></li>
        
        <li><a href="/post/bog18-notes/">#BoG18: Talk Notes</a></li>
        
        <li><a href="/post/quick-guide-to-gene-name-conversion/">Quick Guide to Gene Name Conversion</a></li>
        
        <li><a href="/post/simple-heatmaps-with-complexheatmaps/">Let&#39;s Plot 6: Simple guide to heatmaps with ComplexHeatmaps</a></li>
        
      </ul>
    </div>
    

    

    
<section id="comments">
  <div id="disqus_thread"></div>
<script>
    var disqus_config = function () {
    
    
    
    };
    (function() {
        if (["localhost", "127.0.0.1"].indexOf(window.location.hostname) != -1) {
            document.getElementById('disqus_thread').innerHTML = 'Disqus comments not available by default when the website is previewed locally.';
            return;
        }
        var d = document, s = d.createElement('script'); s.async = true;
        s.src = '//' + "davemcg-gitub-io" + '.disqus.com/embed.js';
        s.setAttribute('data-timestamp', +new Date());
        (d.head || d.body).appendChild(s);
    })();
</script>
<noscript>Please enable JavaScript to view the <a href="https://disqus.com/?ref_noscript">comments powered by Disqus.</a></noscript>
<a href="https://disqus.com" class="dsq-brlink">comments powered by <span class="logo-disqus">Disqus</span></a>
</section>



  </div>
</article>

<footer class="site-footer">
  <div class="container">
    <p class="powered-by">

      &copy; 2018 &middot; 

      Powered by the
      <a href="https://sourcethemes.com/academic/" target="_blank" rel="noopener">Academic theme</a> for
      <a href="https://gohugo.io" target="_blank" rel="noopener">Hugo</a>.

      <span class="pull-right" aria-hidden="true">
        <a href="#" id="back_to_top">
          <span class="button_icon">
            <i class="fa fa-chevron-up fa-2x"></i>
          </span>
        </a>
      </span>

    </p>
  </div>
</footer>


<div id="modal" class="modal fade" role="dialog">
  <div class="modal-dialog">
    <div class="modal-content">
      <div class="modal-header">
        <button type="button" class="close btn-large" data-dismiss="modal">&times;</button>
        <h4 class="modal-title">Cite</h4>
      </div>
      <div>
        <pre><code class="modal-body tex"></code></pre>
      </div>
      <div class="modal-footer">
        <a class="btn btn-primary btn-outline js-copy-cite" href="#" target="_blank">
          <i class="fa fa-copy"></i> Copy
        </a>
        <a class="btn btn-primary btn-outline js-download-cite" href="#" target="_blank">
          <i class="fa fa-download"></i> Download
        </a>
        <div id="modal-error"></div>
      </div>
    </div>
  </div>
</div>

    

    
    
    <script id="dsq-count-scr" src="//davemcg-gitub-io.disqus.com/count.js" async></script>
    

    

    
    <script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.2.1/jquery.min.js" integrity="sha512-3P8rXCuGJdNZOnUx/03c1jOTnMn3rP63nBip5gOP2qmUh5YAdVAvFZ1E+QLZZbC1rtMrQb+mah3AfYW11RUrWA==" crossorigin="anonymous"></script>
    <script src="https://cdnjs.cloudflare.com/ajax/libs/jquery.imagesloaded/4.1.3/imagesloaded.pkgd.min.js" integrity="sha512-umsR78NN0D23AzgoZ11K7raBD+R6hqKojyBZs1w8WvYlsI+QuKRGBx3LFCwhatzBunCjDuJpDHwxD13sLMbpRA==" crossorigin="anonymous"></script>
    <script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.3.7/js/bootstrap.min.js" integrity="sha512-iztkobsvnjKfAtTNdHkGVjAYTrrtlC7mGp/54c40wowO7LhURYl3gVzzcEqGl/qKXQltJ2HwMrdLcNUdo+N/RQ==" crossorigin="anonymous"></script>
    <script src="https://cdnjs.cloudflare.com/ajax/libs/jquery.isotope/3.0.4/isotope.pkgd.min.js" integrity="sha512-VDBOIlDbuC4VWxGJNmuFRQ0Li0SKkDpmGyuhAG5LTDLd/dJ/S0WMVxriR2Y+CyPL5gzjpN4f/6iqWVBJlht0tQ==" crossorigin="anonymous"></script>
    <script src="https://cdnjs.cloudflare.com/ajax/libs/fancybox/3.2.5/jquery.fancybox.min.js" integrity="sha256-X5PoE3KU5l+JcX+w09p/wHl9AzK333C4hJ2I9S5mD4M=" crossorigin="anonymous"></script>
    
    
    <script src="/js/hugo-academic.js"></script>
    

    
    
      
      
      <script src="//cdnjs.cloudflare.com/ajax/libs/highlight.js/9.12.0/highlight.min.js" integrity="sha256-/BfiIkHlHoVihZdc6TFuj7MmJ0TWcWsMXkeDFwhi0zw=" crossorigin="anonymous"></script>
      

      

      

      <script>hljs.initHighlightingOnLoad();</script>
    

    
    
    <script type="text/x-mathjax-config">
        MathJax.Hub.Config({ tex2jax: { inlineMath: [['$','$'], ['\\(','\\)']] } });
    </script>
    
    <script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.1/MathJax.js?config=TeX-AMS_CHTML" integrity="sha512-tOav5w1OjvsSJzePRtt2uQPFwBoHt1VZcUq8l8nm5284LEKE9FSJBQryzMBzHxY5P0zRdNqEcpLIRVYFNgu1jw==" crossorigin="anonymous"></script>
    
    

  </body>
</html>

