<!DOCTYPE html>
<html lang="en-us">
<head>

  <meta charset="utf-8">
  <meta name="viewport" content="width=device-width, initial-scale=1">
  <meta http-equiv="X-UA-Compatible" content="IE=edge">
  <meta name="theme" content="hugo-academic">
  <meta name="generator" content="Hugo 0.40" />
  <meta name="author" content="David McGaughey">

  
  
  
  
    
      
    
  
  <meta name="description" content="Day 2 - Session 6 - TRANSCRIPTOMICS The functional iso-transcriptomics analysis framework to assess the functional impact of alternative isoform usage Multi-resolution, interactive, atlas-scale integration of single-cell assays and experiments Efficient and robust transcriptome reconstruction from long-read RNA-seq alignments Deconvolving the pervasive transcription from jumping genes in RNA-seq and unveiling their role in tumors Alignment and mapping methodology influence transcript abundance estimation A technology-agnostic long-read analysis pipeline for transcriptome discovery and quantification Quantifying isoform expression in single-cell RNA-seq data with STARsolo-Quant Full-length transcript characterization for single cell RNA-seq analysis   Day 2 - Session 6 - TRANSCRIPTOMICS Genome Informatics 2019 at CSHL">

  
  <link rel="alternate" hreflang="en-us" href="/./post/gi2019-day02-night/">

  


  

  
  
  <meta name="theme-color" content="#0095eb">
  
  
  
  
    
  
  
    
    
      
        <link rel="stylesheet" href="//cdnjs.cloudflare.com/ajax/libs/highlight.js/9.12.0/styles/github.min.css">
      
    
  
  
  <link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.3.7/css/bootstrap.min.css" integrity="sha512-6MXa8B6uaO18Hid6blRMetEIoPqHf7Ux1tnyIQdpt9qI5OACx7C+O3IVTr98vwGnlcg0LOLa02i9Y1HpVhlfiw==" crossorigin="anonymous">
  <link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/academicons/1.8.6/css/academicons.min.css" integrity="sha256-uFVgMKfistnJAfoCUQigIl+JfUaP47GrRKjf6CTPVmw=" crossorigin="anonymous">
  <link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/4.7.0/css/font-awesome.min.css" integrity="sha512-SfTiTlX6kk+qitfevl/7LibUOeJWlt9rbyDn92a1DqWOw9vWG2MFoays0sgObmWazO5BQPiFucnnEAjpAB+/Sw==" crossorigin="anonymous">
  <link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/fancybox/3.2.5/jquery.fancybox.min.css" integrity="sha256-ygkqlh3CYSUri3LhQxzdcm0n1EQvH2Y+U5S2idbLtxs=" crossorigin="anonymous">
  
  
  
  
  <link rel="stylesheet" href="//fonts.googleapis.com/css?family=Montserrat:400,700%7cRoboto:400,400italic,700%7cRoboto&#43;Mono">
  
  <link rel="stylesheet" href="/styles.css">
  

  
    <script>
      window.ga=window.ga||function(){(ga.q=ga.q||[]).push(arguments)};ga.l=+new Date;
      ga('create', 'UA-113461893-1', 'auto');
      
      ga('require', 'eventTracker');
      ga('require', 'outboundLinkTracker');
      ga('require', 'urlChangeTracker');
      ga('send', 'pageview');
    </script>
    <script async src="//www.google-analytics.com/analytics.js"></script>
    
    <script async src="https://cdnjs.cloudflare.com/ajax/libs/autotrack/2.4.1/autotrack.js" integrity="sha512-HUmooslVKj4m6OBu0OgzjXXr+QuFYy/k7eLI5jdeEy/F4RSgMn6XRWRGkFi5IFaFgy7uFTkegp3Z0XnJf3Jq+g==" crossorigin="anonymous"></script>
    
  

  
  <link rel="alternate" href="/./index.xml" type="application/rss+xml" title="eye Bioinformatician">
  <link rel="feed" href="/./index.xml" type="application/rss+xml" title="eye Bioinformatician">
  

  <link rel="manifest" href="/site.webmanifest">
  <link rel="icon" type="image/png" href="/img/icon.png">
  <link rel="apple-touch-icon" type="image/png" href="/img/icon-192.png">

  <link rel="canonical" href="/./post/gi2019-day02-night/">

  <meta property="twitter:card" content="summary_large_image">
  
  <meta property="twitter:site" content="@https://twitter.com/David_McGaughey">
  <meta property="twitter:creator" content="@https://twitter.com/David_McGaughey">
  
  <meta property="og:site_name" content="eye Bioinformatician">
  <meta property="og:url" content="/./post/gi2019-day02-night/">
  <meta property="og:title" content="#GI2019 2019 11 07 Night Session | eye Bioinformatician">
  <meta property="og:description" content="Day 2 - Session 6 - TRANSCRIPTOMICS The functional iso-transcriptomics analysis framework to assess the functional impact of alternative isoform usage Multi-resolution, interactive, atlas-scale integration of single-cell assays and experiments Efficient and robust transcriptome reconstruction from long-read RNA-seq alignments Deconvolving the pervasive transcription from jumping genes in RNA-seq and unveiling their role in tumors Alignment and mapping methodology influence transcript abundance estimation A technology-agnostic long-read analysis pipeline for transcriptome discovery and quantification Quantifying isoform expression in single-cell RNA-seq data with STARsolo-Quant Full-length transcript characterization for single cell RNA-seq analysis   Day 2 - Session 6 - TRANSCRIPTOMICS Genome Informatics 2019 at CSHL">
  <meta property="og:locale" content="en-us">
  
  <meta property="article:published_time" content="2019-11-07T00:00:00&#43;00:00">
  
  <meta property="article:modified_time" content="2019-11-07T00:00:00&#43;00:00">
  

  
  

  <title>#GI2019 2019 11 07 Night Session | eye Bioinformatician</title>

</head>
<body id="top" data-spy="scroll" data-target="#toc" data-offset="71" >

<nav class="navbar navbar-default navbar-fixed-top" id="navbar-main">
  <div class="container">

    
    <div class="navbar-header">
      
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse"
              data-target=".navbar-collapse" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      
      <a class="navbar-brand" href="/">eye Bioinformatician</a>
    </div>

    
    <div class="collapse navbar-collapse">

      
      
      <ul class="nav navbar-nav navbar-right">
        

        
        
        
        
        
          
        

        <li class="nav-item">
          <a href="/#about">
            
            <span>Home</span>
            
          </a>
        </li>

        
        

        
        <li class="dropdown">
          <a href="#" class="dropdown-toggle" data-toggle="dropdown" aria-haspopup="true">
            
            <span>Posts</span>
            
            <span class="caret"></span>
          </a>
          <ul class="dropdown-menu">
            
            <li class="nav-item">
              <a href="/#posts">
                
                <span>Recent Posts</span>
                
              </a>
            </li>
            
            <li class="nav-item">
              <a href="/post">
                
                <span>All Posts</span>
                
              </a>
            </li>
            
          </ul>
        </li>

        
        

        
        
        
        
        
          
        

        <li class="nav-item">
          <a href="/#projects">
            
            <span>Projects</span>
            
          </a>
        </li>

        
        

        
        
        
        
        
          
        

        <li class="nav-item">
          <a href="/#culture">
            
            <span>Culture</span>
            
          </a>
        </li>

        
        

        
        
        
        
        
          
        

        <li class="nav-item">
          <a href="/#publications_selected">
            
            <span>Publications</span>
            
          </a>
        </li>

        
        
      

      
      </ul>

    </div>
  </div>
</nav>


<article class="article" itemscope itemtype="http://schema.org/Article">

  


  <div class="article-container">
    <h1 itemprop="name">#GI2019 2019 11 07 Night Session</h1>

    

<div class="article-metadata">

  <span class="article-date">
    
    <time datetime="2019-11-07 00:00:00 &#43;0000 UTC" itemprop="datePublished dateModified">
      Nov 7, 2019
    </time>
  </span>
  <span itemscope itemprop="author publisher" itemtype="http://schema.org/Person">
    <meta itemprop="name" content="David McGaughey">
  </span>

  
  <span class="middot-divider"></span>
  <span class="article-reading-time">
    6 min read
  </span>
  

  
  
  <span class="middot-divider"></span>
  <a href="/./post/gi2019-day02-night/#disqus_thread"></a>
  

  
  
  
  <span class="middot-divider"></span>
  <span class="article-categories">
    <i class="fa fa-folder"></i>
    
    <a href="/./categories/bioinformatics/">bioinformatics</a
    >, 
    
    <a href="/./categories/gi2019/">gi2019</a
    >, 
    
    <a href="/./categories/conferences/">conferences</a
    >, 
    
    <a href="/./categories/talk/">talk</a
    >
    
  </span>
  
  

  
  
<div class="share-box" aria-hidden="true">
  <ul class="share">
    <li>
      <a class="twitter"
         href="https://twitter.com/intent/tweet?text=%23GI2019%202019%2011%2007%20Night%20Session&amp;url=%2f.%2fpost%2fgi2019-day02-night%2f"
         target="_blank" rel="noopener">
        <i class="fa fa-twitter"></i>
      </a>
    </li>
    <li>
      <a class="facebook"
         href="https://www.facebook.com/sharer.php?u=%2f.%2fpost%2fgi2019-day02-night%2f"
         target="_blank" rel="noopener">
        <i class="fa fa-facebook"></i>
      </a>
    </li>
    <li>
      <a class="linkedin"
         href="https://www.linkedin.com/shareArticle?mini=true&amp;url=%2f.%2fpost%2fgi2019-day02-night%2f&amp;title=%23GI2019%202019%2011%2007%20Night%20Session"
         target="_blank" rel="noopener">
        <i class="fa fa-linkedin"></i>
      </a>
    </li>
    <li>
      <a class="weibo"
         href="http://service.weibo.com/share/share.php?url=%2f.%2fpost%2fgi2019-day02-night%2f&amp;title=%23GI2019%202019%2011%2007%20Night%20Session"
         target="_blank" rel="noopener">
        <i class="fa fa-weibo"></i>
      </a>
    </li>
    <li>
      <a class="email"
         href="mailto:?subject=%23GI2019%202019%2011%2007%20Night%20Session&amp;body=%2f.%2fpost%2fgi2019-day02-night%2f">
        <i class="fa fa-envelope"></i>
      </a>
    </li>
  </ul>
</div>


  

</div>


    <div class="article-style" itemprop="articleBody">
      <div id="TOC">
<ul>
<li><a href="#day-2---session-6---transcriptomics">Day 2 - Session 6 - TRANSCRIPTOMICS</a></li>
<li><a href="#the-functional-iso-transcriptomics-analysis-framework-to-assess-the-functional-impact-of-alternative-isoform-usage">The functional iso-transcriptomics analysis framework to assess the functional impact of alternative isoform usage</a></li>
<li><a href="#multi-resolution-interactive-atlas-scale-integration-of-single-cell-assays-and-experiments">Multi-resolution, interactive, atlas-scale integration of single-cell assays and experiments</a></li>
<li><a href="#efficient-and-robust-transcriptome-reconstruction-from-long-read-rna-seq-alignments">Efficient and robust transcriptome reconstruction from long-read RNA-seq alignments</a></li>
<li><a href="#deconvolving-the-pervasive-transcription-from-jumping-genes-in-rna-seq-and-unveiling-their-role-in-tumors">Deconvolving the pervasive transcription from jumping genes in RNA-seq and unveiling their role in tumors</a></li>
<li><a href="#alignment-and-mapping-methodology-influence-transcript-abundance-estimation">Alignment and mapping methodology influence transcript abundance estimation</a></li>
<li><a href="#a-technology-agnostic-long-read-analysis-pipeline-for-transcriptome-discovery-and-quantification">A technology-agnostic long-read analysis pipeline for transcriptome discovery and quantification</a></li>
<li><a href="#quantifying-isoform-expression-in-single-cell-rna-seq-data-with-starsolo-quant">Quantifying isoform expression in single-cell RNA-seq data with STARsolo-Quant</a></li>
<li><a href="#full-length-transcript-characterization-for-single-cell-rna-seq-analysis">Full-length transcript characterization for single cell RNA-seq analysis</a></li>
</ul>
</div>

<div id="day-2---session-6---transcriptomics" class="section level1">
<h1>Day 2 - Session 6 - TRANSCRIPTOMICS</h1>
<p>Genome Informatics 2019 at CSHL</p>
<p><strong>Bold</strong> is the speaker</p>
</div>
<div id="the-functional-iso-transcriptomics-analysis-framework-to-assess-the-functional-impact-of-alternative-isoform-usage" class="section level1">
<h1>The functional iso-transcriptomics analysis framework to assess the functional impact of alternative isoform usage</h1>
<p>Lorena de la Fuente, Manuel Tardáguila, Hector del Risco, Angeles Arzalluz, Francisco Pardo Palacios, Pedro Salguero, Cristina Martín, Sonia Tarazona, <strong>Ana Conesa.</strong></p>
<p>Missed most of this talk to FaceTime goodnight to my kids.</p>
<p>Not certain what is happening - is this some sort of analysis package?</p>
<p>OK, went to twitter.</p>
<p><a href="http://tappas.org" class="uri">http://tappas.org</a></p>
<p>GUI system for analysis?</p>
</div>
<div id="multi-resolution-interactive-atlas-scale-integration-of-single-cell-assays-and-experiments" class="section level1">
<h1>Multi-resolution, interactive, atlas-scale integration of single-cell assays and experiments</h1>
<p><strong>Akshay Balsubramani</strong>, Anshul Kundaje.</p>
<p>So many sc data types</p>
<ul>
<li>CITE, FACS, MERFISH, scATAC…..etc forever</li>
</ul>
<p>Want to unify datasets from different experiments</p>
<p>Using Tabula Muris data</p>
<ul>
<li>Has both Droplet and well based data with rougly equal n</li>
<li>also both annotated with similar-ish cells</li>
</ul>
<ol style="list-style-type: decimal">
<li>Connect cells of similar cell types (anchors)</li>
<li>Impute measured features within clusters</li>
<li>Refine co-embeddings on subpopulations as needed</li>
</ol>
<p>(I think only this last point may be new? I think many do the first two…)</p>
<p>Use “Batch mxiing entropy” to quantify whether tissue types are clustering together</p>
<p>Show how Seurat works OK (“overmixes” some cells)</p>
<p>“Musica”</p>
<p>kNN (nearest neighbors)</p>
<p>Need existing labels to integrate? This would be a deal breaker for many….</p>
<p>Is this a package/software tool???</p>
<p>Want to align all single cell data, agnostic of tech.</p>
<p>No software.</p>
<p>So impossible to evaluate whether this is useful.</p>
</div>
<div id="efficient-and-robust-transcriptome-reconstruction-from-long-read-rna-seq-alignments" class="section level1">
<h1>Efficient and robust transcriptome reconstruction from long-read RNA-seq alignments</h1>
<p>Sam Kovaka, Aleksey V. Zimin, Geo M. Pertea, Roham Razaghi, Steven L. Salzberg, <strong>Mihaela Pertea.</strong></p>
<p>Going over StringTie, which makes tx-ome from short-read (via graph) and high quality ref genome.</p>
<p>Short-reads have two big limitations</p>
<ul>
<li>rarely span more than two exons</li>
<li>multi-mapping</li>
</ul>
<p>Limitations of long read</p>
<ul>
<li>error rate (PacBio simualated looks WAY worse than I would have expected….not certain why)</li>
<li>low throughput (can only get info on highly expressed genes)</li>
</ul>
<p>Effect of noise on the splice graph</p>
<ul>
<li>makes the graph crazy complicated very quickly</li>
<li>which is why StringTie has so much trouble with long-read data</li>
</ul>
<p>StringTie2</p>
<ul>
<li>represent splice graph as matrix (now is sparse matrix unlike StringTie1)</li>
<li>correct splice site error by using consensus</li>
<li>remove edges from splice graph which are “poorly” supported</li>
<li>handles downsampled truth annotations pretty well (StringTie has guided mode which uses known genes to build transcripts)</li>
</ul>
</div>
<div id="deconvolving-the-pervasive-transcription-from-jumping-genes-in-rna-seq-and-unveiling-their-role-in-tumors" class="section level1">
<h1>Deconvolving the pervasive transcription from jumping genes in RNA-seq and unveiling their role in tumors</h1>
<p><strong>Fabio Navarro</strong>, Jacob Hoops, Lauren Bellfy, Eliza Cerveira, Qihui Zhu, Chengsheng Zhang, Charles Lee, Mark Gersten.</p>
<p>More than half of the genome is repetitive</p>
<ul>
<li>SINE, LINE, LTRs, Simple repeats</li>
</ul>
<p>Focus on LINE1 (most common, right?)</p>
<p>Estimation of TE transcription in genome is hard</p>
<ul>
<li>tossing out multi-mappers throws out real data</li>
</ul>
<p>TeXP: <a href="https://journals.plos.org/ploscompbiol/article?id=10.1371/journal.pcbi.1007293" class="uri">https://journals.plos.org/ploscompbiol/article?id=10.1371/journal.pcbi.1007293</a></p>
<p>How can we reliably quantify TE tx?</p>
</div>
<div id="alignment-and-mapping-methodology-influence-transcript-abundance-estimation" class="section level1">
<h1>Alignment and mapping methodology influence transcript abundance estimation</h1>
<p>Avi Srivastava, Laraib Malik, Hirak Sarkar, Mohsen Zakeri, Fatemeh Almodaresi, Charlotte Soneson, Michael I. Love, Carl Kingsford, <strong>Rob Patro.</strong></p>
<p>Starts of with BO Li RSEM equation to estimate abundance</p>
<p>This talk focuses on mapping approaches and how they influence tx quant</p>
<p>“The sample is not the reference”</p>
<p>Tested custom tx-ome vs ref</p>
<p>Bowtie2 and STAR have largest drop in performance</p>
<ul>
<li>still difference isn’t huge</li>
</ul>
<p>Used real data</p>
<ul>
<li>both bulk and single cell (full length)</li>
<li>no ground truth
<ul>
<li>hand check alignments</li>
</ul></li>
<li>found some themes….and we are on the next slide</li>
</ul>
<p>Made “decoy-aware” tx-ome</p>
<ul>
<li>mask exonic regions</li>
<li>take rest and find similar regions, use this for decoy</li>
<li>all reads that map best to decoy are tossed</li>
</ul>
<p>“Oracle”</p>
<ul>
<li>use bowtie2 against txome and STAR against genome (aware of tx)</li>
<li>if some filter thingy happens, then toss read</li>
</ul>
<p>sorry these notes sucks….too busy listening…to this information dense talk….</p>
<p>STAR seems the best…across the testing</p>
<p><a href="https://www.biorxiv.org/content/10.1101/657874v2" class="uri">https://www.biorxiv.org/content/10.1101/657874v2</a></p>
</div>
<div id="a-technology-agnostic-long-read-analysis-pipeline-for-transcriptome-discovery-and-quantification" class="section level1">
<h1>A technology-agnostic long-read analysis pipeline for transcriptome discovery and quantification</h1>
<p><strong>Dana Wyman</strong>, Gabriela Balderrama-Gutierrez, Fairlie Reese, Shan
Jiang, Brian Williams, Barbara Wold, Ali Mortazavi.</p>
<p>TALON!</p>
<p><em>quantification</em> of long read RNA-sdq</p>
<p>also makes high quality annotations</p>
<p>can add new data without re-running existing</p>
<p>used SQANTI tx category names</p>
<p>short tx can (often) be artifacts</p>
<ul>
<li>rna degradation, polyA priming issues</li>
</ul>
<p>lots of filtering to toss likely erroneous tx</p>
<p>PacBio is consistent with PacBio (pearson 0.9) at gene and tx level (0.7)</p>
<ul>
<li>can also reliably detect egnes above TPM 5</li>
</ul>
<p>Direct-RNA ONT also reproducible</p>
<ul>
<li>a bit worse than PacBio</li>
<li>misses many lower read level genes</li>
<li>maybe because reads that get stuck in pore look like short tx</li>
<li>fewer RTPCR artifacts (as expected…since there’s no RTPCR)</li>
</ul>
<p>github.com/dewyman/TALON
github.com/dewyman/TranscriptClean</p>
</div>
<div id="quantifying-isoform-expression-in-single-cell-rna-seq-data-with-starsolo-quant" class="section level1">
<h1>Quantifying isoform expression in single-cell RNA-seq data with STARsolo-Quant</h1>
<p><strong>Nathan Castro-Pacheco</strong>, Alexander Dobin.</p>
<p>Going over droplet based scRNA</p>
<ul>
<li>e.g. DropSeq, 10X</li>
<li>high dropout, short reads</li>
</ul>
<p>Discussing tx isoforms….not many quantify</p>
<ul>
<li>exception: TCC (Ntranos….Pachter)</li>
</ul>
<p>Want to do tx isoform quant with sc data</p>
<p>But short reads at 3’ end…don’t tend to map uniquely</p>
<p>Propose merging clusters of scRNA to get higher counts</p>
<p>quant -&gt; cluster -&gt; do quant again with pseudo bulk? (this is me guessing)</p>
<p>EM based quant, models 3’ end bias</p>
<p>STARsolo -&gt; cluster -&gt; STARsolo-Quant -&gt; combine reads from each cluster -&gt; EMaximization (well I wasn’t super far off)</p>
<p>Showing a bunch of case studies with individual genes and the effect of low counts</p>
<p>If they a bunch of tx have identical end, then you are screwed (so much multi-mapping) - they propose just merging these together</p>
<p>With splatter (simulated data) spearman R 0.93</p>
</div>
<div id="full-length-transcript-characterization-for-single-cell-rna-seq-analysis" class="section level1">
<h1>Full-length transcript characterization for single cell RNA-seq analysis</h1>
<p><strong>Elizabeth Tseng</strong>, Jason Underwood.</p>
<p><span class="citation">@magdoll</span></p>
<p>Iso-Seq</p>
<ul>
<li>full length RNA-seq</li>
<li>uses CCS</li>
<li>8 passes get QV30</li>
</ul>
<p>Did Illumina and PacBio single cell</p>
<ul>
<li>droplet? what? SMARTseq? didn’t say….</li>
<li>anyways, spearman R 0.96</li>
</ul>
<p>scRNA is shorter than bulk (for tx length)</p>
<ul>
<li>clear left-ward shift in scRNA data</li>
</ul>
<p>github.com/magdoll/sqanti2</p>
<p>Talks about orthogonal valdation of full length, 5’, 3’ matching</p>
<p>How to represent isoforms</p>
<ul>
<li>so many (&gt;200k, depending how you count)</li>
<li>top 3 for each gene?</li>
<li>just unique ORFs?
<ul>
<li>uh, what? is that a good idea? I guess I’m not certain why you HAVE to trim at this step</li>
</ul></li>
</ul>
</div>

    </div>

    


<div class="article-tags">
  
  <a class="btn btn-primary btn-outline" href="/./tags/bioinformatics/">bioinformatics</a>
  
  <a class="btn btn-primary btn-outline" href="/./tags/gi2019/">gi2019</a>
  
  <a class="btn btn-primary btn-outline" href="/./tags/conferences/">conferences</a>
  
  <a class="btn btn-primary btn-outline" href="/./tags/talk/">talk</a>
  
</div>




    
    
    <div class="article-widget">
      <div class="hr-light"></div>
      <h3>Related</h3>
      <ul>
        
        <li><a href="/post/gi2019-day02/">#GI2019 2019 11 07</a></li>
        
        <li><a href="/post/gi2019-day02-night/">#GI2019 2019 11 08 Morning Session</a></li>
        
        <li><a href="/post/gi2019-day01/">#GI2019 2019 11 06</a></li>
        
        <li><a href="/post/one-developer-portal-genesis/">One Developer Portal: eyeIntegration Genesis</a></li>
        
        <li><a href="/post/one-developer-portal-optimization/">One Developer Portal: eyeIntegration Web Optimization</a></li>
        
      </ul>
    </div>
    

    

    
<section id="comments">
  <div id="disqus_thread"></div>
<script>
    var disqus_config = function () {
    
    
    
    };
    (function() {
        if (["localhost", "127.0.0.1"].indexOf(window.location.hostname) != -1) {
            document.getElementById('disqus_thread').innerHTML = 'Disqus comments not available by default when the website is previewed locally.';
            return;
        }
        var d = document, s = d.createElement('script'); s.async = true;
        s.src = '//' + "davemcg-gitub-io" + '.disqus.com/embed.js';
        s.setAttribute('data-timestamp', +new Date());
        (d.head || d.body).appendChild(s);
    })();
</script>
<noscript>Please enable JavaScript to view the <a href="https://disqus.com/?ref_noscript">comments powered by Disqus.</a></noscript>
<a href="https://disqus.com" class="dsq-brlink">comments powered by <span class="logo-disqus">Disqus</span></a>
</section>



  </div>
</article>

<footer class="site-footer">
  <div class="container">
    <p class="powered-by">

      &copy; 2018 &middot; 

      Powered by the
      <a href="https://sourcethemes.com/academic/" target="_blank" rel="noopener">Academic theme</a> for
      <a href="https://gohugo.io" target="_blank" rel="noopener">Hugo</a>.

      <span class="pull-right" aria-hidden="true">
        <a href="#" id="back_to_top">
          <span class="button_icon">
            <i class="fa fa-chevron-up fa-2x"></i>
          </span>
        </a>
      </span>

    </p>
  </div>
</footer>


<div id="modal" class="modal fade" role="dialog">
  <div class="modal-dialog">
    <div class="modal-content">
      <div class="modal-header">
        <button type="button" class="close btn-large" data-dismiss="modal">&times;</button>
        <h4 class="modal-title">Cite</h4>
      </div>
      <div>
        <pre><code class="modal-body tex"></code></pre>
      </div>
      <div class="modal-footer">
        <a class="btn btn-primary btn-outline js-copy-cite" href="#" target="_blank">
          <i class="fa fa-copy"></i> Copy
        </a>
        <a class="btn btn-primary btn-outline js-download-cite" href="#" target="_blank">
          <i class="fa fa-download"></i> Download
        </a>
        <div id="modal-error"></div>
      </div>
    </div>
  </div>
</div>

    

    
    
    <script id="dsq-count-scr" src="//davemcg-gitub-io.disqus.com/count.js" async></script>
    

    

    
    <script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.2.1/jquery.min.js" integrity="sha512-3P8rXCuGJdNZOnUx/03c1jOTnMn3rP63nBip5gOP2qmUh5YAdVAvFZ1E+QLZZbC1rtMrQb+mah3AfYW11RUrWA==" crossorigin="anonymous"></script>
    <script src="https://cdnjs.cloudflare.com/ajax/libs/jquery.imagesloaded/4.1.3/imagesloaded.pkgd.min.js" integrity="sha512-umsR78NN0D23AzgoZ11K7raBD+R6hqKojyBZs1w8WvYlsI+QuKRGBx3LFCwhatzBunCjDuJpDHwxD13sLMbpRA==" crossorigin="anonymous"></script>
    <script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.3.7/js/bootstrap.min.js" integrity="sha512-iztkobsvnjKfAtTNdHkGVjAYTrrtlC7mGp/54c40wowO7LhURYl3gVzzcEqGl/qKXQltJ2HwMrdLcNUdo+N/RQ==" crossorigin="anonymous"></script>
    <script src="https://cdnjs.cloudflare.com/ajax/libs/jquery.isotope/3.0.4/isotope.pkgd.min.js" integrity="sha512-VDBOIlDbuC4VWxGJNmuFRQ0Li0SKkDpmGyuhAG5LTDLd/dJ/S0WMVxriR2Y+CyPL5gzjpN4f/6iqWVBJlht0tQ==" crossorigin="anonymous"></script>
    <script src="https://cdnjs.cloudflare.com/ajax/libs/fancybox/3.2.5/jquery.fancybox.min.js" integrity="sha256-X5PoE3KU5l+JcX+w09p/wHl9AzK333C4hJ2I9S5mD4M=" crossorigin="anonymous"></script>
    
    
    <script src="/js/hugo-academic.js"></script>
    

    
    
      
      
      <script src="//cdnjs.cloudflare.com/ajax/libs/highlight.js/9.12.0/highlight.min.js" integrity="sha256-/BfiIkHlHoVihZdc6TFuj7MmJ0TWcWsMXkeDFwhi0zw=" crossorigin="anonymous"></script>
      

      

      

      <script>hljs.initHighlightingOnLoad();</script>
    

    
    
    <script type="text/x-mathjax-config">
        MathJax.Hub.Config({ tex2jax: { inlineMath: [['$','$'], ['\\(','\\)']] } });
    </script>
    
    <script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.1/MathJax.js?config=TeX-AMS_CHTML" integrity="sha512-tOav5w1OjvsSJzePRtt2uQPFwBoHt1VZcUq8l8nm5284LEKE9FSJBQryzMBzHxY5P0zRdNqEcpLIRVYFNgu1jw==" crossorigin="anonymous"></script>
    
    

  </body>
</html>

