---
title: Let's Plot 6: Simple guide to heatmaps with ComplexHeatmaps
author: David McGaughey
date: '2018-05-07'
slug: simple-heatmaps-with-complexheatmaps
categories:
  - Letâ€™s Plot
  - R
tags:
  - bioinformatics
  - R
  - tidyverse
  - heatmaps
output:
  blogdown::html_page:
    toc: true
---

## Introduction

Heatmaps are a core competency for a bioinformatician. They are a compact way to visually demonstrate relationships and changes in values across conditions. 

We are going to use RNA-seq data to make two kinds of common heatmaps:

1. Sample - Sample relationships
2. Gene expression changes across samples for a subset of genes

## Data processing
The two Rdata datasets we are going to use can be found here:

1. [Expression data](https://github.com/davemcg/Let_us_plot/blob/master/006_heatmaps/expression.Rdata)
2. [Metadata](https://github.com/davemcg/Let_us_plot/blob/master/006_heatmaps/metadata.Rdata)

## Load data
```{r}
load('~/git/Let_us_plot/006_heatmaps/expression.Rdata')
load('~/git/Let_us_plot/006_heatmaps/metadata.Rdata')
```

## Peek at expression
Rows are genes, columns are samples. Notice how the *number* sample names got X appended to the beginning. This is because R variable names cannot start with a number. Generally R will 'fix' your sample names during a *base* R handling.
```{r}
head(expression)
# check number of rows (genes) and column (samples)
dim(expression)
```

## Peek at metadata
Sample names with the serum conditions (Human Serum or Heat Inactivated HS) as well as the time the cells spend in the serum
```{r}
metadata
```

## Brief outline on how the RNA-seq data was processed before we see it

1. [Salmon](http://salmon.readthedocs.io)
2. [tximport](https://bioconductor.org/packages/release/bioc/html/tximport.html)
3. [DESeq2](https://bioconductor.org/packages/release/bioc/html/DESeq2.html)
4. [variance stabilizing transformation](http://www.bioconductor.org/packages//2.10/bioc/vignettes/DESeq/inst/doc/vst.pdf)
5. Filter genes to only retain high variance genes (often not necessary, but used to reduce the amount of data I share)

## Load libraries
I've rapidly become a fan of [ComplexHeatmap](https://www.bioconductor.org/packages/release/bioc/html/ComplexHeatmap.html). It is capaable of creating just about anything **and** has excellent documentation and examples. 
```{r, echo=FALSE}
library(tidyverse)
library(ComplexHeatmap)
library(viridis)
```

## Create a Sample - Sample distance heatmap
An powerful way to see sample to sample relationships is to use the expression data to arrange the samples. The distance between each sample can be calculated by taking the [euclidean distance](https://en.wikipedia.org/wiki/Euclidean_distance) between each sample. This reduces the space from a n x m matrix of values to a single number for each sample pair. 
```{r}
expression_dists <- expression %>% 
  # drop the Gene and variance column
  select(-Gene, -var) %>% 
  # flip axes as the dist() function does all of row-row relationships
  # we want samples as the rows....
  t() %>% 
  # this actually calculates the euclidean distance (other types of distance are possible - read the docs with ?dist)
  dist() %>% 
  # coerce to matrix, then data frame
  as.matrix() %>% data.frame() 

# now a 9 x 9 data frame
expression_dists %>% dim()
```

### Easy heatmap with ComplexHeatmap
I have a few problems with this plot:

1. Red / blue is more for distinguishing over / under in my opinion - we are talking about close to far
2. Don't really need to see key for the colors. Distance is unit-less.
3. We can't map between sample name and the metadata
```{r}
Heatmap(expression_dists)
```

### Complex heatmap
The metadata needs some extensive reworking to make it work with the sample distance data for the heatmap

1. `mutate` the sample to add the 'X' to match the dist data column names
2. `filter` to only keep metadata samples that are in the distance data (very common to have metadata for *all* samples)
3. `select` to grab the relevant columns (also common to have many extra columns that aren't used)
4. `mutate` again to coerce the sample order in the metadata to match the sample order in the distance data
5. `arrange` to use the order set above for the metadata (try erasing this and see what happens....)
6. `unique` to remove duplicated rows

After that the HeatmapAnnotation take the `Time` and `Serum` as a data.frame and I am manually setting the colors by grabbing colors from the magma scheme from viridis. 

```{r}

metadata_heatmap <- metadata  %>% 
  mutate(sample = paste0('X', sample)) %>% 
  filter(sample %in% colnames(expression_dists)) %>% 
  dplyr::select(sample, treatment_hours, serum) %>% 
  mutate(sample=factor(sample, levels=colnames(expression_dists))) %>% 
  arrange(sample) %>% 
  unique() 

ha_column = HeatmapAnnotation(df = data.frame(Time = metadata_heatmap$treatment_hours,
                                              Serum = metadata_heatmap$serum), 
                              col = list(Serum = c("HS" =  magma(20)[5], "HIHS" = magma(20)[7]),
                                         Time = c("24" = magma(20)[12], "48" = magma(20)[18])))


Heatmap(expression_dists, 
        col=viridis(100),
        name = 'Sample\nDistances',
        top_annotation = ha_column)
```

### Finished heatmap
Now I am going to drop the 'X' from the sample names and remove the Sample Distances key
```{r}
# substitute '' (nothing) for 'X' for the column and row names
colnames(expression_dists) <- gsub('X','',colnames(expression_dists))
rownames(expression_dists) <- gsub('X','',rownames(expression_dists))
Heatmap(expression_dists, 
        col=viridis(100),
        show_heatmap_legend = F,
        top_annotation = ha_column)
```

## Gene Heatmaps
The next common type of heatmap is to show how a set of genes changes expression between conditions

We will look at a small set of genes and see how their expression changes when put in Heat Inactivated Human Serum (HIHS) or Human Serum (HS), only at the 48 hour time point. 

First, make gene_set, HS, and HIHS sample vectors
```{r}

gene_set <- c('EP300','ABCA4','CLCN6','CRX','HIST1H1T','IGSF9','SLC16A7','MYO5C','NEURL3')

HS_samples <- metadata %>% filter(serum=='HS', treatment_hours==48) %>% pull(sample) %>% unique() %>% paste0('X',.)
HIHS_samples <- metadata %>% filter(serum=='HIHS', treatment_hours==48) %>% pull(sample) %>% unique() %>% paste0('X',.)
```

Now calculate the change in expression across the HIHS <-> HS comparison and prep the metadata
```{r}
gene_set = expression %>% filter(Gene %in% gene_set) %>% select(-var)

# HIHS is the 'reference'
# we'll look at how expression changes with you use HS
row_mean_HIHS = gene_set[,HIHS_samples] %>% rowMeans()

# now subtract the mean HIHS expression from each row,
# and only keep HS samples for the plot
heatmap_expression = gene_set[,HS_samples] - row_mean_HIHS
# add Gene back as rownames
row.names(heatmap_expression) <- gene_set$Gene
# prep metadata as above
metadata_heatmap <- metadata  %>% 
  mutate(sample = paste0('X', sample)) %>% 
  filter(sample %in% colnames(heatmap_expression)) %>% 
  dplyr::select(sample, treatment_hours, serum) %>% 
  mutate(sample=factor(sample, levels=colnames(heatmap_expression))) %>% 
  arrange(sample) %>% 
  unique() 
```

The clustering of the rows is *really* useful to quickly see what sets of genes are moving in the same direction
```{r}
ha_column = HeatmapAnnotation(df = data.frame(Time = metadata_heatmap$treatment_hours,
                                              Serum = metadata_heatmap$serum), 
                              col = list(Serum = c("HS" =  magma(20)[5], "HIHS" = magma(20)[7]),
                                         Time = c("24" = magma(20)[12], "48" = magma(20)[18])))

# Drop the X in sample names
colnames(heatmap_expression) <- gsub('X','',colnames(heatmap_expression))
Heatmap(heatmap_expression, 
        cluster_columns = FALSE,
        name = 'log2(Fold\nChange)', 
        top_annotation = ha_column)

```

### A bit simpler
You may think the annotations are superfluous, as they are all the same. If you are making just one, I would agree. Usually I am making a bunch of plots and I'd rather keep the annotations attached so I don't get confused later. 

It is easy enough to drop them.
```{r}
Heatmap(heatmap_expression, 
        cluster_columns = FALSE,
        name = 'log2(Fold\nChange)')
```

## Session Info
```{r}
devtools::session_info()
```