---
title: '#BoG18: Talk Notes'
author: David McGaughey
date: '2018-05-08'
slug: bog18-notes
categories:
  - bog18
  - bioinformatics
tags:
  - conference
  - bog18
  - bioinformatics
  - talks
output:
  blogdown::html_page:
    toc: true
---

# Intro
Very sparse and poorly written notes covering [#BoG18](https://twitter.com/search?q=%23BoG18&src=tyah). 

Typos everywhere. Things may change dramatically over time as I scan back through notes.

I've tried to respect #notwitter. Will be updated periodically. 

Speaker (Last Author)

# Genome Engineering and Genome Editing (Tuesday Night)

## Jef Boeke
### Writing Genomes

Building synthetic yeast genomes. Contig/chr one by one. All designed. *Sc2.0*

80+% complete for each of the 16. 

### "dark matter"

Can we use 'big dna' to functionally query mammalian genomes?
'Synthetic haplotypes'

Building dif combinations of haplotype blocks

synthetic hypervariation:

    - query enhancers
    - alt splicing
  
built 102kb locus (human) and put into yeast

    - built in 3kb chunks and can assemble dif combiations
    
### big dna
Can build big dna pieces
CEGS grant
will build 3 100kb+ loci / year
want community input

## Greg Findlay (Jay Shendure)
Accurate classficiation of thousands of BRCA1 variants with saturation genome editing

vous a problem
BRCA1:

  - 4243 clinvar snvs
  - >50% VOUS
    
How to functionally validate?

Use Homology-direct repair (HDR). Can engineer precise edits.

Use a library of SNVs for the HDR

Over time selecdtion removes non functional edits

Each experimente:

  - millions of cells
  - millions of sequencing reads to count SNVs
    
Variable effects at splicing junctions

  - sometimes just 2-3 bp
  - sometimes 9 base pairs
    
also have matched rna-seq data

aberrant splicing causes RNA depletion

matches up really well with clinvar designations

question: 

  - hdr effeciency rate? 10-90% effectiveness

## Stephen Levene (Andrew Fire)
eccDNA is a possible mediator of chromosomal polymorphism at multple loci

'physical chemist by training'

e. coli genome: 1 femtoliter volume

100 fold compaction problem (DNA)

~10k fold for mammalian

Techniques for DNA/chromiatin flexibility:

  - Hi-C
  - FISH
  - SLICE (Beagrie et al Nature 2017)
  
eccDNA (circular dna outside chromosome/nucleus(?))

  - unclear how it forms
  - elevated levels associatd with genome instability
  
how to capture?

  - DNA -> SDS lysis -> isolate gDNA -> cscl gradient -> bottom bit
  - or exoV treatment (leaves circular alone)
  
take pictures of the loops with high resolution microscopy

sequenced a few (**unfortunately with short read illumina**)

modeled with molecular dynamics

## David Truong (Jef Boeke)
Resurrection of Histone H3k27 Me in brewer's yeast by human prc2 and plant atxr6

human pathway reconstruction in yeast

  - avoid pleiotropy (hopefully)
  - more real than *in vitro*
  
Yeast (s. cerevisiae) lost histone mods 

**Adding them back???**

yes

humanize yeast histones

add synthetic human histones

force out wt histones with +5FOA

*20 days later* ..... one colony

Keep growing the colony out

WGS: mutations in cell cycle regulation 

  - bypassing histone cycle checks?

brewer's yeast does not have H3K27 methylation 

  - PRC2 complex (to methylate h3k27)

add the stuffs - what happens?

  - made artifical chr with PRC2 complex
    - and a slightly broken one
  - not much happens in WT yeast
    - no me changes
  - deleted H3k36me3 (might antagonize artificial chr)
    - nope
  - can you jump start with atxr6 (does mono me)?
    - yes (confirmed with mass spec)
      - not super high levels (0.054% tri me)
      
      
## Feng Zhang
Advances in genome editing technologies

two major classes of CRISPR

  - class 1 (multi subunut)
  - class 2 (single subunit crRNA-effector)
  
trying to find new class 2

  - bioinformatic screen with BLAST of cas1
  - found a bunch (Shmakov collaboration)
  
cas13

  - added into e. coli
  - modify to only edit RNA?
  
rna editing

  - reversible
  - nuclease based editing inefficient in post mitotic cells 
  - dCas13 linked with [ADAR](https://en.wikipedia.org/wiki/ADAR) (adenosine to inosine) + guideRNA --> A to I conversion in RNA
  - 90+% conversion
  - 1732 off target incidents
    - 925 off target with non-targeting guide!
    - so protein itself is not ideal....
    - identified non-binding residues of ADAR
      - mutated them
  - v2 works better
    - 18385 off target (v1) to 20 (v2)
  - still developing
  
## Molly Gasperini (Jay Shendure)
**my fav of the night**

crisprQTL mapping as a genome-wide association framework for cellular genetic screens
  
lots of guideRNA to made mutations, check for dif in expression

nuclease inactive cas9

want to test all enhancers against all genes

scRNA-seq + guideRNA (multiplex gRNA)

  - thus multiple perturbations per 'assay'
  - 15-30 / cell!
  
targeted 1,119 candidate enhancers

  - 15 guides / cell
  - 47k cells
  - 10X
  - CROP-seq 
  - works really well
  - crisprQTL usually targets closest gene
    - sometimes not....
  - matches up with histone chip-seq
  - 34.3kb average distance from enhancer <-> gene
  
** new data! **
4,801 enhancers

  - built logistic regression model on pilot to pick new candidates
  - 30 guides / cell
  - correlates with pilot

manolis kellis q:

  - why doesn't work so well? expected more
  - what about multiple SNPS / block?
  
## Eilon Sharon (Hunter Fraser)

Testing genetic var effect on fitness using precise genome editing

high throughput edigint

crispey:
cas9 retron precise parallel editing via homology

use bacterial reverse transcriptase and RNA retron to covalently link ssDNA donor to guide-tracrRNA

can insert long sequences

(yeast)

measure fitness of genetic variants (growth competition)

sequence every 2-3 gens

model at linear relative strain abundance / time (generation #)

 # missense var ~ # synonymous var for effecting fitness!

## Luca Pinello
CRISPR-SURF exploratory and interactive software for analyzing CRISPR-base tiling screens

Uncover non-coding functional regions

** Nice overview of CRISPR tiling strategy **
Mutate (tile across region) -> Measure pheno change (somehow) -> Assess (sequence gRNA)

No unified framework to analyze these kind of assays

many challenges

  - biological noise
  - sgRNA efficiencies
  - non-uniform spacing
  - perturbation / assay differences
  - epigenetic perturbation can be wide (changing 200bp or so)  

deconvolve with generalized lasso

fastq -> score -> segmentation -> deconvolution -> region ID

# Population Genomics (Wednesday morning)

## Mattias Joakobsson 

** out of my field here **

Sequence based approaches utilizing complete modern and ancient genomes to investigate early human history

Use full genomes on ancient pops

Population divergence models

```{ascii}
                      X
     +                X
     |                X
     |                X
     |                X
     |               XXXX
     |              XX  XXX
time |           XXX      XX
     |         XXX         XX
     |        XX            XX
     |
     |      A B              C
     |
     v
```

Can model whether discordant or concordant (a,b,c) over time

estimate pop divergence (time) in generations

use genes from different populations to estimate divergence 

  - 'tt method'
  
stone age humans from sourthern africa

  - 13 genomes from 3 people
  - a bit 'right' of yoruba
  - admixture with east africa **missed something**

## Jaemin Kim (Elaine Ostrander)
Genetic Selection of Athletic Success in Sport Hunting Dogs

WGS of sport hunting (10 breeds), terrier (i breeds), and 'village' dogs (unselected - an outgroup)

  - 14 million SNPs

59 genes under strong selectdion in hunting dogs (compare to terrier and village)

  - blood circulation GO terms
  - and a bunch of 'process' GO terms

ASIC3 - resistance to muscle fatigue? 

  - maybe?
  - a guess based on known gene function (I think)
  
dogs do agility performance competitions

  - made a metric to find breeds good at winning
  - WGS of 92 breeds of 299 dogs
  - ROBO1 significant 3e-4 (FDR corrected? Don't know)
    - neuronal migration, axon guidance
  - 1243 SNP chip
    - dogs classified by agility performance
    - ** do only pure breeds do agility? **
    - ROBO1 SNP AF increases with more winning breeds

racing speeds (whippet)

  - not ROBO1
  - TRPM3 (1.6e-3)
  
CDH23 - increased tolerance to loud noise and low startle reflex
  - **do hunting dogs have poor hearing?**
  
**Useful stuff maybe for competitive dog breeders**

Q (Kellis?): polymorphic nature of traits across dogs. **what's the question?**

A: complex traits, incomplete answers right now

Q (Kellis?): enrollment bias for dogs that will win

A: tried to control by grouping breeds

Q: what kind of mutations?

A: mostly noncoding (**answer in LD I guess**)

Q: project personality onto dog ... look at dog behavior/traits relating to this?

A: try to objectively test dogs (can't trust owners....)

Elaine: people developing stanrdard tests for dogs (yes, owners lie)

## Ipsita Agarwal (Molly Przeworski)
Widespread differences in the mtation spectrum of X and autosomes

Males contrigute more germline mutations than females 

  - epigenetic differences for gamete development (methylation)
  - sperm in mitosis all the time
 
```{ascii} 
+
|
|
|                         XX
|                       XXX
|                    XX
|                  X
|               XX
|             X
|       X XX
|   XXXX
|  XX
|                 XXXX
|   XXXX XXX  XXXXX
|
+------------------------+
```

Mutation rates get wider as males age (top line male, bottom female)

  - **eyeball 3x worse?**
  
GnomAD:

  - 120 million SNPs
  - 60% singleton (50%) and doubletons (10%) for variants
  
Looked for X-autosome difs

X/A diversity = (mutations(x) / all X) / (mutation(a) / all a)
(a is autosome, x is chrX)

Bootstrap test for mutation types to make null distribution

  - big shift in X (more than expected)
  - T->A and C->A more common in X versus autosome
  
Replication timing

 - inactive X has more mutations
 
Enriched of C>G (meiotic recombination / DSB) mutations

## Amnon Koren
Genetic architecture of human DNA replication origin activity

We have extensive maps of human genomic / epigenomic

But where are replication origins?

  - **yeah....good q**
  - yeast have them
    - yeast have a DnaA/OriC signatures
  - how to find?
    - many techniques - don't agree well (or at all)
    
different parts of genomic replicate at different rates
  
  - **can measure coverage across time, right**
    - yes
  - sort G / S phase
    - check coverage
  - Did in 2012 with human - but not precise enough (low resolution)
  
Is this a polymorphic trait?

  - skipped cell sorting
  - works well enough
    - and way faster to do
  - **uh, wait, this has been done already (if cells are growing)**
      - again, yes, LCL from 1000G

But still, did WGS >140 hESC lines 
  
  - oooooo, reproduces REALLY well
  - find 'master' ORI that are pretty much always present
      - crucial regions in replication?

GWAS of DNA replication timing

  - 'rtQTL' (replication timing)
  - big hit on chr7
  - 756 with FDR < 0.1
  - most fall within replication origin
    - **direct relationship, then, cool**
  - getting causal SNPs with CAVIAR
  - enriched with active chromatin states
  
perhaps some QTL stabilize TF binding motifs
  - stuff happening in motifs 

Q: look for associations with structural variation? (1000G data)

A: looking at this now

Q: cell type specific? (**thank you**)

A: 20-30% are dif across cell types (**cool**)

Q: is ORI piggybacking on enhancer / regulatory system? (**or other way around??**)

A: maybe (or other way around)

Q: cis effects - did you find any trans or pleiotropy?

A: nothing strong (spurious stuff maybe?)

Q: HiC/3C data profiles comparisons?

A: not *yet*


## Sarah Tishkoff
Novel loci associated with skin pigmentation identified in African populations

Integrative omics of copmlex traits

  - epigenomics
  - transcriptomes
  - microbiome
  - metabolomics
  - proteomics
  - genomics
  
LOTS OF POSTERS

200 WGS Aricans
  - 35 million SNPS
  - 20% novel
  
81% GWAS european

Skin color is adaptive trait
  - spectrophotometry of skin color
  - and take DNA
  - boom GWAS
  - 1600 people
  - found 8 regions
  
SLC24A5

MFSD12
  - novel
  - transmembrane transporters
  - found enhnacer activity
  - functional work!
    - KO mRNA in melanocyptes
      - get more melanin
      - colocalizes with lysosome
    - ZF KO
      - yellow gone
    - mouse KO
      - diff colors
      - looks like gr/gr mouse
        - 9bp deletion in MFSD12!
        
DDB1
  - DNA repair in UV damage
  - pigmentation in tomatoes
  - fine mapping hits TMEM128
  - luciferase assay with enhancer activity
  - huge haplotype blocks of low het in europeans/asians
    - selective sweep, near complete fixation
    
OCA2/HERC2
- exon10 SNP alt splicing
- rs1800404

convergent evolution of very dark skin
  - african and south asia
  
q: speculate about lysosome?

a: pheomelanin made in lysosome (like) structure?

q: surprised to not find var close to mcr1 
a: no

q: chimp with no hair has vitiligo?

a: dont' think so - have been assured chimps have light skin

##  Patrick Albers (Gil McVean)
Non-parametric estimation of allele age for variants in pop-scale seq data

Want to know history of allele at single locus

Genealogical approach (GEVA)

  - look for coalescent events
      - concordant and discordance allele pairs
  - use HMM to detecdt haplotypes segments
    - non parametric
  - assess allele age...not sure how
    - incorporate time some how?
    - **model with real data to pick parameters??**
    
model
  - simulation with simple pop, const size, fixed mut rate
  - compare estimate with actual
  - good cor (0.953)
  
also tested in error-filled data
  - haplotypes, errors, phasing issues
  - still works, but overestimate young alleles (think they are older)
  
ran in simons genetci diversity project and 1000G
  - oldest var ~37.5k years in AFR
  - SAS 12.5K
  - EAS 11K
  - AMR 7.5K
  - EUR 6K (why so young? - admixture)
  
cumulative coalescent decoding (CCD)
  - what frac of your genome share with another genome back in time?
  - ran this pair-wise across 1000G dataset
  
*will release a genome-wide atlas of allele age >16 million variants*

## Laura Hayward (Guy Sella)
Polygenic adaptation in response to sudden change in environment

What time scale? Mentioned Human expansion out of Africa, which is ~100k years

stabilizing selection reduces pheontypic variance (omega is width of trait distribution)
  - kingsolver et al. 2001
  
model change of phenotpye in environment change

OK...not following this talk well at all (pretty sure it's me). Laura is copiously using cartoons, which is usually works for fools like me. 

not cetain whether this stuff is driven by theory or data or both
  - no, no data
  - equations from first princples
  
conclusions:
  - polygenic adaption is rapid
  - short term: large effects drive change
  - long term: moderate effect alleles replace them
  
topol Q: is this like dogs? big changes in short term

a: no, not modeling dramatic changes like this

# Functional Genetics and Epigenomics

## Job Dekker
Folding, unfolding, and refolding genomes

How does the genome work? **structured**

Dekker et al. 2002: 3C. **I remember this paper. And totally failing at doing this tech myself**

A/B compartments....TAD.....enhancer - gene loops. One slide summarzing many publications and of cool work. 

```{ascii}
                                TAD
                                                loop (cohesin-mediated)
                 loop (cohesin-mediated)       XXXX 
                XXXX                         XX   XX
              XXX   XXX                     XX      XXX
            XX        XX                  XX           XX
          XX            XX              XXX              XX
        XX                X           XXX                  XXX
      XX                   XX        XX                      XX
    XXX                      XX    XXX                         XXX
  XX                          XXXXX                              XX
 XXXX                           X                XXXXXXXX       XXXX
    XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX       XXX X XX
ctcf                           ctcf                                 ctcf
```

Interphase / metaphase
  - interphase has structure
  - metaphase erases structure (fat diagonal of 3c map)
  
meiotic chr fold as helical nested loop arrays (helix with loops coming off)
  - Gibcus...Dekker Science 2018
  
Using ATAC-seq, cut&ruhn to assay CTCF binding patterning in interphase / metaphase

FRAP to measure stable CTCF binding in interphase / metaphase
  - unstable in metaphase
  
Q: CTCF sites occupied in mitosis - why? Nucleosomes taking over?

A: Good q, don't know. Think nucleosomes 'sliding in'

Q: **missed it**

A: protein levels of CTCF doesn't change during what phase cell is in

Q: what is keeping the promoters open during mitosis

A: don't think pol is bound....promoters more open to begin with....something bookmarking the site????

## Flora Vaccarino
Integrative multi-omics analyses of iPSC-derived brain organioids

 #notwitter
 
## Carninci
RADICL-seq: novel tech for genome-wide mapping of RNA-chromatin interactions

Many many (~20k) functiona lncRNA

But what is role? Activate genes, promoter, enhancer? Repression of genes? Establish insulation?

RADICL-seq
  - capture RNA-DNA interactions with crosslinking (formaldehyde, 1-2%)
  
Where do they map?

  - Lots in the intron
  - open chromatin
  - 380k RNA-DNA interactions
  - enriched in TF family members
  - **looks like lots of trans interactions**
    - **but way fewer than cis,as you would expect**
  - some genes like MALAT1 interact with *entire* genome
  - compartments **sort of** like TADS
    - weak cor with Hi-C (0.27)

## Johnathan Griffiths (Berthold Gottgens)
Charting the Diversification of Mammalian Cells at whole genome scale 

Gastrulation focus (mouse)

>350 whole embryos

Collected every 6 hours: e6.5 to e8.5

10X chromium, 94k cells (post QC), 15,000 UMI (median), 3.5k detected genes (median)

Big t-SNE plot

  - talk of 'direction' and 'trajectory' which I dislike for t-SNE...
  - but can back up with time point data

Clustered with association of cell types to each other

Chimera embryos

  - KO gene, but only chimerically (inject KO mESC into blastocyst)
  - **whoah**
  - can compare KO **cells** vs not KO **cells** within full atlas
  - **damn**
  
## Emma Farley
Regulatory principles governing enhancer specificity during development

Otxa (neural enhancer)

How do enhancers encode function?

Need to test in embryos across time

*Ciona*
  - have notochord, heart, nerver chord
  
enhnacer + prom + gfp -> electroporation == inside embryo

made 2.5 million synthetic enhancers (barcoded)

electroporate 100k eggs -> mRNA -> sequence (remember, we have barcodes) -> identify functional enhancers 
  
  - **pooling of embryos?**
  - **doesn't seem like it...but seems like you have too for $**
  
Can we we make inert enhancers functional with small tweaks (change to optimal seq)

  - lights up EVERYTHING
  - *need mix of optimal and sub-optimal sites to maintain proper expression*

spacing between enhancers also important

  - adding just a few bp between motifs >>> expression

interplay between spacing and motif 'strength' (canonical-ness)

and *orientation* (flipping motifs can break function)

*suboptimization* as design parameter hilarious and terrifying (great, let's make things worse to control things more precisely)

wut - SHH enhancer that causes polydactyly....points out that SNP makes binding site *better*

Deboever et al. Cell Stem Cell 2017
D'Antonion et al. Nature communication 2017

## Jake Yeung (Felix Naef)
Clock dependent chromatin topology modulates circadian transcdription and behavior

promoter - enhancer loops

cry1 24 cycle in liver (high at night, low at day)

how are enhancers used?

4c-seq on mice collected every 4 hours for 24 hours
  - contacts change over time
  
h3k27ac changes also at same location

removed that region in mouse: cry1deltaE mouse
  - clock runs faster (15 mins over 24 hours)
  - corresponding mRNA difs in cry1
  
## Minal Caliskan (Casey Brown)
Genetic and epigenetic fine mapping of complex trait associated loci in the human liver

Genotype <-> RNA-seq <-> H3K4me3 <-> H3K27ac
50 <-> 50 <-> 10 <-> 10 (people)

Found eQTLs specific to liver (vs GTEX)

Found hQTLS also (histone)

eQTL + hQTL + RNA-seq to 'fine map' GWAS loci

  - blood pressure
  - coronary artery disease

## Parisa Razaz (Talkwoski)
Tissue-specific molecular sigmnature of 16p11.2 reciprocal genomic disorder

engineer rgd with CRISPR (make microdels and microdups)

iPSCs models and mouse models with tx profiling (brain and not brain)

# Evolutionary and Non-human genomics

## Monica Justice

## Arang Rhie (Erich Jarvis)

## Olga Dudchenko (Erez Lieberman Aiden)

## Kasper Munch

## Gavin Sherlock

## Anne Ruxandra Carvunis

## Elaine Ostrander

## Bobbie Cansdale (Claire Wade)