---
title: '#GI2018 - Day Two'
author: David McGaughey
date: '2018-09-18'
slug: gi2018-day-two
categories:
  - GI2018
  - bioinformatics
tags:
  - conference
  - GI2018
  - bioinformatics
  - talks
header:
  caption: ''
  image: ''
output:
  blogdown::html_page:
    toc: true
---


<div id="TOC">
<ul>
<li><a href="#day-two">Day Two</a><ul>
<li><a href="#katie-pollard-keynote">Katie Pollard Keynote</a></li>
<li><a href="#sri-kosuri-kosuri">Sri Kosuri (Kosuri)</a></li>
<li><a href="#patrick-brennan-nationwide-childrens">Patrick Brennan (Nationwide Children’s)</a></li>
</ul></li>
<li><a href="#kaitlin-samocha-hurles">Kaitlin Samocha (Hurles)</a></li>
<li><a href="#tracy-ballinger-semple">Tracy Ballinger (Semple)</a></li>
<li><a href="#lucia-spangenberg-naya">Lucia Spangenberg (Naya)</a></li>
</ul>
</div>

<div id="day-two" class="section level1">
<h1>Day Two</h1>
<p>Very sparse and poorly written notes covering <a href="https://twitter.com/hashtag/GI2018?src=hash">#GI2018</a>.</p>
<p>Typos everywhere. Things may change dramatically over time as I scan back through notes.</p>
<p>I’ve tried to respect #notwitter. Will be updated periodically.</p>
<p>Speaker (Lab | Group)</p>
<div id="katie-pollard-keynote" class="section level2">
<h2>Katie Pollard Keynote</h2>
<p>A population genetic view of human chromatin organization</p>
<p>Noncoding human mutation and disease</p>
<ul>
<li>which are causal?</li>
<li>how do they alter gene expression?</li>
<li>what genes / pathways do they affect?</li>
</ul>
<p>core idea:</p>
<ul>
<li>motif disruption (i.e. mutation alter binding site)</li>
</ul>
<p>but, how about we think about genome as structure?</p>
<ul>
<li>if structure (3D folding) is functional, any variation that modifies that would be deleterious</li>
</ul>
<p>chromatin structure and pop genetics</p>
<p>discussion of TADs / HiC / BE (boundary elements)</p>
<p>compare rare variants in patients, variants in healthy humans, fixed diffs between primates</p>
<ul>
<li>deletions depleted at boundary elements (strong negative selection)</li>
<li>also holds up across non-primates</li>
<li>less depletion in disease cohort (autism) variants</li>
</ul>
<p>“suggestion broad role for enhancer hijacking in disease”</p>
<p>noncoding variation assessment should be TAD aware</p>
<p>try to predict how mutations change TAD/BE</p>
<ul>
<li>kind of working (not as “sharp” as the real data)</li>
</ul>
<p>chromatin and genetic structure (LD) both have similar struture</p>
<ul>
<li>are they correlated?</li>
<li>not the same sizes (LD blocks shorter)</li>
<li>not correlated at 5kb to 2Mb</li>
</ul>
<p>chromatin interactions (HiC) more enriched for (compared to closest gene or LD):</p>
<ul>
<li>eQTL</li>
<li>GO enrichment terms</li>
<li>“most distal noncoding variants do not target closest expressed gene”</li>
</ul>
<p>sequence motifs fail to explain key aspects of protein-DNA binding</p>
<ul>
<li>30% of top 2000 ChIP-seq peaks have no sequence motif</li>
<li>many DNA binding proteins have very similar motifs but have disparate binding</li>
<li>seq motifs don’t capture DNA shape well (major / minor groove)</li>
</ul>
<p>hypothesis: “DNA binding proteins recognize DNA shape beyond seq. motifs”</p>
<ul>
<li>ShapeMF (<a href="https://www.bioconductor.org/packages/release/bioc/html/DNAshapeR.html">DNAshape R package</a>)</li>
<li>compare to classic motif with ENCODE ChIP-seq and HT-SELEX</li>
</ul>
<p>shape motifs match up between ENCODE and HT-SELEX</p>
<p>shape motifs can occur without sequence motifs</p>
<ul>
<li>most have both shape and motif patterns</li>
<li>some shape only and some motif only</li>
</ul>
<p>Machine learning in genomics comments (Pollard kindly taking time out her talk to discuss geneeral ML issues)</p>
<ul>
<li>try to model enhancer interaction with konwn enhancers, expressed genes, Hi-C data (TargetFinder)</li>
<li>common issues:
<ul>
<li>balanced training data (genomic data <a href="http://davemcg.github.io/post/are-you-in-genomics-stop-using-roc-use-pr/">usually isn’t</a>)</li>
<li>select features from whole dataset (feature selection should be <em>inside</em> cross validation)</li>
<li>use biologically appropriate examples</li>
<li>use the right statistic (PR, ROC, etc?)</li>
<li>most ML models assume observations are <em>independent</em> and <em>identically distributed</em></li>
<li>genomic data is inherently <strong>not</strong> independent and identically distributed</li>
<li>not trivial….different genome regions are inherently different (model doesn’t transfer)</li>
</ul></li>
</ul>
</div>
<div id="sri-kosuri-kosuri" class="section level2">
<h2>Sri Kosuri (Kosuri)</h2>
<p>Functional Impacts Of Rare Genetic Variation On Exome Recognition</p>
<p>Experimental assays (but huge)</p>
<p>multiplexed reporter assays (100k+ experiments) + DNA synthesis to do reverse genomics</p>
<p><a href="https://www.biorxiv.org/content/early/2018/03/10/199927" class="uri">https://www.biorxiv.org/content/early/2018/03/10/199927</a></p>
<p>Average genome has 4-5 million common variants and 50-100K rare variants (&lt;0.5% AF)</p>
<p>how do rare variants exert large effect changes?</p>
<ul>
<li>you see them in GTEx (under and over expression outliers)</li>
</ul>
<p>Massively parallel exon splicing reporter</p>
<ul>
<li>which mutations cause exon splicing issues</li>
<li>assayed 28k across 2,200 exon</li>
<li>mostly at splice junctions</li>
<li>but you still see plenty in exons and further away</li>
<li><em>in silico</em> predictions are pretty bad (splicing specific models OK)</li>
<li>3.8% (1,505 / 27,900) led to large loss of exon recognition</li>
<li>can’t predict them…. (with existing tools)</li>
<li>nice to see a whole slide on why this isn’t the most perfect experiment ever
<ul>
<li>short (100nt) exons used (biased)</li>
</ul></li>
</ul>
</div>
<div id="patrick-brennan-nationwide-childrens" class="section level2">
<h2>Patrick Brennan (Nationwide Children’s)</h2>
<p>Integration of whole genome, whole exome, and transcriptome pipeline for comprehensive genomic profiling of 60 pediatric cancer subjects</p>
<p>250x WES, 60X WGS (tumor + normal) RNA-seq (tumor only)</p>
<p>7 diff fusion detection algorithms</p>
<ul>
<li>ugh, 7 way venn</li>
<li>they focus on the few which are consistently called (or rather ranked on overlap and evidence support)</li>
</ul>
<p>RNA-seq</p>
<ul>
<li>DESeq, Salmon, then Ingenuity for pathway analysis (TPM as input)</li>
</ul>
<p>Differential comparison (RNA)</p>
<ul>
<li>but to what???</li>
<li>oh the ‘treehouse’ data (not sure what this is)</li>
</ul>
<p>Case study matching indel in gene with RNA expression difference, which helped to suggest tyrosine kinase inhibitor treatment</p>
<p>Case study IDing fusion gene between MET and RBPMS (chr 7 &lt;-&gt; chr 8)</p>
<p>Keep using ‘Churchill’ term for their software….annoying they don’t credit the tools that actually power it (GATK?)</p>
</div>
</div>
<div id="kaitlin-samocha-hurles" class="section level1">
<h1>Kaitlin Samocha (Hurles)</h1>
<p>Evaluating the role of rare variation in children with developmental disorders</p>
<p>DDD</p>
<ul>
<li>represent UK well</li>
<li>diagnostic rate of 35%</li>
</ul>
<p>4,300 trios have 94 genes with high gene burden</p>
<ul>
<li>gene test with poisson tests outputting p values for LoF, missense, synonymous</li>
</ul>
<p>New approach</p>
<ul>
<li>weight different types of mutations separately</li>
<li>gene score composed of weighted scores</li>
<li>multiple hypothesis correction (use prior knowledge)</li>
</ul>
<p>where do weights come from?</p>
<ul>
<li>observations from DDD cohort, comparing DDD enrichment by class (LoF, etc) to CADD</li>
</ul>
<p>new method</p>
<ul>
<li>finds more genes relative to poisson (old) method</li>
<li>applying to 10k trios</li>
</ul>
<p>inherited variation</p>
<ul>
<li>do cases have more rare damaging variants</li>
<li>cadd &gt; 21</li>
<li>observed once or more ExAC</li>
<li>effect size of ~1.7 for probands with affect parents</li>
<li>~1.2 for other probands</li>
</ul>
<p>causative enriched in high pLI genes</p>
<p>is this from cryptic recessive diagnoses?</p>
<ul>
<li>probably not</li>
<li>high pLI genes not enriched for recessive disease genes</li>
</ul>
<p>digenic inheritance?</p>
<ul>
<li>parents each have damaging mutations (but dif from each other)</li>
<li>probably not (no strong enrichment in undiagnosed vs diagnosed)</li>
</ul>
</div>
<div id="tracy-ballinger-semple" class="section level1">
<h1>Tracy Ballinger (Semple)</h1>
<p>Modeling Double Strand Breaks to Interrogate Structural Variation in Cancer</p>
<p>SV (structural variation) prevalent across cancer</p>
<p>Nonrandom distribution</p>
<ul>
<li>chromatin structure, sequence composition, and nuclear organization influence SV</li>
</ul>
<p>built Random Forest model that correlates well with real data</p>
<ul>
<li>Replication seq, open chromatin (DNase), pol2db, most important features</li>
<li>predicts well across different cell types</li>
</ul>
<p>Compare SV prediction to cancer breakpoints</p>
<ul>
<li>ID regions with more or less SV breakpoints than expected</li>
<li>also “passenger” regions which are not enriched</li>
</ul>
<p>cancer ‘coldspots’:</p>
<ul>
<li>enriched for active chromatin and deplted in fragile sites</li>
</ul>
</div>
<div id="lucia-spangenberg-naya" class="section level1">
<h1>Lucia Spangenberg (Naya)</h1>
<p>Retrieving Charras genomic tracts from the Uruguayan admixed population</p>
<p>Charras all killed by state 187 years ago</p>
<p>80 Uruguayan genomes (Just 10 for this project?)</p>
<p>Q1: Still have Charrua in Uruguayan genomes?</p>
<p>Calculated local ancestry (rel to EUR or AFR)</p>
<p>PCA to see clustering of Native Uruguayan. Also had 20 genomes from Simons</p>
<p>Want to calculate Fst Tree with other ancient/original ameican populations</p>
<ul>
<li>clusters with Guarani and Chane (expected!)</li>
</ul>
<p>Build protogenome by merging the 10 WGS they did</p>
<ul>
<li>99% covered</li>
</ul>
</div>
