---
title: '#GI2018 - Day Three'
author: David McGaughey
date: '2018-09-19'
slug: gi2018-day-three
categories:
  - GI2018
  - bioinformatics
tags:
  - conference
  - GI2018
  - bioinformatics
  - talks
header:
  caption: ''
  image: ''
output:
  blogdown::html_page:
    toc: true
---


<div id="TOC">
<ul>
<li><a href="#day-3">Day 3</a></li>
<li><a href="#rafael-irizarry-keynote">Rafael Irizarry Keynote</a></li>
<li><a href="#session-4-transcriptomics-alternative-splicing-and-gene-predictions">Session 4: Transcriptomics, Alternative Splicing and Gene Predictions</a><ul>
<li><a href="#mark-robinson-robinson">Mark Robinson (Robinson)</a></li>
<li><a href="#hagen-tilgner-tilgner">Hagen Tilgner (Tilgner)</a></li>
<li><a href="#nikka-keivanfar-church-10x">Nikka Keivanfar (Church, 10X)</a></li>
<li><a href="#koen-van-den-berge-clement">Koen Van den Berge (Clement)</a></li>
</ul></li>
</ul>
</div>

<div id="day-3" class="section level1">
<h1>Day 3</h1>
<p>Very sparse and poorly written notes covering <a href="https://twitter.com/hashtag/GI2018?src=hash">#GI2018</a>.</p>
<p>Typos everywhere. Things may change dramatically over time as I scan back through notes.</p>
<p>I’ve tried to respect #notwitter. Will be updated periodically.</p>
<p>Speaker (Lab | Group)</p>
</div>
<div id="rafael-irizarry-keynote" class="section level1">
<h1>Rafael Irizarry Keynote</h1>
<p>Understanding variabliity in high throughput data</p>
<p>Typical workflow raw data -&gt; pre-processing -&gt; analysis -&gt; discovery (false?????)</p>
<p>“When you find an <strong>unexpected result</strong>, be skeptical, check for systematic errors”</p>
<p>“Always, look at the data”</p>
<p>“Dynamite plots needs to die” (barplot + SE/SD)</p>
<ul>
<li>Plug for a blog post by me: <a href="http://davemcg.github.io/post/let-s-plot-4-r-vs-excel/" class="uri">http://davemcg.github.io/post/let-s-plot-4-r-vs-excel/</a></li>
<li>Show the data (boxplot + individual points)</li>
</ul>
<p>Showing batch effect example where huge diff in gene expression between groups is actually derived from technical issues (two groups were effectively done at different times)</p>
<p>Lin et al. 2014 (notorious paper comparing tx between human and mouse the claimed dif tissues within species more common to each other)</p>
<p>Yoav Gilad quickly pointed out that the mouse and human were done on different sequencers</p>
<p>So they re-did on same sequencer machine….got similar result</p>
<p>Not a new “result” (yanai 2004 had similar conclusion with microarray)</p>
<p>Rafael pointing out that high genes are high and low are low…always, so correlation will always be high.</p>
<p>Once you fix probe bias (microarray), then tissues cluster together.</p>
<p>But what about Lin RNA-seq?</p>
<p>A lot is explained by:</p>
<ul>
<li>number of transcripts</li>
<li>GC content</li>
</ul>
<p>Single Cell RNA-seq</p>
<ul>
<li>trying to disentangle batch effect from tumor type</li>
<li>actually proportion of zeros is driving variability
<ul>
<li>this is probably not biology</li>
</ul></li>
</ul>
<p>Do DNAm(ethylation) changes drive gene expression changes?</p>
<p>Discussing Ford et al. 2017 <em>bioRxiv</em></p>
<p>“Distinguishing consistent differences from random ones is challenging”</p>
<p>Talking about how technical variability and biological variability are different and must be considered when designing experiment (moar replicates!)</p>
<p>Do we trust individual measurements?</p>
<ul>
<li>when you do lots of tests, individual tests can be wrong</li>
<li>region test was done for CpG methylation (DMR test Jaffe 2012)</li>
<li>can also shuffle data (shuffle samples, not points) and see how often a diff can occur by chance</li>
<li>region + bootstrapping powerful approach (plus looking at replicates individually!!!)</li>
</ul>
</div>
<div id="session-4-transcriptomics-alternative-splicing-and-gene-predictions" class="section level1">
<h1>Session 4: Transcriptomics, Alternative Splicing and Gene Predictions</h1>
<div id="mark-robinson-robinson" class="section level2">
<h2>Mark Robinson (Robinson)</h2>
<p>On the analysis of long-read sequencing data for gene expression</p>
<p>Illumina &lt;-&gt; ONT (Oxford Nanopore) cheap &lt;-&gt; expensive short fragments &lt;-&gt; full length lower error rates &lt;-&gt; higher error rates</p>
<p>DataSet</p>
<ul>
<li>WT vs Srpk1-KO</li>
<li>ONT (various preps) and Illumina</li>
<li>1M reads / sample (ONT); 30M reads / sample (Illumina)</li>
</ul>
<p>25% mismatch with ONT vs &lt;1% for Illumina (but they still map)</p>
<p>ONT doesn’t look useful for variation in sequence (way too noisy)</p>
<p>Do we actually get full length transcripts?</p>
<ul>
<li>for longer ones…not really</li>
<li>showing how a longer transcript has worse coverage with ONT</li>
</ul>
<p>Quantification ONT &lt;-&gt; Illumina</p>
<ul>
<li>looks pretty good for protein coding genes</li>
<li>more discrepancy at the tx level (still OK)</li>
</ul>
<p>ONT</p>
<ul>
<li>minimap2 + salmon current choice for tooling</li>
</ul>
<p>ONT direct RNA or ONT cDNA???</p>
<ul>
<li>look pretty comparable</li>
<li>direct RNA requires crazy input amounts, probably not worth it unless you are looking at RNA base modifications</li>
<li>ONT has new stranded cDNA protocol / kit</li>
</ul>
<p>Diff expression analysis</p>
<ul>
<li>once you account for way lower depth (again 1M vs 30M) for ONT, roughly comparable</li>
</ul>
<p>Definitely not time to jump over for more standard DE type RNAseq tests</p>
</div>
<div id="hagen-tilgner-tilgner" class="section level2">
<h2>Hagen Tilgner (Tilgner)</h2>
<p>Single-cell isoform RNA sequencing (ScISOr-Seq) across thousands of cells reveals isoforms of cerebellar cell types.</p>
<p>Need long-read RNA seq to confidently ID isoforms</p>
<p>ScISOr-Seq</p>
<ul>
<li>10X uses 3’ sequencing</li>
<li>Add PacBio ISO-Seq and find the 10X barcode</li>
<li>blend 10X and PacBio data</li>
</ul>
<p>Barcode ID (long read data can be noisy…)</p>
<ul>
<li>99.99% specificity with PacBio</li>
<li>Only 58% of molecules have barcode (74% in Illumina)
<ul>
<li>not catastrophic, more annoying</li>
</ul></li>
</ul>
<p>It works</p>
<ul>
<li>validate some junctions with MS/MS</li>
</ul>
<p>Plug for <a href="isoformAtlas.com" class="uri">isoformAtlas.com</a></p>
<p>See some compelling cell type specific isoform usage</p>
</div>
<div id="nikka-keivanfar-church-10x" class="section level2">
<h2>Nikka Keivanfar (Church, 10X)</h2>
<p>Bootstrapping Biology: Quick and easy de novo genome assembly to enable single cell gene expression analysis</p>
<p>Linked-Reads + Supernova + Cactus/CAT</p>
<p>Supernova does assembly (with phase)</p>
<p>Supernova v2 can get N50 over 100+ kb for many species (HUman, Hummingbird, Drosophila)</p>
<ul>
<li>Zebrafish is tougher (only around 20kb)</li>
</ul>
<p>CAT</p>
<ul>
<li>Comparative Annotation Toolkit</li>
<li>“provides a flexible way to simultaneously annotate entire clades and identify orthology relationships”</li>
<li>doi.org/10.1101/231118</li>
<li><a href="https://genome.cshlp.org/content/28/7/1029" class="uri">https://genome.cshlp.org/content/28/7/1029</a></li>
</ul>
<p>“Blackjack” (Donkey) blood sequenced with 10X</p>
<ul>
<li>made assembly with Supernova</li>
<li>better metrics than other donkey genomes</li>
</ul>
<p>Used assembly (+CAT to get gene info) for scRNA-seq of the blood</p>
<ul>
<li>labels major blood types (T cells, B cells, NK, monocytes, plasma, etc)</li>
</ul>
</div>
<div id="koen-van-den-berge-clement" class="section level2">
<h2>Koen Van den Berge (Clement)</h2>
<p>Discrete and continuous differential expression analysis for single-cell RNA-seq data</p>
<p>scRNA-seq data is WAY noiser than bulk RNA-seq (at least cell to cell)</p>
<p>Bulk RNAseq generally use negative binomial models (edgeR, DESeq2)</p>
<p>Sort of surprisingly, these bulk RNAseq methods seem to work well in scRNA</p>
<p>But zeros…..showing simulations where adding zeros getting weird dispersion patterns</p>
<ul>
<li><strong>Why not ZINB-WAaE??</strong></li>
</ul>
<p>In real also see some odd dispersion</p>
<p>Propose zero inflated negative binomal (ZINB) distribution</p>
<ul>
<li><strong>Oh, this is the ZINB-WaVE author</strong></li>
</ul>
<p>Downweighting excess zeros helps reduce variance and improve power</p>
<p><a href="https://genomebiology.biomedcentral.com/articles/10.1186/s13059-018-1406-4" class="uri">https://genomebiology.biomedcentral.com/articles/10.1186/s13059-018-1406-4</a></p>
<p>Continuous DE</p>
<ul>
<li>following SC across developmental trajectory</li>
<li>slingshot tool (Street et al. 2018)</li>
</ul>
<p>What do you do downstream of the trajectory analyses???</p>
<ul>
<li>most do DE between clusters</li>
<li>which he argues is sub-optimal
<ul>
<li>no fixed biological meaning</li>
<li>hetergeneous</li>
<li>inflate FDR with so many comparisons required</li>
</ul></li>
</ul>
<p>Propose NB-GAM</p>
<ul>
<li>so smooth out pseudotime trajectory with offset of sequencing depth (or other covariates?)</li>
<li>cubic spline smoother right now</li>
<li><strong>this is pretty slick</strong></li>
<li>“very preliminary”</li>
</ul>
</div>
</div>
