---
title: '#BoG18: Talk Notes'
author: David McGaughey
date: '2018-05-08'
slug: bog18-notes
categories:
  - bog18
  - bioinformatics
tags:
  - conference
  - bog18
  - bioinformatics
  - talks
output:
  blogdown::html_page:
    toc: true
---


<div id="TOC">
<ul>
<li><a href="#intro">Intro</a></li>
<li><a href="#genome-engineering-and-genome-editing">Genome Engineering and Genome Editing</a><ul>
<li><a href="#jef-boeke">Jef Boeke</a><ul>
<li><a href="#writing-genomes">Writing Genomes</a></li>
<li><a href="#dark-matter">“dark matter”</a></li>
<li><a href="#big-dna">big dna</a></li>
</ul></li>
<li><a href="#greg-findlay-jay-shendure">Greg Findlay (Jay Shendure)</a></li>
<li><a href="#stephen-levene">Stephen Levene</a></li>
<li><a href="#david-truong">David Truong</a></li>
<li><a href="#feng-zhang">Feng Zhang</a></li>
<li><a href="#molly-gasperini-jay-shendure">Molly Gasperini (Jay Shendure)</a></li>
<li><a href="#eilon-sharon">Eilon Sharon</a></li>
<li><a href="#luca-pinello">Luca Pinello</a></li>
</ul></li>
<li><a href="#population-genomics">Population Genomics</a><ul>
<li><a href="#mattias-joakobsson">Mattias Joakobsson</a></li>
</ul></li>
</ul>
</div>

<div id="intro" class="section level1">
<h1>Intro</h1>
<p>Very sparse and poorly written notes covering <a href="https://twitter.com/search?q=%23BoG18&amp;src=tyah">#BoG18</a>. I’ve tried to respect #notwitter. Will be updated periodically.</p>
</div>
<div id="genome-engineering-and-genome-editing" class="section level1">
<h1>Genome Engineering and Genome Editing</h1>
<div id="jef-boeke" class="section level2">
<h2>Jef Boeke</h2>
<div id="writing-genomes" class="section level3">
<h3>Writing Genomes</h3>
<p>Building synthetic yeast genomes. Contig/chr one by one. All designed. <em>Sc2.0</em></p>
<p>80+% complete for each of the 16.</p>
</div>
<div id="dark-matter" class="section level3">
<h3>“dark matter”</h3>
<p>Can we use ‘big dna’ to functionally query mammalian genomes? ‘Synthetic haplotypes’</p>
<p>Building dif combinations of haplotype blocks</p>
<p>synthetic hypervariation:</p>
<pre><code>- query enhancers
- alt splicing</code></pre>
<p>built 102kb locus (human) and put into yeast</p>
<pre><code>- built in 3kb chunks and can assemble dif combiations</code></pre>
</div>
<div id="big-dna" class="section level3">
<h3>big dna</h3>
<p>Can build big dna pieces CEGS grant will build 3 100kb+ loci / year want community input</p>
</div>
</div>
<div id="greg-findlay-jay-shendure" class="section level2">
<h2>Greg Findlay (Jay Shendure)</h2>
<p>Accurate classficiation of thousands of BRCA1 variants with saturation genome editing</p>
<p>vous a problem BRCA1:</p>
<ul>
<li>4243 clinvar snvs</li>
<li><blockquote>
<p>50% VOUS</p>
</blockquote></li>
</ul>
<p>How to functionally validate?</p>
<p>Use Homology-direct repair (HDR). Can engineer precise edits.</p>
<p>Use a library of SNVs for the HDR</p>
<p>Over time selecdtion removes non functional edits</p>
<p>Each experimente:</p>
<ul>
<li>millions of cells</li>
<li>millions of sequencing reads to count SNVs</li>
</ul>
<p>Variable effects at splicing junctions</p>
<ul>
<li>sometimes just 2-3 bp</li>
<li>sometimes 9 base pairs</li>
</ul>
<p>also have matched rna-seq data</p>
<p>aberrant splicing causes RNA depletion</p>
<p>matches up really well with clinvar designations</p>
<p>question:</p>
<ul>
<li>hdr effeciency rate? 10-90% effectiveness</li>
</ul>
</div>
<div id="stephen-levene" class="section level2">
<h2>Stephen Levene</h2>
<p>eccDNA is a possible mediator of chromosomal polymorphism at multple loci</p>
<p>‘physical chemist by training’</p>
<ol start="5" style="list-style-type: lower-alpha">
<li>coli genome: 1 femtoliter volume</li>
</ol>
<p>100 fold compaction problem (DNA)</p>
<p>~10k fold for mammalian</p>
<p>Techniques for DNA/chromiatin flexibility:</p>
<ul>
<li>Hi-C</li>
<li>FISH</li>
<li>SLICE (Beagrie et al Nature 2017)</li>
</ul>
<p>eccDNA (circular dna outside chromosome/nucleus(?))</p>
<ul>
<li>unclear how it forms</li>
<li>elevated levels associatd with genome instability</li>
</ul>
<p>how to capture?</p>
<ul>
<li>DNA -&gt; SDS lysis -&gt; isolate gDNA -&gt; cscl gradient -&gt; bottom bit</li>
<li>or exoV treatment (leaves circular alone)</li>
</ul>
<p>take pictures of the loops with high resolution microscopy</p>
<p>sequenced a few (<strong>unfortunately with short read illumina</strong>)</p>
<p>modeled with molecular dynamics</p>
</div>
<div id="david-truong" class="section level2">
<h2>David Truong</h2>
<p>Resurrection of Histone H3k27 Me in brewer’s yeast by human prc2 and plant atxr6</p>
<p>human pathway reconstruction in yeast</p>
<ul>
<li>avoid pleiotropy (hopefully)</li>
<li>more real than <em>in vitro</em></li>
</ul>
<p>Yeast (s. cerevisiae) lost histone mods</p>
<p><strong>Adding them back???</strong></p>
<p>yes</p>
<p>humanize yeast histones</p>
<p>add synthetic human histones</p>
<p>force out wt histones with +5FOA</p>
<p><em>20 days later</em> ….. one colony</p>
<p>Keep growing the colony out</p>
<p>WGS: mutations in cell cycle regulation</p>
<ul>
<li>bypassing histone cycle checks?</li>
</ul>
<p>brewer’s yeast does not have H3K27 methylation</p>
<ul>
<li>PRC2 complex (to methylate h3k27)</li>
</ul>
<p>add the stuffs - what happens?</p>
<ul>
<li>made artifical chr with PRC2 complex
<ul>
<li>and a slightly broken one</li>
</ul></li>
<li>not much happens in WT yeast
<ul>
<li>no me changes</li>
</ul></li>
<li>deleted H3k36me3 (might antagonize artificial chr)
<ul>
<li>nope</li>
</ul></li>
<li>can you jump start with atxr6 (does mono me)?
<ul>
<li>yes (confirmed with mass spec)</li>
<li>not super high levels (0.054% tri me)</li>
</ul></li>
</ul>
</div>
<div id="feng-zhang" class="section level2">
<h2>Feng Zhang</h2>
<p>Advances in genome editing technologies</p>
<p>two major classes of CRISPR</p>
<ul>
<li>class 1 (multi subunut)</li>
<li>class 2 (single subunit crRNA-effector)</li>
</ul>
<p>trying to find new class 2</p>
<ul>
<li>bioinformatic screen with BLAST of cas1</li>
<li>found a bunch (Shmakov collaboration)</li>
</ul>
<p>cas13</p>
<ul>
<li>added into e. coli</li>
<li>modify to only edit RNA?</li>
</ul>
<p>rna editing</p>
<ul>
<li>reversible</li>
<li>nuclease based editing inefficient in post mitotic cells</li>
<li>dCas13 linked with <a href="https://en.wikipedia.org/wiki/ADAR">ADAR</a> (adenosine to inosine) + guideRNA –&gt; A to I conversion in RNA</li>
<li>90+% conversion</li>
<li>1732 off target incidents
<ul>
<li>925 off target with non-targeting guide!</li>
<li>so protein itself is not ideal….</li>
<li>identified non-binding residues of ADAR</li>
<li>mutated them</li>
</ul></li>
<li>v2 works better
<ul>
<li>18385 off target (v1) to 20 (v2)</li>
</ul></li>
<li>still developing</li>
</ul>
</div>
<div id="molly-gasperini-jay-shendure" class="section level2">
<h2>Molly Gasperini (Jay Shendure)</h2>
<p><strong>my fav of the night</strong></p>
<p>crisprQTL mapping as a genome-wide association framework for cellular genetic screens</p>
<p>lots of guideRNA to made mutations, check for dif in expression</p>
<p>nuclease inactive cas9</p>
<p>want to test all enhancers against all genes</p>
<p>scRNA-seq + guideRNA (multiplex gRNA)</p>
<ul>
<li>thus multiple perturbations per ‘assay’</li>
<li>15-30 / cell!</li>
</ul>
<p>targeted 1,119 candidate enhancers</p>
<ul>
<li>15 guides / cell</li>
<li>47k cells</li>
<li>10X</li>
<li>CROP-seq</li>
<li>works really well</li>
<li>crisprQTL usually targets closest gene
<ul>
<li>sometimes not….</li>
</ul></li>
<li>matches up with histone chip-seq</li>
<li>34.3kb average distance from enhancer &lt;-&gt; gene</li>
</ul>
<p>** new data! ** 4,801 enhancers</p>
<ul>
<li>built logistic regression model on pilot to pick new candidates</li>
<li>30 guides / cell</li>
<li>correlates with pilot</li>
</ul>
<p>manolis kellis q:</p>
<ul>
<li>why doesn’t work so well? expected more</li>
<li>what about multiple SNPS / block?</li>
</ul>
</div>
<div id="eilon-sharon" class="section level2">
<h2>Eilon Sharon</h2>
<p>Testing genetic var effect on fitness using precise genome editing</p>
<p>high throughput edigint</p>
<p>crispey: cas9 retron precise parallel editing via homology</p>
<p>use bacterial reverse transcriptase and RNA retron to covalently link ssDNA donor to guide-tracrRNA</p>
<p>can insert long sequences</p>
<p>(yeast)</p>
<p>measure fitness of genetic variants (growth competition)</p>
<p>sequence every 2-3 gens</p>
<p>model at linear relative strain abundance / time (generation #)</p>
<p># missense var ~ # synonymous var for effecting fitness!</p>
</div>
<div id="luca-pinello" class="section level2">
<h2>Luca Pinello</h2>
<p>CRISPR-SURF exploratory and interactive software for analyzing CRISPR-base tiling screens</p>
<p>Uncover non-coding functional regions</p>
<p>** Nice overview of CRISPR tiling strategy ** Mutate (tile across region) -&gt; Measure pheno change (somehow) -&gt; Assess (sequence gRNA)</p>
<p>No unified framework to analyze these kind of assays</p>
<p>many challenges</p>
<ul>
<li>biological noise</li>
<li>sgRNA efficiencies</li>
<li>non-uniform spacing</li>
<li>perturbation / assay differences</li>
<li>epigenetic perturbation can be wide (changing 200bp or so)</li>
</ul>
<p>deconvolve with generalized lasso</p>
<p>fastq -&gt; score -&gt; segmentation -&gt; deconvolution -&gt; region ID</p>
</div>
</div>
<div id="population-genomics" class="section level1">
<h1>Population Genomics</h1>
<div id="mattias-joakobsson" class="section level2">
<h2>Mattias Joakobsson</h2>
<p>** out of my field here **</p>
<p>Sequence based approaches utilizing complete modern and ancient genomes to investigate early human history</p>
<p>Use full genomes on ancient pops</p>
<p>Population divergence models</p>
<pre><code>                  X
 +                X
 |                X
 |                X
 |                X
 |               XXXX
 |              XX  XXX</code></pre>
<p>time | XXX XX | XXX XX | XX XX | | A B C | v</p>
<p>Can model whether discordant or concordant (a,b,c) over time</p>
<p>estimate pop divergence (time) in generations</p>
<p>use genes from different populations to estimate divergence - ‘tt method’</p>
<p>stone age humans from sourthern afridca - 13 genomes from 3 people - a bit ‘right’ of yoruba - admixture with east africa <strong>missed something</strong></p>
</div>
</div>
