---
title: "#GI2019 2019 11 07"
author: David McGaughey
date: '2019-11-07'
slug: gi2019-day02
categories:
  - bioinformatics
  - gi2019
  - conferences
  - talk
tags:
  - bioinformatics
  - gi2019
  - conferences
  - talk
header:
  caption: ''
  image: ''
output:
  blogdown::html_page:
    toc: true
---


<div id="TOC">
<ul>
<li><a href="#day-2---session-2---sequencing-algorithms-variant-discovery-and-genome-assembly">Day 2 - Session 2 - SEQUENCING ALGORITHMS, VARIANT DISCOVERY AND GENOME ASSEMBLY</a></li>
<li><a href="#genomic-sketching-with-hyperloglog">Genomic sketching with HyperLogLog</a></li>
<li><a href="#centroflyeassembling-centromeres-with-long-error-prone-reads">centroFlye—Assembling centromeres with long error-prone reads</a></li>
<li><a href="#genotyping-structural-variants-in-pangenome-graphs-using-the-vg-toolkit">Genotyping structural variants in pangenome graphs using the vg toolkit</a></li>
<li><a href="#rapidly-mapping-raw-nanopore-signal-with-uncalled-to-enable-real-time-targeted-sequencing">Rapidly mapping raw nanopore signal with UNCALLED to enable real-time targeted sequencing</a></li>
<li><a href="#the-construct-and-utility-of-reference-pan-genome-graphs">The construct and utility of reference pan-genome graphs</a></li>
<li><a href="#princessa-framework-for-comprehensive-detection-andphasing-of-snps-and-structural-variants">PRINCESS—A framework for comprehensive detection andphasing of SNPs and structural variants</a></li>
<li><a href="#efficient-chromosome-scale-haplotype-resolved-assembly-of-human-individuals">Efficient chromosome-scale haplotype-resolved assembly of human individuals</a></li>
<li><a href="#utilization-of-an-ensemble-approach-for-identification-of-driver-fusions-in-pediatric-cancer">Utilization of an ensemble approach for identification of driver fusions in pediatric cancer</a></li>
</ul>
</div>

<div id="day-2---session-2---sequencing-algorithms-variant-discovery-and-genome-assembly" class="section level1">
<h1>Day 2 - Session 2 - SEQUENCING ALGORITHMS, VARIANT DISCOVERY AND GENOME ASSEMBLY</h1>
<p>Genome Informatics 2019 at CSHL</p>
<p><strong>Bold</strong> is the speaker</p>
</div>
<div id="genomic-sketching-with-hyperloglog" class="section level1">
<h1>Genomic sketching with HyperLogLog</h1>
<p>Daniel N. Baker, <strong>Ben Langmead.</strong></p>
<p>Slides at…crap missed it. Email Ben I guess?</p>
<p>Sketching</p>
<ul>
<li>algorithm to collapse genomes into summary info</li>
<li>fasta -&gt; kmers (“shingle”) -&gt; sample (smartly)</li>
<li>showing set relationships</li>
<li>the “minimum” of a set of sketches gives you useful info on the range you would expect</li>
</ul>
<p>Unions and Intersections</p>
<ul>
<li>space of possible coincidences requires multiple samples to get useful info</li>
<li>bottom 3?</li>
<li>or minimum in 3 partitions?</li>
</ul>
<p>MinHash uses bottom-k approach (Ondov / Phillipy)</p>
<p>Tremendous talk - I feel like I understand what is happening. Invite Ben to talk at your place…I’d imagine you’d get a great talk.</p>
<p>OK, so take <em>log log minimum</em></p>
<p>Daniel Baker wrote:</p>
<ul>
<li>bit.ly/dash_pre</li>
<li>github.com/dnbaker/dashing</li>
<li>github.com/dnbaker/sketch</li>
</ul>
<p>HyperLogLog *HLL)</p>
<ul>
<li>k partition</li>
<li>log<sub>2</sub>n</li>
<li>exponent</li>
<li>average, bias correction</li>
</ul>
<p>Highly vectorized (for speed! Massively parallel calculation)</p>
<ul>
<li>only on intel?</li>
</ul>
<p>Comparing to MinHash</p>
<ul>
<li>HLL deals with lopsided sets better (bottom k not ideal, better minimum k partition)</li>
<li>Uses WAY less memory and wall clock</li>
</ul>
<p>Future stuffs</p>
<ul>
<li>multi-k (kmer lengths)</li>
<li>weighted jaccard</li>
</ul>
</div>
<div id="centroflyeassembling-centromeres-with-long-error-prone-reads" class="section level1">
<h1>centroFlye—Assembling centromeres with long error-prone reads</h1>
<p><strong>Andrey Bzikadze</strong>, Pavel A. Pevzner.</p>
</div>
<div id="genotyping-structural-variants-in-pangenome-graphs-using-the-vg-toolkit" class="section level1">
<h1>Genotyping structural variants in pangenome graphs using the vg toolkit</h1>
<p><strong>Jean Monlong</strong>, Glenn Hickey, David Heller, Jonas Andreas Sibbesen, Jouni Siren, Jordan Eizenga, Eric T. Dawson, Erik Garrison, Adam Novak, Benedict Paten.</p>
</div>
<div id="rapidly-mapping-raw-nanopore-signal-with-uncalled-to-enable-real-time-targeted-sequencing" class="section level1">
<h1>Rapidly mapping raw nanopore signal with UNCALLED to enable real-time targeted sequencing</h1>
<p><strong>Sam Kovaka</strong>, Yunfan Fan, Winston Timp, Michael C. Schatz.</p>
</div>
<div id="the-construct-and-utility-of-reference-pan-genome-graphs" class="section level1">
<h1>The construct and utility of reference pan-genome graphs</h1>
<p><strong>Heng Li.</strong></p>
</div>
<div id="princessa-framework-for-comprehensive-detection-andphasing-of-snps-and-structural-variants" class="section level1">
<h1>PRINCESS—A framework for comprehensive detection andphasing of SNPs and structural variants</h1>
<p><strong>Medhat Mahmoud</strong>, Winston Timp, Fritz J. Sedlazeck.</p>
</div>
<div id="efficient-chromosome-scale-haplotype-resolved-assembly-of-human-individuals" class="section level1">
<h1>Efficient chromosome-scale haplotype-resolved assembly of human individuals</h1>
<p><strong>Shilpa Garg</strong>, Arkarachai Fungtammasan, Anthony Schmitt, Andrew
Carroll, Paul Pelusol, Emily Hatas, Fritz Sedlazeck, Justin Zook, Mike
Chou, John Aach, Jason Chin, Heng Li, George Church.</p>
</div>
<div id="utilization-of-an-ensemble-approach-for-identification-of-driver-fusions-in-pediatric-cancer" class="section level1">
<h1>Utilization of an ensemble approach for identification of driver fusions in pediatric cancer</h1>
<p><strong>Stephanine LaHaye</strong>, Kyle Voytovich, James Fitch, Natalie Bir, Sean D. McGrath, Anthony Miller, Amy Wetzel, Vincent Magrini, Catherine E. Cottrell, Elaine R. Mardis, Richard K. Wilson.</p>
</div>
