---
title: "2019 11 09 GI2019 Personal and Medical Genomics"
author: David McGaughey
date: '2019-11-09'
slug: gi2019-day04-morning
categories:
  - bioinformatics
  - gi2019
  - conferences
  - talk
tags:
  - bioinformatics
  - gi2019
  - conferences
  - talk
header:
  caption: ''
  image: ''
output:
  blogdown::html_page:
    toc: true
---


<div id="TOC">
<ul>
<li><a href="#day-4---session-9---personal-and-medical-genomics">Day 4 - Session 9 - PERSONAL AND MEDICAL GENOMICS</a></li>
<li><a href="#characterization-of-prevalence-and-health-consequences-of-uniparental-disomy-in-four-million-individuals-from-the-general-population">Characterization of prevalence and health consequences of uniparental disomy in four million individuals from the general population</a></li>
<li><a href="#sub-continental-ancestry-inference-based-on-the-gnomad-dataset-accurately-classifies-patients-at-nch">Sub-continental ancestry inference based on the gnomAD dataset accurately classifies patients at NCH</a></li>
<li><a href="#inferring-clone--and-haplotype-specific-chromosomal-organization-in-rearranged-cancer-genomes-with-multiple-sequencing-technologies">Inferring clone- and haplotype-specific chromosomal organization in rearranged cancer genomes with multiple sequencing technologies</a></li>
<li><a href="#identification-and-interpretation-of-common-and-rare-variants-in-relation-to-rare-disease-phenotype-and-outcome">Identification and interpretation of common and rare variants in relation to rare disease phenotype and outcome</a></li>
<li><a href="#somatic-mutation-status-prediction-by-a-splicing-alteration-based-machine-learning-technique">Somatic mutation status prediction by a splicing-alteration-based machine learning technique</a></li>
<li><a href="#beyond-accessibilityatac-seq-footprinting-analysis-reveals-dynamics-of-transcription-factor-binding-during-preimplantation-development">Beyond accessibility—ATAC-seq footprinting analysis reveals dynamics of transcription factor binding during preimplantation development</a></li>
</ul>
</div>

<div id="day-4---session-9---personal-and-medical-genomics" class="section level1">
<h1>Day 4 - Session 9 - PERSONAL AND MEDICAL GENOMICS</h1>
</div>
<div id="characterization-of-prevalence-and-health-consequences-of-uniparental-disomy-in-four-million-individuals-from-the-general-population" class="section level1">
<h1>Characterization of prevalence and health consequences of uniparental disomy in four million individuals from the general population</h1>
<p>Priyanka Nakka, Samuel Pattillo Smith, Anne H. O’Donnell-Luria, Kimberly F. McManus, 23andMe Research Team, Joanna L. Mountain, <strong>Sohini Ramachandran</strong>, Fah Sathirapongsasuti.</p>
<p><span class="citation">@s_ramach</span></p>
<p>Paper came out two days ago AJHG</p>
<p>“Population-level summaries of ROH increase with divergnce from Agrica”
ROH == Runs of Homozygosity (LONG runs where both chr are identical)</p>
<p>“Class C ROH”</p>
<ul>
<li>don’t know what this is…something about consanguinity?</li>
<li>oh, &gt;0.8MB?</li>
</ul>
<p>Group interested in distribution across human genomes</p>
<p>23andMe partially sort of funded (personel) this study</p>
<p>Also uses 23andMe data</p>
<ul>
<li>obv</li>
<li>not even MacArthur has 4 million genomes</li>
</ul>
<p>Posting IBD karyogram….from reddit (imgur.com/a/hl6Cd)</p>
<p>UPD</p>
<ul>
<li>both homologs of chr come from one parent</li>
<li>meiotic oopsie</li>
</ul>
<p>3,300 reported in clinical literature</p>
<ul>
<li>upd-tl.com/upd.html</li>
<li>more common on chr 6,7,11, and 15 (most common)</li>
</ul>
<p>What is incidence in population (at least 23andMe pop)?</p>
<p>What are rates of maternal/paternal UPD?</p>
<p>Three subtypes of UPD</p>
<ul>
<li>HetUPD, IsoUPD, Partial IsoUPD</li>
</ul>
<p>In 916,712 duos/trios in 23andMe found 199 with UPD</p>
<ul>
<li>Maternal UPD 3X more common than Paternal UPD</li>
<li>different dist across chr than the upd-tl.com data</li>
</ul>
<p>Estimate 1/2000 rate (more common than previously estimated)</p>
<p>Trained log. regression classifier for each chr for each pop to diagnose ROH from UPD (instead of consanguinity)</p>
<ul>
<li>used ROH length</li>
<li>ratio of ROH across chr</li>
<li>other things I missed</li>
</ul>
<p>Found 304 <strong>putative</strong> UPD cases in 1.3 million “singletons”</p>
<ul>
<li>recapitulates chr dist. patterns</li>
<li>found 172 more in UK biobank</li>
<li>found 0 in the 4,000 duos (not a big sample size)</li>
</ul>
</div>
<div id="sub-continental-ancestry-inference-based-on-the-gnomad-dataset-accurately-classifies-patients-at-nch" class="section level1">
<h1>Sub-continental ancestry inference based on the gnomAD dataset accurately classifies patients at NCH</h1>
<p><strong>Andrei Rajkovic</strong>, Defne Ceyhan, Greg Wheeler, Tara Lichtenberg, Ben Kelly, Peter White.</p>
<p>#Patient stratification in the UK’s 100k Genomes Project—Using WGS and machine learning to predict cancer outcomes
<strong>Kate Ridout</strong>, Pauline Robbe, CLL pilot consortium, Anna Schuh.
Presenter affiliation: University of Oxford, Oxford, United Kingdom. 231</p>
</div>
<div id="inferring-clone--and-haplotype-specific-chromosomal-organization-in-rearranged-cancer-genomes-with-multiple-sequencing-technologies" class="section level1">
<h1>Inferring clone- and haplotype-specific chromosomal organization in rearranged cancer genomes with multiple sequencing technologies</h1>
<p><strong>Sergey Aganezov</strong>, Fritz Sedlazeck, Sara Goodwin, Gayatri Arun, Isac Lee, Sam Kovaka, Michael Kitsche, Rachel Scherman, Ilya Zban, Vitaly Aksenov, Nikita Alexeev, Robert Wappel, Melissa Kramer, Karen Kostroff, David L. Spector, Winston Timp, Michael S. Schatz, W. Richard McCombie.</p>
</div>
<div id="identification-and-interpretation-of-common-and-rare-variants-in-relation-to-rare-disease-phenotype-and-outcome" class="section level1">
<h1>Identification and interpretation of common and rare variants in relation to rare disease phenotype and outcome</h1>
<p>Mana Gajapathy, Brandon Wilk, Arthur Weborg, Angelina Uno- Antonison, Alex Moss, Matt Holt, Donna Brown, Melissa Wilk, Camille Birch, Nadiya Sosonkina, <strong>Elizabeth Worthey</strong></p>
</div>
<div id="somatic-mutation-status-prediction-by-a-splicing-alteration-based-machine-learning-technique" class="section level1">
<h1>Somatic mutation status prediction by a splicing-alteration-based machine learning technique</h1>
<p>Raul N. Mateos, Naoko Iida, Kenichi Chiba, Ai Okada, Yuichi Shiraishi.</p>
<p>NOT HERE</p>
<p>Model selection and permutation testing for association studies within a large direct-to-consumer genetic cohort
<strong>Elena P. Sorokin</strong>, Danny S. Park, Kristin A. Rand, Julie M. Granka, Eurie L. Hong, Kenneth G. Chahine, Catherine A. Ball.</p>
<p>NOT TWEETABLE</p>
</div>
<div id="beyond-accessibilityatac-seq-footprinting-analysis-reveals-dynamics-of-transcription-factor-binding-during-preimplantation-development" class="section level1">
<h1>Beyond accessibility—ATAC-seq footprinting analysis reveals dynamics of transcription factor binding during preimplantation development</h1>
<p>Mette Bentsen, Philipp Goymann, Anastasiia Petrova, Hendrik Schultheis, Kathrin Klee, Jens Preussner, Carsten Kuenne, Mario Looso.</p>
<p>NOT HERE</p>
</div>
