---
title: "#GI2019 2019 11 07"
author: David McGaughey
date: '2019-11-07'
slug: gi2019-day02
categories:
  - bioinformatics
  - gi2019
  - conferences
  - talk
tags:
  - bioinformatics
  - gi2019
  - conferences
  - talk
header:
  caption: ''
  image: ''
output:
  blogdown::html_page:
    toc: true
---


<div id="TOC">
<ul>
<li><a href="#day-2---session-2---sequencing-algorithms-variant-discovery-and-genome-assembly">Day 2 - Session 2 - SEQUENCING ALGORITHMS, VARIANT DISCOVERY AND GENOME ASSEMBLY</a></li>
<li><a href="#genomic-sketching-with-hyperloglog">Genomic sketching with HyperLogLog</a></li>
<li><a href="#centroflyeassembling-centromeres-with-long-error-prone-reads">centroFlye—Assembling centromeres with long error-prone reads</a></li>
<li><a href="#genotyping-structural-variants-in-pangenome-graphs-using-the-vg-toolkit">Genotyping structural variants in pangenome graphs using the vg toolkit</a></li>
<li><a href="#rapidly-mapping-raw-nanopore-signal-with-uncalled-to-enable-real-time-targeted-sequencing">Rapidly mapping raw nanopore signal with UNCALLED to enable real-time targeted sequencing</a></li>
<li><a href="#the-construct-and-utility-of-reference-pan-genome-graphs">The construct and utility of reference pan-genome graphs</a></li>
<li><a href="#princess-a-framework-for-comprehensive-detection-and-phasing-of-snps-and-structural-variants">PRINCESS — A framework for comprehensive detection and phasing of SNPs and structural variants</a></li>
<li><a href="#efficient-chromosome-scale-haplotype-resolved-assembly-of-human-individuals">Efficient chromosome-scale haplotype-resolved assembly of human individuals</a></li>
<li><a href="#utilization-of-an-ensemble-approach-for-identification-of-driver-fusions-in-pediatric-cancer">Utilization of an ensemble approach for identification of driver fusions in pediatric cancer</a></li>
</ul>
</div>

<div id="day-2---session-2---sequencing-algorithms-variant-discovery-and-genome-assembly" class="section level1">
<h1>Day 2 - Session 2 - SEQUENCING ALGORITHMS, VARIANT DISCOVERY AND GENOME ASSEMBLY</h1>
<p>Genome Informatics 2019 at CSHL</p>
<p><strong>Bold</strong> is the speaker</p>
</div>
<div id="genomic-sketching-with-hyperloglog" class="section level1">
<h1>Genomic sketching with HyperLogLog</h1>
<p>Daniel N. Baker, <strong>Ben Langmead.</strong></p>
<p>Slides at…crap missed it. Email Ben I guess?</p>
<p>Sketching</p>
<ul>
<li>algorithm to collapse genomes into summary info</li>
<li>fasta -&gt; kmers (“shingle”) -&gt; sample (smartly)</li>
<li>showing set relationships</li>
<li>the “minimum” of a set of sketches gives you useful info on the range you would expect</li>
</ul>
<p>Unions and Intersections</p>
<ul>
<li>space of possible coincidences requires multiple samples to get useful info</li>
<li>bottom 3?</li>
<li>or minimum in 3 partitions?</li>
</ul>
<p>MinHash uses bottom-k approach (Ondov / Phillipy)</p>
<p>Tremendous talk - I feel like I understand what is happening. Invite Ben to talk at your place…I’d imagine you’d get a great talk.</p>
<p>OK, so take <em>log log minimum</em></p>
<p>Daniel Baker wrote:</p>
<ul>
<li>bit.ly/dash_pre</li>
<li>github.com/dnbaker/dashing</li>
<li>github.com/dnbaker/sketch</li>
</ul>
<p>HyperLogLog *HLL)</p>
<ul>
<li>k partition</li>
<li>log<sub>2</sub>n</li>
<li>exponent</li>
<li>average, bias correction</li>
</ul>
<p>Highly vectorized (for speed! Massively parallel calculation)</p>
<ul>
<li>only on intel?</li>
</ul>
<p>Comparing to MinHash</p>
<ul>
<li>HLL deals with lopsided sets better (bottom k not ideal, better minimum k partition)</li>
<li>Uses WAY less memory and wall clock</li>
</ul>
<p>Future stuffs</p>
<ul>
<li>multi-k (kmer lengths)</li>
<li>weighted jaccard</li>
</ul>
</div>
<div id="centroflyeassembling-centromeres-with-long-error-prone-reads" class="section level1">
<h1>centroFlye—Assembling centromeres with long error-prone reads</h1>
<p><strong>Andrey Bzikadze</strong>, Pavel A. Pevzner.</p>
<p><span class="citation">@AndreyBzikadze</span></p>
<p>biorxiv preprint (missed link)</p>
<p>brings up t2t consortium (telomere to telomere)</p>
<p>github.com/nanopore-wgs-consortium/CHM13</p>
<p>mentions karen miga’s chrX t2t biorxiv preprint</p>
<p>centromere:</p>
<ul>
<li>assembler’s favoriate region
<ul>
<li>probably not, but keeps them gainfully employed for now</li>
</ul></li>
<li>highly repetitive….etc etc.</li>
<li>3% of genome - and because are unassembled, poor idea how they influence human diseasse</li>
</ul>
<p>centroFlye</p>
<ul>
<li>classify reads (prefix…something…suffix… what is going on? Andrey already moved on)
<ul>
<li>guessing something to do with begining / middle / end of repeat?</li>
</ul></li>
<li>find rare k-mers that can be used to anchor assembly</li>
<li>but rare k-mers are usually <strong>errors</strong>
<ul>
<li>if the <strong>distance between two rare k-mers is conserved</strong> then they are likely real</li>
<li>neat idea!</li>
<li>but I’m not certain what conservation is…</li>
</ul></li>
<li>throw everything away except for rare k-mers</li>
</ul>
<p>centroFlye used to improve chrX centromere assembly</p>
<ul>
<li>greater length</li>
<li>but how do you more robustly define success?</li>
<li>one metric is to count shared (or discrepant) k-mers between assembly and a read</li>
<li>discorance (a,b) = sharedRead(assembly A) - sharedRead(assembly B)</li>
</ul>
</div>
<div id="genotyping-structural-variants-in-pangenome-graphs-using-the-vg-toolkit" class="section level1">
<h1>Genotyping structural variants in pangenome graphs using the vg toolkit</h1>
<p><strong>Jean Monlong</strong>, Glenn Hickey, David Heller, Jonas Andreas Sibbesen, Jouni Siren, Jordan Eizenga, Eric T. Dawson, Erik Garrison, Adam Novak, Benedict Paten.</p>
<p>Yo time for graphs!</p>
<p>Intro on value over linear genomes</p>
<ul>
<li>highly polymorphic regions should have better geontyping</li>
<li>points out graphs not for discovery but for genotyping</li>
</ul>
<p>Increasing <em>n</em> of SV catalogs with long read data</p>
<ul>
<li>HGSVC (3 samples)</li>
<li>SVPOP (15)</li>
<li>GiaB (1)</li>
</ul>
<p>Probably lots of missing SV from 50-500bp (cmparing HGSVC against gnomadSV which is short-read based)</p>
<p>github.com/vgteam/vg</p>
<p>vg toolkit can make graphs, map reads, and call variants</p>
<p>“can we genotype SVs from short-read datasets with vg toolkit”</p>
<p>Build graph with long-read data, genotype SVs against it with short-reads</p>
<p>R package to evaluate calls</p>
<ul>
<li>github.com/jmonlong/sveval</li>
</ul>
<p>with simulated data</p>
<ul>
<li>vg vs paragraph vs bayestyper vs delly vs SVTyper (last two are non-graph tolls)</li>
<li>basestyper does best (not their tool)</li>
<li>f1 score</li>
</ul>
<p>with real data from HGSVC</p>
<ul>
<li>all do worse</li>
</ul>
<p>simple repeat regions are hard to genotype</p>
<ul>
<li>line, sine, alu,</li>
<li>for both insertion/deletion</li>
</ul>
<p>hard to work with vcf…representation confusion with equivalent representation and oversimplification</p>
<ul>
<li>just use the assemblies?!</li>
<li><p>testing with yeast right now</p></li>
<li><p>vg performance best/near best</p></li>
</ul>
</div>
<div id="rapidly-mapping-raw-nanopore-signal-with-uncalled-to-enable-real-time-targeted-sequencing" class="section level1">
<h1>Rapidly mapping raw nanopore signal with UNCALLED to enable real-time targeted sequencing</h1>
<p><strong>Sam Kovaka</strong>, Yunfan Fan, Winston Timp, Michael C. Schatz.</p>
<p>Don’t want unwanted DNA sequence</p>
<ul>
<li>especially for low throughput sequencers (like ONT)</li>
<li>targeted seq techniques not ideal for ONT
<ul>
<li>length not enough</li>
<li>erase DNA info (methylation, etc.)</li>
</ul></li>
</ul>
<p>ReadUntil</p>
<ul>
<li>selectively start/stop</li>
<li>davemcg: uh, does this actually work yet?
<ul>
<li>was announced a year ago or more??</li>
</ul></li>
</ul>
<p>UNCALLED</p>
<ul>
<li>utility for nanopore current alignment to large expanses of DNA</li>
<li>novel streaming algorithm which maps raw nanopore signale in real-time</li>
<li>works with raw nanopore output (electrical signal)</li>
<li>discussing kmer matching to nanopore output
<ul>
<li>it looks…hard</li>
</ul></li>
<li>insane slide discussing algorith/implementation
<ul>
<li>i am not equipped to summarize</li>
</ul></li>
</ul>
<p>ReadUntil
- enrichment (ejcet a read if it does not map)
- depletion (the converse)</p>
<p>Longer reads get more enrichment</p>
<ul>
<li>saves on seq cost and shorter reads hard to map</li>
</ul>
<p>Testing with bacteria</p>
<ul>
<li>4.5x enrichment of on-target</li>
<li>0.4x off-target</li>
</ul>
<p>Doing a human cancer “panel” for SV</p>
<ul>
<li>28 genes</li>
<li>overall 3X enrichment</li>
<li>and can assemble the genes with much higher success rate</li>
</ul>
<p>Future</p>
<ul>
<li>want to improve yield</li>
<li>is ssDNA “knotting” and blocking ejection?</li>
<li>better API from ONT?</li>
</ul>
<p>github.com/skovaka/UNCALLED</p>
</div>
<div id="the-construct-and-utility-of-reference-pan-genome-graphs" class="section level1">
<h1>The construct and utility of reference pan-genome graphs</h1>
<p><strong>Heng Li.</strong></p>
<p>pangenome = collection of genomes</p>
<ul>
<li>graph (collapse similar seq)</li>
<li>or</li>
<li>compressed full-text index</li>
</ul>
<p>10 years ago (review with Nils Homer): “alignment against multiple genomes will become increasingly important”</p>
<p>“hasn’t happened yet”</p>
<p>vcf doesn’t handle graphs</p>
<p>GFP format (assembly format)</p>
<ul>
<li>davemcg note: looks like a directed(?) acyclic graph</li>
<li>not good because if you split a segment, the <strong>coordinate changes</strong></li>
</ul>
<p>Proposal:</p>
<ul>
<li>reference GFA (GFA <strong>with tags that have coord info and version</strong>)</li>
<li>start with GRCh38, incrementally add other genomes</li>
<li>blacklist and decoy seq for linear tools</li>
<li>updates to preserve coords</li>
</ul>
<p>Incrementally add new assemblies</p>
<ul>
<li>add assembly, make graph</li>
<li>then add another assembly, make new graph</li>
</ul>
<p>Discussion of alignment linear seq to graph…which I don’t quite follow</p>
<ul>
<li>but ideal approach too slow</li>
<li>so approximate with k shortest paths</li>
</ul>
<p>minigraph</p>
<ul>
<li>based on minimap2</li>
<li>limitations!
<ul>
<li>doesn’t work with dense graphs (too many k paths?)</li>
</ul></li>
<li>1.5 hours over 24 CPUs with human graph of 20 haplotypes
<ul>
<li>36k bubbles</li>
<li>94% of GRCh37 is invariable</li>
</ul></li>
<li><a href="https://github.com/lh3/minigraph" class="uri">https://github.com/lh3/minigraph</a> (didn’t give link but it’s on his github)</li>
</ul>
<p>multi-allelic regions and minisatellites are hard to assemble and genotype</p>
<p>applications</p>
<ul>
<li>blacklist regions (SV, etc)</li>
</ul>
</div>
<div id="princess-a-framework-for-comprehensive-detection-and-phasing-of-snps-and-structural-variants" class="section level1">
<h1>PRINCESS — A framework for comprehensive detection and phasing of SNPs and structural variants</h1>
<p><strong>Medhat Mahmoud</strong>, Winston Timp, Fritz J. Sedlazeck.</p>
<p>Burn. Winston Timp not on the slide.</p>
<p><span class="citation">@MedhatHelmy7</span></p>
<p>Points out there are diff tech to detect SNV, SV, Phasing, Methylation</p>
<p>Would prefer just use one platform</p>
<ul>
<li>ONT?</li>
</ul>
<p>PRINCESS</p>
<ul>
<li><strong>framework</strong> to integrate tools to analyze Long Reads</li>
<li>mapping: minimap2 and/or NGMLR</li>
<li>SNVs: Clair (deep NN) - first NN mention?</li>
<li>SVs: Sniffles</li>
<li>Phasing: WhatsHap, Princess-subtools</li>
<li>can also use trio info (SNPs)</li>
<li>Nanopolish for Methyl C</li>
<li>outputs….statistics!</li>
<li>runs on workstation or hpc or cloud</li>
<li>outputs phase SNV, SV with optional MethylC</li>
</ul>
<p>Benchmarking with GiaB HG002</p>
<ul>
<li>PacBio CCS, PacBio CLR, ONT</li>
<li>LOOOOOONG tail for ONT (read length dist)</li>
<li>remove &lt;500bp in length</li>
<li>try diff coverage (10x, 25x, 50x, 95x)</li>
<li>with one tech (70-80% sensitivity, 85-93% precision for SNP)
<ul>
<li>PacBio CCS &gt; CLR &gt; ONT</li>
</ul></li>
<li>SV calling
<ul>
<li>CCS &gt; CLR == ONT (80% sensitivity, 85-95% precision)</li>
</ul></li>
<li>Phasing
<ul>
<li>Use parental SNPs</li>
<li>N50 better with ONT (loooong reads)</li>
<li>but better accuracy with PacBio</li>
</ul></li>
</ul>
<p>Now real data</p>
<ul>
<li>Common Disease Genomics</li>
<li>focus on cardiovascular and … something else missed it</li>
<li>44k+ short read WGS -&gt; do some with long read -&gt; merge to make comprehensive genome -&gt; which can be used to improve catalogues</li>
<li>SVcollector (github.com/fritzsedlazeck/SVCollector)</li>
<li>then Princess (github.com/MeHelmy/princess)</li>
<li>then Paragraph (github.com/Illumina/paragraph)
<ul>
<li>uh, what is paragraph</li>
<li>from github: &quot; A graph-based structural variant genotyper for short-read sequence data.&quot;</li>
<li>oh, that makes sense</li>
<li>so use 44k short reads on graph?</li>
</ul></li>
</ul>
</div>
<div id="efficient-chromosome-scale-haplotype-resolved-assembly-of-human-individuals" class="section level1">
<h1>Efficient chromosome-scale haplotype-resolved assembly of human individuals</h1>
<p><strong>Shilpa Garg</strong>, Arkarachai Fungtammasan, Anthony Schmitt, Andrew
Carroll, Paul Pelusol, Emily Hatas, Fritz Sedlazeck, Justin Zook, Mike
Chou, John Aach, Jason Chin, Heng Li, George Church.</p>
<p>Discussing PacBio HiFi tech</p>
<ul>
<li>circularize DNA</li>
<li>15k long</li>
<li>99% with 5 passes</li>
</ul>
<p>Now Hi-C discussion/intro</p>
<ul>
<li>wild how common Hi-C is. I guess I’m getting old - in grad school this stuff was just IMPOSSIBLE to do</li>
</ul>
<p>now genome assembly discussion</p>
<ul>
<li>graph geneeration …. getting longer reads gives you fewer branches on your graph (contiguous assembly)</li>
</ul>
<p>PacBio alone not enough for full assembly/phasing</p>
<ul>
<li>but can use trios with Koren/Phillipy approach (at least works in goats)
<ul>
<li>trio canu</li>
</ul></li>
<li>but want full haplotypes <strong>without</strong> trios</li>
</ul>
<p>workflow</p>
<ul>
<li>PacBio CCS</li>
<li>make contigs</li>
<li>use Hi-C to scaffold</li>
<li>peregrine -&gt; hirrise -&gt; deepvariant -&gt; whatshap + hapcut2 =&gt; whatshap -&gt; peregrine</li>
<li>walltime: <strong>one day</strong></li>
</ul>
<p>benchmark with PGP1, HG002, NA12878</p>
<ul>
<li>contig n50, NGA50, phasing hamming error, phasing switch error</li>
<li>compare against trio canu</li>
<li>generally better than trio canu across all metrics</li>
<li>consistent across diff genomes</li>
</ul>
<p>checking in HLA region</p>
<ul>
<li>can build HLA in two contigs</li>
</ul>
<p>Link!</p>
<ul>
<li>github.com/shilpagarg/WHdenovo</li>
</ul>
</div>
<div id="utilization-of-an-ensemble-approach-for-identification-of-driver-fusions-in-pediatric-cancer" class="section level1">
<h1>Utilization of an ensemble approach for identification of driver fusions in pediatric cancer</h1>
<p><strong>Stephanine LaHaye</strong>, Kyle Voytovich, James Fitch, Natalie Bir, Sean D. McGrath, Anthony Miller, Amy Wetzel, Vincent Magrini, Catherine E. Cottrell, Elaine R. Mardis, Richard K. Wilson.</p>
<p>Last talk of the session!</p>
<p>ID fusions in cancer (pediatric)</p>
<p>Interesting….pediatric cancer have fwer mutations (which I guess makes sense)</p>
<ul>
<li>SN Grobner et al Nature 2018</li>
<li>fusions are helpful to classify cancers</li>
<li>difficult via RNA-seq</li>
</ul>
<p>Used 4 (6?) different callers for an ensembl approach</p>
<ul>
<li>oh god a multi venn diagram</li>
<li>USE AN UpSET plot!!!!</li>
<li>anyways, not much agreement (but low n is a feature if they are true)</li>
<li>not clear how they actually <strong>know</strong> most are false positive</li>
<li>specificity != sensitivity</li>
</ul>
<p>LOL Runtimes</p>
<ul>
<li>vary from 30 minutes to 75 hours</li>
<li>dragen the fast one</li>
</ul>
<p>Synthetic fusion gene cDNA for benchmarking</p>
<ul>
<li>more dilution == harder to detect</li>
<li>1:50 dilution == OK, 1:250 == not so good</li>
</ul>
<p>Running this crazy big complicated pipeline in AWS serveeless config</p>
<ul>
<li>looks like they are paying AWS a lot of $</li>
</ul>
</div>
