---
title: "#GI2019 2019 11 08 Morning Session"
author: David McGaughey
date: '2019-11-08'
slug: gi2019-day03-morning
categories:
  - bioinformatics
  - gi2019
  - conferences
  - talk
tags:
  - bioinformatics
  - gi2019
  - conferences
  - talk
header:
  caption: ''
  image: ''
output:
  blogdown::html_page:
    toc: true
---


<div id="TOC">
<ul>
<li><a href="#day-3---session-7---microbial-and-metagenomics">Day 3 - Session 7 - MICROBIAL AND METAGENOMICS</a></li>
<li><a href="#strain-level-metagenomic-assignment-and-compositional-estimation-for-long-reads-with-metamaps">Strain-level metagenomic assignment and compositional estimation for long reads with MetaMaps</a></li>
<li><a href="#metaflyescalable-long-read-metagenome-assembly-using-repeat-graphs">metaFlye—Scalable long-read metagenome assembly using repeat graphs</a></li>
<li><a href="#detecting-microbial-transmission-and-engraftment-after-faecal-microbiota-transplants-using-long-read-metagenomics-and-reticulatus">Detecting microbial transmission and engraftment after faecal microbiota transplants using long-read metagenomics and reticulatus</a></li>
<li><a href="#entropy-of-a-bacterial-stress-response-is-a-generalizable-predictor-for-fitness-and-antibiotic-sensitivity">Entropy of a bacterial stress response is a generalizable predictor for fitness and antibiotic sensitivity</a></li>
<li><a href="#the-use-of-kmer-counts-to-train-random-forests-to-predict-country-of-origin-for-bacterial-pathogen-sequencing-data">The use of kmer counts to train random forests to predict country of origin for bacterial pathogen sequencing data</a></li>
<li><a href="#real-time-assembly-using-nanopore-sequencing-data-for-microbial-communities">Real-time assembly using Nanopore sequencing data for microbial communities</a></li>
<li><a href="#exploring-the-role-of-ribosomal-gene-repeats-in-the-context-of-regeneration">Exploring the role of ribosomal gene repeats in the context of regeneration</a></li>
<li><a href="#genomic-epidemiology-of-west-nile-virus-in-california">Genomic epidemiology of West Nile virus in California</a></li>
</ul>
</div>

<div id="day-3---session-7---microbial-and-metagenomics" class="section level1">
<h1>Day 3 - Session 7 - MICROBIAL AND METAGENOMICS</h1>
<p>Genome Informatics 2019 at CSHL</p>
<p><strong>Bold</strong> is the speaker</p>
<p><strong>If you dislike/disagree with my notes/sentiment and you are the speaker/PI then contact me. I very much could be mis-understanding some important points.</strong></p>
<p>My notes will likely reflect my background knowledge of microbes. Which is nil.</p>
</div>
<div id="strain-level-metagenomic-assignment-and-compositional-estimation-for-long-reads-with-metamaps" class="section level1">
<h1>Strain-level metagenomic assignment and compositional estimation for long reads with MetaMaps</h1>
<p><strong>Alexander T. Dilthey</strong>, Chirag Jain, Sergey Koren, Adam M. Phillippy.</p>
<p>Discussion of how MinHash works.</p>
<p>kmers, which are hashed, then sorted (take the smallest), compare kmers between query and ref</p>
<p>MinHash ~ Jaccard similarity (point to assess similarity, e.g. “alignment”)</p>
<p>“Mapping qualities”</p>
<ul>
<li>want prob. dist. of all mapping locations for a read</li>
<li>assume
<ul>
<li>read error rate known</li>
</ul></li>
<li>compute jaccard similarity</li>
<li>model as Bernoulli</li>
</ul>
<p>MetaMaps</p>
<ul>
<li>integrated metagenomic classification system</li>
<li>takes lots from MashMap</li>
<li>auto-tunes minimum read length &amp; identity (EM)</li>
</ul>
<p>Benchmark</p>
<ul>
<li>precision better than others (Kraken, Kraken2, Centrifuge, MEGAN-LR)</li>
<li><strong>REQUIRES LONG READS (&gt; 1000bp)</strong></li>
<li>slower than other methods (and takes more mem)
<ul>
<li>but are working on this</li>
</ul></li>
</ul>
<p>Going through some case studies on finding novel things</p>
<ul>
<li>find mismatches from errors in ref databases</li>
<li>location mapping allows you to quantify categories of genes (or I totally misheard/misread)</li>
</ul>
<p>github.com/DiltheyLab/MetaMaps</p>
</div>
<div id="metaflyescalable-long-read-metagenome-assembly-using-repeat-graphs" class="section level1">
<h1>metaFlye—Scalable long-read metagenome assembly using repeat graphs</h1>
<p><strong>Mikhail Kolmogorov</strong>, Mikhail Rayko, Jeffrey Yuan, Evgeny Polevikov, Pavel Pevzner.</p>
<p>Genome Assembly, with graphs</p>
<p>Flye intro:
- glue repeats into the graph</p>
<p>OK, I’m not following this. Sorry Mikhail!</p>
<p>Isn’t this a metagenome session? This is all genome assembly.</p>
<p>Oh, he is shifting now.</p>
<p>How to detect repeats?</p>
<ul>
<li>in genomes can use coverage</li>
<li>not reliable enough in metagenomes</li>
<li>need to use graph</li>
</ul>
<p>Emphasizes look at graph viz</p>
<ul>
<li>want one loop for bacteria chr</li>
<li>reality sucks</li>
<li>bubbles oh my
<ul>
<li>strain hetergeneity?</li>
<li>or repeat -&gt; bubble -&gt; repeat?!</li>
</ul></li>
<li>BUBBLES CAN HAVE BUBBLES</li>
</ul>
<p>Showing cow rumen microbiome</p>
<p>Now Sheep microbiome</p>
<p>Looks like many successes and many crazy things</p>
<p>Emphasizes metaFlye is still very much in dev.</p>
<p>Recommends Bandage (Wick et al) and AGB (Mikheenko &amp; Kolmogorov) for viz of assemblies</p>
<p>Asks us - would you prefer simple consensus or bubbly haplotype resolved assembly?</p>
<p>github.com/fenderglass/Flye</p>
</div>
<div id="detecting-microbial-transmission-and-engraftment-after-faecal-microbiota-transplants-using-long-read-metagenomics-and-reticulatus" class="section level1">
<h1>Detecting microbial transmission and engraftment after faecal microbiota transplants using long-read metagenomics and reticulatus</h1>
<p><strong>Samuel M. Nicholls</strong>, Joshua C. Quick, Tariq H. Iqbal, Nicholas J. Loman.</p>
<p><span class="citation">@samstudio8</span></p>
<p>Intro on why microbiome is important</p>
<ul>
<li>health</li>
<li>exciting as field isn’t fully define for best practices</li>
</ul>
<p>“three peaks challenge”</p>
<ul>
<li>want extraction method that maintains composition (equal representation of strain?) and long reads</li>
</ul>
<p>(everyone is so coy about what is a faecal transplant)</p>
<p>anyways….checking human microbiomes pre/post transplant</p>
<p>get looooong reads</p>
<p>showing n50, contig nums, etc. they look uh, big? (don’t have a great sense about what is acceptable)</p>
<p>diving into communities. oooooo very cool plot (bubble plot?) with families of bacteria. so many complete (?) genomes</p>
<p>need to do assembly polishing, but expensive computation</p>
<p>github.com/samstudio8/reticulatus (snakemake pipeline for polishing)</p>
<p>tech discussion of GPU acceleration of steps in pipeline</p>
<ul>
<li>100 minutes -&gt; 5 minutes with GPU acceleration</li>
<li>10 - 5 hours on promethION</li>
</ul>
<p>File parsers!</p>
<ul>
<li>poor gzip FQ parser</li>
<li>port heng li readfq…save 40 minutes</li>
</ul>
<p>Huge time savings on real data</p>
<p>back to data</p>
<ul>
<li>showing diffs pre/post transplantation
<ul>
<li>novel communities?</li>
</ul></li>
<li>how do they find these? eyeballing? diff testing?</li>
</ul>
</div>
<div id="entropy-of-a-bacterial-stress-response-is-a-generalizable-predictor-for-fitness-and-antibiotic-sensitivity" class="section level1">
<h1>Entropy of a bacterial stress response is a generalizable predictor for fitness and antibiotic sensitivity</h1>
<p><strong>Defne Surujon</strong>, Zeyu Zhu, Juan C. Ortiz-Marquez, José Bento, Tim van Opijnen.</p>
</div>
<div id="the-use-of-kmer-counts-to-train-random-forests-to-predict-country-of-origin-for-bacterial-pathogen-sequencing-data" class="section level1">
<h1>The use of kmer counts to train random forests to predict country of origin for bacterial pathogen sequencing data</h1>
<p><strong>Lauren Cowley</strong>, Jordan Taylor, Claire Jenkins, Tim Dallman.</p>
</div>
<div id="real-time-assembly-using-nanopore-sequencing-data-for-microbial-communities" class="section level1">
<h1>Real-time assembly using Nanopore sequencing data for microbial communities</h1>
<p><strong>Son H. Nguyen</strong>, Minh D. Cao, Lachlan J. Coin.</p>
</div>
<div id="exploring-the-role-of-ribosomal-gene-repeats-in-the-context-of-regeneration" class="section level1">
<h1>Exploring the role of ribosomal gene repeats in the context of regeneration</h1>
<p><strong>Sofia N. Barreira</strong>, Andreas D. Baxevanis.</p>
</div>
<div id="genomic-epidemiology-of-west-nile-virus-in-california" class="section level1">
<h1>Genomic epidemiology of West Nile virus in California</h1>
<p><strong>Karthik Gangavarapu</strong>, Nathaniel Matteson, on behalf of the WestNile 4K Project.</p>
</div>
