---
title: "#GI2019 2019 11 06"
author: David McGaughey
date: '2019-11-06'
slug: gi2019-day01
categories:
  - bioinformatics
  - gi2019
  - conferences
  - talk
tags:
  - bioinformatics
  - gi2019
  - conferences
  - talk
header:
  caption: ''
  image: ''
output:
  blogdown::html_page:
    toc: true
---


<div id="TOC">
<ul>
<li><a href="#day-1---session-1---genome-structure-and-function">Day 1 - Session 1 - GENOME STRUCTURE AND FUNCTION</a><ul>
<li><a href="#comparative-3d-genome-organization-in-apicomplexan-parasites">Comparative 3D genome organization in Apicomplexan parasites</a></li>
<li><a href="#unscrambling-the-tumor-genome-via-integrated-analysis-of-structural-variation-and-copy-number">Unscrambling the tumor genome via integrated analysis of structural variation and copy number</a></li>
<li><a href="#tissue-specific-enhancer-functional-networks-for-associating-distal-regulatory-regions-to-disease">Tissue-specific enhancer functional networks for associating distal regulatory regions to disease</a></li>
<li><a href="#targeted-nanopore-sequencing-with-cas9-for-studies-of-methylation-structural-variants-and-mutations">Targeted Nanopore sequencing with Cas9 for studies of methylation, structural variants, and mutations</a></li>
<li><a href="#long-read-sequencing-of-structurally-variant-genomes">Long-read sequencing of structurally variant genomes</a></li>
<li><a href="#exploring-the-3d-spatial-dependency-of-gene-expression-using-markov-random-fields">Exploring the 3D spatial dependency of gene expression using Markov random fields</a></li>
<li><a href="#mapping-cis-eqtl-from-rna-seq-data-with-no-genotypes">Mapping cis-eQTL from RNA-seq data with no genotypes</a></li>
<li><a href="#exploring-short-tandem-repeat-expansions-at-both-known-and-novel-loci-in-the-human-genome">Exploring short tandem repeat expansions at both known and novel loci in the human genome</a></li>
</ul></li>
</ul>
</div>

<div id="day-1---session-1---genome-structure-and-function" class="section level1">
<h1>Day 1 - Session 1 - GENOME STRUCTURE AND FUNCTION</h1>
<p>Genome Informatics 2019 at CSHL</p>
<p><strong>Bold</strong> is the speaker</p>
<p>349 attendees. Largest?</p>
<p>189 posters</p>
<p>39 (?) talks</p>
<p>90 countries, 20 (US) states</p>
<p>High(est) % female, URM for gi</p>
<div id="comparative-3d-genome-organization-in-apicomplexan-parasites" class="section level2">
<h2>Comparative 3D genome organization in Apicomplexan parasites</h2>
<p>Evelien M. Bunnik, Aarthi Venkat, Ferhat Ay, <strong>Karine G. Le Roch.</strong></p>
<p>Apicomplexan parasites:</p>
<ul>
<li>toxoplasma gondii</li>
<li>babesia</li>
<li>most prevalent nad morbidity causing pathogens</li>
</ul>
<p>Try to understand at genomic level</p>
<p>Malaria (plasmodium) genome published 18 years ago</p>
<ul>
<li>82% AT rich</li>
<li>grab bag of major -omics techniques</li>
<li>still poor understanding of gene regulation</li>
<li>believe under-representation of TFs identified (lower rate than human, yeast)</li>
</ul>
<p>Did Hi-C (chrmatin conformation capture)</p>
<ul>
<li>at many stages</li>
<li>100kb resolution (at least for later work?)</li>
<li>going through their literature of chromosome organization (Ay, Bunnik Genome Research 2014)</li>
</ul>
<p><em>Virulence</em> Genes</p>
<ul>
<li>mediates adherence of infected RGC to endothelium of infected RBC</li>
<li>altered chromatin structure at some of these genes</li>
</ul>
<p>More examples of loops for other pathogenesis processes</p>
<ul>
<li>see power of doing many stages (stage specific patterning of contact maps)</li>
</ul>
<p>Did more Hi-C across other parasites, unfortunately had to re-assemble several genomes</p>
<ul>
<li>Bunnik PNAS 2019</li>
<li>more expression at genes near (in 3D space) telomore</li>
<li>which is where many virulence genes are …</li>
</ul>
</div>
<div id="unscrambling-the-tumor-genome-via-integrated-analysis-of-structural-variation-and-copy-number" class="section level2">
<h2>Unscrambling the tumor genome via integrated analysis of structural variation and copy number</h2>
<p><strong>Daniel L. Cameron</strong>, Charles Shale, Jonathan Baber, Anthony T. Papenfuss, Peter Priestly.</p>
<p>Hartwig has a fully automated oncology report (“no touch”)</p>
<ul>
<li>needs to be highly sensitive and specific</li>
<li>use 110x tumor, 45x germline</li>
<li>SV and CNV calling not ideal</li>
</ul>
<p>GRIDSS -&gt; PURPLE -&gt; LINX</p>
<p>SV-&gt; CNV -&gt; Interpretation</p>
<p><a href="https://github.com/PapenfussLab/gridss" class="uri">https://github.com/PapenfussLab/gridss</a></p>
<p>full pipeline: <a href="https://github.com/hartwigmedical/gridss-purple-linx" class="uri">https://github.com/hartwigmedical/gridss-purple-linx</a></p>
<p>work on phasing SV (cis vs trans vs unknown)</p>
<ul>
<li>only good up to about 300bp</li>
<li>but events are clustered, so works better than you’d expect</li>
</ul>
<p>Benchmarking (includes GRIDSS)</p>
<ul>
<li><a href="https://genomebiology.biomedcentral.com/articles/10.1186/s13059-019-1720-5" class="uri">https://genomebiology.biomedcentral.com/articles/10.1186/s13059-019-1720-5</a></li>
</ul>
</div>
<div id="tissue-specific-enhancer-functional-networks-for-associating-distal-regulatory-regions-to-disease" class="section level2">
<h2>Tissue-specific enhancer functional networks for associating distal regulatory regions to disease</h2>
<p><strong>Xi Chen</strong>, Jian Zhou, Chandra L. Theesfeld, Ran Zhang, Aaron K. Wong, Olga G. Troyanskaya.</p>
<p>How to predict function of enhancers?</p>
<p>Different ways enhancers associated with disease:</p>
<ul>
<li>enhancer -&gt; gene (near)</li>
<li>enhancer(with SNP inside)</li>
<li>enhancer(rare mutation inside)</li>
<li>enhance without direct evidence</li>
</ul>
<p>Hypothesis:</p>
<ul>
<li>enhancer will have a functional relationship with another enhancer if:
<ul>
<li>3d interaction</li>
<li>co-activation with TF</li>
<li>regulate related genes</li>
</ul></li>
<li>if all met, then <em>maybe</em> disease associated</li>
</ul>
<p>bayesian inference to model the above assumptions</p>
<p>how to see if this sort of works?</p>
<ul>
<li>check brain specific super enhancers in non-brain tissues</li>
<li>get some enrichment</li>
</ul>
<p><a href="https://fenrir.flatironinstitute.org" class="uri">https://fenrir.flatironinstitute.org</a></p>
<ul>
<li>web tool</li>
<li>no preprint or paper?</li>
</ul>
<p>dmcg note:</p>
<ul>
<li>seems like you’d really want some en-masse validation. validating a few sites doesn’t seem like enough.</li>
<li>i missed what the input data for these models are (public stuff? Internal stuffs?)</li>
</ul>
</div>
<div id="targeted-nanopore-sequencing-with-cas9-for-studies-of-methylation-structural-variants-and-mutations" class="section level2">
<h2>Targeted Nanopore sequencing with Cas9 for studies of methylation, structural variants, and mutations</h2>
<p><strong>Timothy E. Gilpatrick</strong>, Isac Lee, Fritz J. Sedlazek, Winston Timp.</p>
<p>Want to enrich regions for genome</p>
<p>ROI -&gt; cut out with CRISPR (cas9 on both sides of ROI) -&gt; dA tail -&gt; adapter ligate</p>
<ul>
<li>oh even cutting just once helps with enrichment. cool.</li>
<li>can target 10 sites simulateously (400X+ coverage with MinION)</li>
</ul>
<p>Tried with cancer model</p>
<p>On-target perentage (reads) is about 1-4%</p>
<ul>
<li>most off-target randomly distributed</li>
<li>very little off-target cutting</li>
</ul>
<p>5% error rate!</p>
<ul>
<li>homopolymers sucks</li>
<li>guppy v3 does better job at SNV calling</li>
</ul>
<p>did downsampling test to check SNV performance</p>
<ul>
<li>tools
<ul>
<li>mpileup</li>
<li>Nanopolish (Simpson et al. ?)</li>
<li>Medaka (ONT)</li>
<li>Clair (Luo et al.)</li>
</ul></li>
<li>can get 95% sesnitivity…doesn’t seem to great</li>
<li>Clair gets worse with more coverage (trained on 25X apparently)</li>
</ul>
<p>HE IS GOING SO FAST OMG</p>
<p>More discussion of benchmarking
- still in SNV calling</p>
<p>Now we are talking phasing</p>
<ul>
<li>in TP53 find all true variants, no FP</li>
</ul>
<p>Suggestion of CNV with phased data, more copies of one allele (60X vs 20X)</p>
<p>Now CpG</p>
<ul>
<li>enriched stuff can still get detectable CpG calls</li>
</ul>
<p>SV &amp; Repetitive</p>
<ul>
<li>still works</li>
</ul>
<p>dmcg note:</p>
<ul>
<li>everything still works (isn’t that expected?)</li>
<li>on-target percentage VERY low (1-4% means you are “throwing” <strong>lots</strong> of data away)</li>
<li>and yield is getting hammered…so doubly bad</li>
<li>not doing anything like getting hundreds of genes…which I would think would be really necessary to be useful clinically</li>
<li>size smaller? (length of DNA reads)</li>
</ul>
</div>
<div id="long-read-sequencing-of-structurally-variant-genomes" class="section level2">
<h2>Long-read sequencing of structurally variant genomes</h2>
<p><em>Evan E. Eichler.</em></p>
<p>4% duplication</p>
<p>49% of normal CNV maps to this 4% of genome</p>
<p>Long reads increase resolution of SV</p>
<p>SV detection comaprison between Illumina and PacBio (30k with PB, 11k with Illumina)</p>
<ul>
<li><strong>70% of SV missed with just Illumina data</strong></li>
</ul>
<p>Did SV analysis with LR seq method across diverse human genomes</p>
<ul>
<li>found near 100k resolved SV</li>
<li>exciting to see a plateau of novel SV! (east africa most unique, as expected)</li>
</ul>
<p>Improve SV calling with short read data with improved genomes</p>
<ul>
<li>can find new eQTL</li>
</ul>
<p>Goal: t2t (telomere 2 telomere for chromosome)</p>
<ul>
<li>t2t consortium</li>
</ul>
<p>Solution:</p>
<ul>
<li>HiFi PacBio 99.9% acccurate 18kbp</li>
<li>ONT gives you ultra long (whale!) reads</li>
</ul>
<p>HiFi &amp; Strandseq(?) to build referene-free phased human genomes</p>
<p>hifi &amp; canu -&gt; saarclust &amp; strandseq -&gt; deepvariant -&gt; whatshap &amp; strandseq -&gt; canu/peregrine</p>
<p>assemboy -&gt; contigs -&gt; snp calls -&gt; haplotype CCS reads -&gt; assembly</p>
<p>23 chr (“clusters”)</p>
<p>95% SNP phase</p>
<p>N50 = 28 Mbp</p>
<p>problematic regions</p>
<ul>
<li>canu and peregrine fail</li>
<li>250 in count</li>
<li>60% are supplemental dups</li>
<li>18% SV</li>
<li>23% acrocentric</li>
</ul>
<p>solutions</p>
<ul>
<li>build graghs around high dup regions of human genome</li>
<li>“SDA” assembler</li>
</ul>
<p>Long reads getting centrome seq of chr8 (t2t consortium)</p>
</div>
<div id="exploring-the-3d-spatial-dependency-of-gene-expression-using-markov-random-fields" class="section level2">
<h2>Exploring the 3D spatial dependency of gene expression using Markov random fields</h2>
<p><strong>Naihui Zhou</strong>, Iddo Friedberg, Mark S. Kaiser.</p>
<p>The quantitative expression of gene regulation could manifest as spatial dependency</p>
<p>Hypothesis:</p>
<ul>
<li>TF factories (clustering of genes for active transcription)</li>
<li>hub-enhancers (shared enhancers for many genes)</li>
</ul>
<p>Propose adding spatial location of genes to improve modeeling of gene-gene variation (which is already modelled in most diff expression analyses, e.g. limma)</p>
<p>Poisson Hierarchical Markov Random Field Model (PHiMRF)</p>
<ul>
<li>uses HiC data to get pairwise gene interactions</li>
<li>two genes called neighbors if HiC interaction &gt; threshold</li>
<li>making a <em>spatial gene network</em></li>
<li>genes are nodes, edges are HiC interaction</li>
</ul>
<p>New random variable Y<sub>ik</sub> which follows binomial dist (? missed this)</p>
<p>Do genes in TAD have spatial dependencies?</p>
<ul>
<li>but how to deal with interTAD and intraTAD (within and between)?</li>
<li>run model with both types of edges</li>
<li>Yes…more connected than bootstraping tests</li>
</ul>
<p>davemcg note: Still waiting to see if/how this improves diff expression?</p>
<p>Enriched GO terms for most conncted gene groups</p>
<ul>
<li>tx by RNA polII</li>
<li>GPCR signalling</li>
<li>negative cell proliferation</li>
</ul>
<p>R package!
<a href="https://github.com/ashleyzhou972/PhiMRF" class="uri">https://github.com/ashleyzhou972/PhiMRF</a></p>
</div>
<div id="mapping-cis-eqtl-from-rna-seq-data-with-no-genotypes" class="section level2">
<h2>Mapping cis-eQTL from RNA-seq data with no genotypes</h2>
<p><strong>Elena Vigorito</strong>, Simon White, Chris Wallace.</p>
<p>Can’t always get genotypes….so would like eQTL without them….</p>
<p>Impute cis-SNP (G<sub>i</sub>)</p>
<p>negative binomial weighted by probable genotype</p>
<p>work in 500kb window to compute gene - SNP pairings</p>
<p>benchmark against GEUVADIS reference data</p>
<ul>
<li>only chr22 (why? performance?)</li>
<li>better sensitivity and specificity</li>
</ul>
<p>R package out “soon”</p>
<p><em>only method for eQTL without genotypes</em></p>
</div>
<div id="exploring-short-tandem-repeat-expansions-at-both-known-and-novel-loci-in-the-human-genome" class="section level2">
<h2>Exploring short tandem repeat expansions at both known and novel loci in the human genome</h2>
<p><strong>Harriet Dashnow</strong>, Brent Pedersen, Daniel MacArthur, Alicia Oshlack, Aaron Quinlan.</p>
<p><span class="citation">@hdashnow</span> (twitter handle! good!)</p>
<p>STR is 1-6bp repeats. Microsatellites.</p>
<p>3% of human genome with high mutation rate and high polymorphism</p>
<p>Huntington’s, fragile X, spinocerebellar ataxias</p>
<p>Classically, longer is worse prognosis / higher severity</p>
<p>STR alleles are WAY longer than short reads</p>
<p>Harriet looking for NOVEL (str NOT IN REFERENCE GENOME)</p>
<ul>
<li>not so many methods fornovel (just ExpansionHunter? as of 2 months ago)</li>
</ul>
<p>STR reads often mis-mapped</p>
<ul>
<li>read STR…where does it go???</li>
<li>sends it to just whatever STR or nowhere if really novel</li>
</ul>
<p>STRling!</p>
<ul>
<li><a href="https://github.com/quinlan-lab/STRling" class="uri">https://github.com/quinlan-lab/STRling</a></li>
<li>use kmers to find STR</li>
<li>if read passes kmer threahold, then is a STR … then re-map</li>
<li>for longer pairs use linear regression to predict allele size</li>
</ul>
<p>Validated acdross 9 (7 worked) disease expansions</p>
<p>Found 5/5 novel expansions in CANVAS disease</p>
<p>Also validation with PacBio</p>
<p>Limitations! (yay! seriously love this)</p>
<ul>
<li>limited to large (&gt;50 bp) STR expansions</li>
</ul>
</div>
</div>
