---
title: "#GI2019 2019 11 08 Afternoon Session"
author: David McGaughey
date: '2019-11-08'
slug: gi2019-day03-afternoon
categories:
  - bioinformatics
  - gi2019
  - conferences
  - talk
tags:
  - bioinformatics
  - gi2019
  - conferences
  - talk
header:
  caption: ''
  image: ''
output:
  blogdown::html_page:
    toc: true
---


<div id="TOC">
<ul>
<li><a href="#day-3---session-8---evolution-and-phylogenetics">Day 3 - Session 8 - EVOLUTION AND PHYLOGENETICS</a></li>
<li><a href="#is-pathogen-evolution-predictable-the-role-of-population-genomics">Is pathogen evolution predictable? The role of population genomics</a></li>
<li><a href="#learning-the-properties-of-adaptive-regions-with-functional-data-analysis">Learning the properties of adaptive regions with functional data analysis</a></li>
<li><a href="#creating-pan-human-and-population-specific-consensus-representations-of-the-reference-genome-and-assessing-their-effect-on-functional-genomic-data-analysis">Creating pan-human and population-specific consensus representations of the reference genome and assessing their effect on functional genomic data analysis</a></li>
<li><a href="#gramene-subsitespangenome-browsers-for-crops">Gramene subsites—Pangenome browsers for crops</a></li>
<li><a href="#section"></a></li>
<li><a href="#what-do-we-gain-when-tolerating-loss-the-information-bottleneck-lossy-compression-and-detecting-horizontal-gene-transfer">What do we gain when tolerating loss? The information bottleneck, lossy compression, and detecting horizontal gene transfer</a></li>
<li><a href="#a-recurrent-neural-network-for-inferring-sweeps-and-allele-frequency-trajectories-using-gene-trees-based-on-the-ancestral-recombination-graph">A recurrent neural network for inferring sweeps and allele frequency trajectories using gene trees based on the ancestral recombination graph</a></li>
<li><a href="#assembling-the-y-chromosomes-of-anopheles-mosquitoes">Assembling the Y chromosomes of anopheles mosquitoes</a></li>
</ul>
</div>

<div id="day-3---session-8---evolution-and-phylogenetics" class="section level1">
<h1>Day 3 - Session 8 - EVOLUTION AND PHYLOGENETICS</h1>
<p>Genome Informatics 2019 at CSHL</p>
<p><strong>Bold</strong> is the speaker</p>
<p><strong>If you dislike/disagree with my notes/sentiment and you are the speaker/PI then contact me. I very much could be mis-understanding some important points.</strong></p>
</div>
<div id="is-pathogen-evolution-predictable-the-role-of-population-genomics" class="section level1">
<h1>Is pathogen evolution predictable? The role of population genomics</h1>
<p><strong>William Hanage.</strong></p>
<p>Is pathogen evolution predictable?</p>
<ul>
<li>Probably not</li>
<li>Cool, talk over, I guess.</li>
</ul>
<p>Oh he is still going</p>
<p>“Accessory genome”</p>
<ul>
<li>three <em>e. coli</em> strains only share 39% of the genes
<ul>
<li>Welsh et al…….2001 (?)</li>
</ul></li>
<li>the “mr. potato head” model</li>
</ul>
<p>Sample 4k+ pneumococcal genomes from 4 dif places</p>
<ul>
<li>Corander Nat Eco Evol 2017</li>
<li>found 73 “sequence clusters” based on population (tree?) structure</li>
<li>also 1700+ accessory genes (COGS)</li>
</ul>
<p><strong>Dif strains are correlated when using accessory genes (was effectively R^2 of 0 with full genome)</strong></p>
<p>Negative Frequency Dependent Selection</p>
<ul>
<li>balancing selection</li>
<li>corrolary are antigens/phage receptors - if too common get ID’ed and wiped out</li>
</ul>
<p>Pneumococcal Conjugate vaccines (PCVs) as an evolutionary experiment</p>
<ul>
<li>we have good vaccines against <em>some</em> strains</li>
<li>7</li>
<li>but there are 90+ serotypes</li>
</ul>
<p>Can we predict what types will persist/succeed after vaccination?</p>
<p>Sampling (seq.) per/post vaccination</p>
<ul>
<li>35 strains found</li>
<li>vaccination wiped out the 7 that were covered</li>
<li>uneven “replacement” rate by other strains
<ul>
<li>not super convinced … is this just noise?</li>
</ul></li>
</ul>
<p>Isolates in the same group tend to have the same set of COGS (getting increasingly concerned I don’t know what COGS stands for…)</p>
<p>Check for prevalence of COGS pre/post vaccination and sum distance from linear expected.</p>
<ul>
<li>directional</li>
</ul>
<p>Can predict fitness … and it roughly matches what you’d expect. Not perfect, but the correlation looks real.</p>
<p>doi.org/10.1101/420315 (&lt;- bioRxiv)</p>
<p>What are the genes?! (&lt;- his formatting)</p>
<ul>
<li>aaaah, he didn’t say.</li>
<li>tease</li>
</ul>
</div>
<div id="learning-the-properties-of-adaptive-regions-with-functional-data-analysis" class="section level1">
<h1>Learning the properties of adaptive regions with functional data analysis</h1>
<p><strong>Mehreen R. Mughal</strong>, Hillary Koch, Jinguo Huang, Francesca Chiaromonte, Michael DeGiorgio.</p>
<p>How do organisms adapt to their environment?</p>
<ul>
<li>pos. selection == (over time) fixation</li>
<li>idea is more beneficial == faster fixation</li>
<li>selective sweep reduces diversity at haplotype
<ul>
<li>(chunk of genome)</li>
<li>so one beneficial variant can “drag” a whole lot of other variants into purity.
<ul>
<li>free-loaders</li>
</ul></li>
</ul></li>
</ul>
<p>SURFDAWave</p>
<ul>
<li>ha</li>
<li>learn spatial dist. from statistics of sweep / neutral</li>
<li>doi.org/10.1101/834010</li>
<li>pick dist to match data</li>
<li>penalized regression
<ul>
<li>with wavelet coefficient</li>
<li>cross-validate params to check fit</li>
</ul></li>
<li>predicts neutral/sweep pretty well in TF/TP/FN/FP</li>
<li>each statistic tested gets own model
<ul>
<li>initial freq</li>
<li>selection coef</li>
<li>time of selection</li>
</ul></li>
</ul>
<p>Now running on SLC45A2</p>
<ul>
<li>related to skin color in europeans</li>
<li>estimated time of selection 2000 years ago with initial freq of 0.04</li>
</ul>
<p>Adaptive introgression</p>
<ul>
<li>selective sweep (?)</li>
<li>beneficial mutation has unique pattern</li>
<li>which SURFDAWave has 52% successful classification
<ul>
<li>kind of bleh</li>
<li>can be improved with more info?</li>
<li>with 2D info 52 -&gt; 62%</li>
</ul></li>
</ul>
</div>
<div id="creating-pan-human-and-population-specific-consensus-representations-of-the-reference-genome-and-assessing-their-effect-on-functional-genomic-data-analysis" class="section level1">
<h1>Creating pan-human and population-specific consensus representations of the reference genome and assessing their effect on functional genomic data analysis</h1>
<p><strong>Benjamin Kaminow</strong>, Sara Ballouz, Jesse Gillis, Alex Dobin.</p>
<p>How much variation do we need to improve ref genome?</p>
<p>We can use <strong>all</strong> known…but graph is complicated.</p>
<p>Consensus genome</p>
<ul>
<li>simple</li>
<li>replace minor alleles with major alleles in other pop</li>
<li>have been used on variant calling</li>
<li><strong>here we assess effects of RNA-seq</strong></li>
</ul>
<p>ConsDB download and parse db files which makes consensus VCF and FASTA</p>
<p>Uses 1000G</p>
<ul>
<li>better pop balance</li>
<li>uh, but doesn’t 1000G have loads of technical errors due to seq. errors?
<ul>
<li>was this ever fixed……?</li>
</ul></li>
<li>got RNA-seq data from one person from 4 diff pops</li>
<li>anyways….millions of replacements of SNPS</li>
<li>once replacements made, then # of diff SNPs decreases from one genome &lt;-&gt; consensus genome</li>
</ul>
<p>check mapping with STARconsensus</p>
<ul>
<li>STAR + VCF file to transform the reference</li>
<li>also does compare pre/post correction<br />
= VERY large drop in mapping error rate…(3.5 to 0.5%)
<ul>
<li>much bigger effect than I would have thought….</li>
</ul></li>
<li>wait I’m confused
<ul>
<li>y axis says “% of reads overlapping variant”</li>
<li>that’s not error rate….</li>
</ul></li>
</ul>
</div>
<div id="gramene-subsitespangenome-browsers-for-crops" class="section level1">
<h1>Gramene subsites—Pangenome browsers for crops</h1>
<p>Marcela K. Tello-Ruiz, Sharon Wei, Joshua Stein, Kapeel Chougule, <strong>Andrew Olson</strong>, Yinping Jiao, Bo Wang, Ivar Meijs, Doreen Ware.</p>
</div>
<div id="section" class="section level1">
<h1></h1>
<p><strong>Janet Kelso.</strong></p>
<p>NOT TWEETABLE</p>
</div>
<div id="what-do-we-gain-when-tolerating-loss-the-information-bottleneck-lossy-compression-and-detecting-horizontal-gene-transfer" class="section level1">
<h1>What do we gain when tolerating loss? The information bottleneck, lossy compression, and detecting horizontal gene transfer</h1>
<p><strong>Apurva Narechania</strong>, Rob Desalle, Barun Mathema, Barry Kreiswirth,
Paul Planet.</p>
<p>Can we do horizontal gene transfer without genes?</p>
<p>Make pangenome</p>
<p>count kmers across genomes</p>
<p>clusters kmers</p>
<p>but how do we cluster well? don’t want to lose too much info…but also want to compress…how to get the proper balance?</p>
<p>claude shannon!</p>
<ul>
<li>transmitter -&gt; noisy channel -&gt; receiver</li>
<li>noisy channel ~ noisy clusters</li>
<li>want middle ground where you can interpret but haven’t lost much info</li>
</ul>
<p>can see how clusters stabliize with n kmers</p>
<p>I’m reminded of luke zappias/oshlack clustree work</p>
<ul>
<li>which is ironic bc this talk is all about avoiding trees</li>
<li>done in 2 hours … which I guess is fast?</li>
</ul>
<p>Plot show inflection between cluster n and norm. mutual information</p>
<p>q: you fixed at 19 kmer, why?
a: looking at other kmer n - balance between closeness of variants and kmer n</p>
</div>
<div id="a-recurrent-neural-network-for-inferring-sweeps-and-allele-frequency-trajectories-using-gene-trees-based-on-the-ancestral-recombination-graph" class="section level1">
<h1>A recurrent neural network for inferring sweeps and allele frequency trajectories using gene trees based on the ancestral recombination graph</h1>
<p><strong>Hussein A</strong>. Hejase, Ziyi Mo, Adam Siepel.</p>
</div>
<div id="assembling-the-y-chromosomes-of-anopheles-mosquitoes" class="section level1">
<h1>Assembling the Y chromosomes of anopheles mosquitoes</h1>
<p><strong>Chujia Chen</strong>, Austin Compton, Yang Wu, Jiangtao Liang, Dustin Miller, Xiaoguang Chen, Igor Sharakhov, Chunhong Mao, Zhijian J. Tu.</p>
</div>
