---
title: "#GI2019 2019 11 06"
author: David McGaughey
date: '2019-11-06'
slug: gi2019-day01
categories:
  - bioinformatics
  - gi2019
  - conferences
  - talk
tags:
  - bioinformatics
  - gi2019
  - conferences
  - talk
header:
  caption: ''
  image: ''
---



<div id="day-1" class="section level1">
<h1>Day 1</h1>
<p>Genome Informatica 2019 at CSHL</p>
<p><em>Italics Bold</em> is the speaker</p>
<p>349 attendees. Largest?</p>
<p>189 posters</p>
<p>39 (?) talks</p>
<p>90 countries, 20 (US) states</p>
<p>High(est) % female, URM for gi</p>
</div>
<div id="night-1" class="section level1">
<h1>Night 1</h1>
<div id="comparative-3d-genome-organization-in-apicomplexan-parasites" class="section level2">
<h2>Comparative 3D genome organization in Apicomplexan parasites</h2>
<p>Evelien M. Bunnik, Aarthi Venkat, Ferhat Ay, <strong>Karine G. Le Roch.</strong></p>
<p>Apicomplexan parasites:</p>
<ul>
<li>toxoplasma gondii</li>
<li>babesia</li>
<li>most prevalent nad morbidity causing pathogens</li>
</ul>
<p>Try to understand at genomic level</p>
<p>Malaria (plasmodium) genome published 18 years ago</p>
<ul>
<li>82% AT rich</li>
<li>grab bag of major -omics techniques</li>
<li>still poor understanding of gene regulation</li>
<li>believe under-representation of TFs identified (lower rate than human, yeast)</li>
</ul>
<p>Did Hi-C (chrmatin conformation capture)</p>
<ul>
<li>at many stages</li>
<li>100kb resolution (at least for later work?)</li>
<li>going through their literature of chromosome organization (Ay, Bunnik Genome Research 2014)</li>
</ul>
<p><em>Virulence</em> Genes</p>
<ul>
<li>mediates adherence of infected RGC to endothelium of infected RBC</li>
<li>altered chromatin structure at some of these genes</li>
</ul>
<p>More examples of loops for other pathogenesis processes</p>
<ul>
<li>see power of doing many stages (stage specific patterning of contact maps)</li>
</ul>
<p>Did more Hi-C across other parasites, unfortunately had to re-assemble several genomes</p>
<ul>
<li>Bunnik PNAS 2019</li>
<li>more expression at genes near (in 3D space) telomore</li>
<li>which is where many virulence genes are …</li>
</ul>
</div>
<div id="unscrambling-the-tumor-genome-via-integrated-analysis-of-structural-variation-and-copy-number" class="section level2">
<h2>Unscrambling the tumor genome via integrated analysis of structural variation and copy number</h2>
<p><strong>Daniel L. Cameron</strong>, Charles Shale, Jonathan Baber, Anthony T. Papenfuss, Peter Priestly.</p>
<p>Hartwig has a fully automated oncology report (“no touch”)</p>
<ul>
<li>needs to be highly sensitive and specific</li>
<li>use 110x tumor, 45x germline</li>
<li>SV and CNV calling not ideal</li>
</ul>
<p>GRIDSS -&gt; PURPLE -&gt; LINX</p>
<p>SV-&gt; CNV -&gt; Interpretation</p>
<p><a href="https://github.com/PapenfussLab/gridss" class="uri">https://github.com/PapenfussLab/gridss</a></p>
<p>full pipeline: <a href="https://github.com/hartwigmedical/gridss-purple-linx" class="uri">https://github.com/hartwigmedical/gridss-purple-linx</a></p>
<p>work on phasing SV (cis vs trans vs unknown)</p>
<ul>
<li>only good up to about 300bp</li>
<li>but events are clustered, so works better than you’d expect</li>
</ul>
<p>Benchmarking (includes GRIDSS)</p>
<ul>
<li><a href="https://genomebiology.biomedcentral.com/articles/10.1186/s13059-019-1720-5" class="uri">https://genomebiology.biomedcentral.com/articles/10.1186/s13059-019-1720-5</a></li>
</ul>
</div>
<div id="tissue-specific-enhancer-functional-networks-for-associating-distal-regulatory-regions-to-disease" class="section level2">
<h2>Tissue-specific enhancer functional networks for associating distal regulatory regions to disease</h2>
<p><strong>Xi Chen</strong>, Jian Zhou, Chandra L. Theesfeld, Ran Zhang, Aaron K. Wong, Olga G. Troyanskaya.</p>
<p>How to predict function of enhancers?</p>
<p>Different ways enhancers associated with disease:</p>
<ul>
<li>enhancer -&gt; gene (near)</li>
<li>enhancer(with SNP inside)</li>
<li>enhancer(rare mutation inside)</li>
<li>enhance without direct evidence</li>
</ul>
<p>Hypothesis:</p>
<ul>
<li>enhancer will have a functional relationship with another enhancer if:
<ul>
<li>3d interaction</li>
<li>co-activation with TF</li>
<li>regulate related genes</li>
</ul></li>
<li>if all met, then <em>maybe</em> disease associated</li>
</ul>
<p>bayesian inference to model the above assumptions</p>
<p>how to see if this sort of works?</p>
<ul>
<li>check brain specific super enhancers in non-brain tissues</li>
<li>get some enrichment</li>
</ul>
<p><a href="https://fenrir.flatironinstitute.org" class="uri">https://fenrir.flatironinstitute.org</a></p>
<ul>
<li>web tool</li>
</ul>
</div>
<div id="targeted-nanopore-sequencing-with-cas9-for-studies-of-methylation-structural-variants-and-mutations" class="section level2">
<h2>Targeted Nanopore sequencing with Cas9 for studies of methylation, structural variants, and mutations</h2>
<p><em>Timothy E. Gilpatrick</em>, Isac Lee, Fritz J. Sedlazek, Winston Timp.</p>
</div>
<div id="long-read-sequencing-of-structurally-variant-genomes" class="section level2">
<h2>Long-read sequencing of structurally variant genomes</h2>
<p><em>Evan E. Eichler.</em></p>
</div>
<div id="exploring-the-3d-spatial-dependency-of-gene-expression-using-markov-random-fields" class="section level2">
<h2>Exploring the 3D spatial dependency of gene expression using Markov random fields</h2>
<p><em>Naihui Zhou</em>, Iddo Friedberg, Mark S. Kaiser.</p>
</div>
<div id="mapping-cis-eqtl-from-rna-seq-data-with-no-genotypes" class="section level2">
<h2>Mapping cis-eQTL from RNA-seq data with no genotypes</h2>
<p><em>Elena Vigorito</em>, Simon White, Chris Wallace.</p>
</div>
<div id="exploring-short-tandem-repeat-expansions-at-both-known-and-novel-loci-in-the-human-genome" class="section level2">
<h2>Exploring short tandem repeat expansions at both known and novel loci in the human genome</h2>
<p><em>Harriet Dashnow</em>, Brent Pedersen, Daniel MacArthur, Alicia Oshlack, Aaron Quinlan.</p>
</div>
</div>
