---
title: "#GI2019 2019 11 07"
author: David McGaughey
date: '2019-11-07'
slug: gi2019-day02
categories:
  - bioinformatics
  - gi2019
  - conferences
  - talk
tags:
  - bioinformatics
  - gi2019
  - conferences
  - talk
header:
  caption: ''
  image: ''
output:
  blogdown::html_page:
    toc: true
---

# Day 2 - Session 2 - SEQUENCING ALGORITHMS, VARIANT DISCOVERY AND GENOME ASSEMBLY

Genome Informatics 2019 at CSHL

**Bold** is the speaker

# Genomic sketching with HyperLogLog
Daniel N. Baker, **Ben Langmead.**

Slides at...crap missed it. Email Ben I guess?

Sketching

  - algorithm to collapse genomes into summary info
  - fasta -> kmers ("shingle") -> sample (smartly)
  - showing set relationships
  - the "minimum" of a set of sketches gives you useful info on the range you would expect

Unions and Intersections

  - space of possible coincidences requires multiple samples to get useful info
  - bottom 3? 
  - or minimum in 3 partitions?
  
MinHash uses bottom-k approach (Ondov / Phillipy)

Tremendous talk - I feel like I understand what is happening. Invite Ben to talk at your place...I'd imagine you'd get a great talk. 

OK, so take *log log minimum*

Daniel Baker wrote: 

  - bit.ly/dash_pre
  - github.com/dnbaker/dashing
  - github.com/dnbaker/sketch
  
HyperLogLog *HLL)

  - k partition
  - log<sub>2</sub>n
  - exponent
  - average, bias correction
  
Highly vectorized (for speed! Massively parallel calculation)

  - only on intel? 
  
Comparing to MinHash

  - HLL deals with lopsided sets better (bottom k not ideal, better minimum k partition)
  - Uses WAY less memory and wall clock
  
Future stuffs

  - multi-k (kmer lengths)
  - weighted jaccard

# centroFlye—Assembling centromeres with long error-prone reads 
**Andrey Bzikadze**, Pavel A. Pevzner.

@AndreyBzikadze

biorxiv preprint (missed link)

brings up t2t consortium (telomere to telomere)

github.com/nanopore-wgs-consortium/CHM13

mentions karen miga's chrX t2t biorxiv preprint

centromere:

  - assembler's favoriate region
    - probably not, but keeps them gainfully employed for now
  - highly repetitive....etc etc.
  - 3% of genome - and because are unassembled, poor idea how they influence human diseasse
  
centroFlye

  - classify reads (prefix...something...suffix... what is going on? Andrey already moved on)
    - guessing something to do with begining / middle / end of repeat?
  - find rare k-mers that can be used to anchor assembly
  - but rare k-mers are usually **errors** 
    - if the **distance between two rare k-mers is conserved** then they are likely real
    - neat idea!
    - but I'm not certain what conservation is...
  - throw everything away except for rare k-mers

centroFlye used to improve chrX centromere assembly
  
  - greater length
  - but how do you more robustly define success?
  - one metric is to count shared (or discrepant) k-mers between assembly and a read
  - discorance (a,b) = sharedRead(assembly A) - sharedRead(assembly B)


# Genotyping structural variants in pangenome graphs using the vg toolkit
**Jean Monlong**, Glenn Hickey, David Heller, Jonas Andreas Sibbesen, Jouni Siren, Jordan Eizenga, Eric T. Dawson, Erik Garrison, Adam Novak, Benedict Paten.



# Rapidly mapping raw nanopore signal with UNCALLED to enable real-time targeted sequencing
**Sam Kovaka**, Yunfan Fan, Winston Timp, Michael C. Schatz.

# The construct and utility of reference pan-genome graphs
**Heng Li.**

# PRINCESS—A framework for comprehensive detection andphasing of SNPs and structural variants
**Medhat Mahmoud**, Winston Timp, Fritz J. Sedlazeck.  

# Efficient chromosome-scale haplotype-resolved assembly of human individuals
**Shilpa Garg**, Arkarachai Fungtammasan, Anthony Schmitt, Andrew
Carroll, Paul Pelusol, Emily Hatas, Fritz Sedlazeck, Justin Zook, Mike
Chou, John Aach, Jason Chin, Heng Li, George Church.

# Utilization of an ensemble approach for identification of driver fusions in pediatric cancer
**Stephanine LaHaye**, Kyle Voytovich, James Fitch, Natalie Bir, Sean D. McGrath, Anthony Miller, Amy Wetzel, Vincent Magrini, Catherine E. Cottrell, Elaine R. Mardis, Richard K. Wilson.