<!DOCTYPE html>
<html lang="en-us">
<head>

  <meta charset="utf-8">
  <meta name="viewport" content="width=device-width, initial-scale=1">
  <meta http-equiv="X-UA-Compatible" content="IE=edge">
  <meta name="theme" content="hugo-academic">
  <meta name="generator" content="Hugo 0.40" />
  <meta name="author" content="David McGaughey">

  
  
  
  
    
      
    
  
  <meta name="description" content="Day 4 - Session 9 - PERSONAL AND MEDICAL GENOMICS Characterization of prevalence and health consequences of uniparental disomy in four million individuals from the general population Sub-continental ancestry inference based on the gnomAD dataset accurately classifies patients at NCH Patient stratification in the UK’s 100k Genomes Project—Using WGS and machine learning to predict cancer outcomes Inferring clone- and haplotype-specific chromosomal organization in rearranged cancer genomes with multiple sequencing technologies Identification and interpretation of common and rare variants in relation to rare disease phenotype and outcome Somatic mutation status prediction by a splicing-alteration-based machine learning technique Beyond accessibility—ATAC-seq footprinting analysis reveals dynamics of transcription factor binding during preimplantation development   Day 4 - Session 9 - PERSONAL AND MEDICAL GENOMICS  Characterization of prevalence and health consequences of uniparental disomy in four million individuals from the general population Priyanka Nakka, Samuel Pattillo Smith, Anne H.">

  
  <link rel="alternate" hreflang="en-us" href="/./post/gi2019-day04-morning/">

  


  

  
  
  <meta name="theme-color" content="#0095eb">
  
  
  
  
    
  
  
    
    
      
        <link rel="stylesheet" href="//cdnjs.cloudflare.com/ajax/libs/highlight.js/9.12.0/styles/github.min.css">
      
    
  
  
  <link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.3.7/css/bootstrap.min.css" integrity="sha512-6MXa8B6uaO18Hid6blRMetEIoPqHf7Ux1tnyIQdpt9qI5OACx7C+O3IVTr98vwGnlcg0LOLa02i9Y1HpVhlfiw==" crossorigin="anonymous">
  <link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/academicons/1.8.6/css/academicons.min.css" integrity="sha256-uFVgMKfistnJAfoCUQigIl+JfUaP47GrRKjf6CTPVmw=" crossorigin="anonymous">
  <link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/4.7.0/css/font-awesome.min.css" integrity="sha512-SfTiTlX6kk+qitfevl/7LibUOeJWlt9rbyDn92a1DqWOw9vWG2MFoays0sgObmWazO5BQPiFucnnEAjpAB+/Sw==" crossorigin="anonymous">
  <link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/fancybox/3.2.5/jquery.fancybox.min.css" integrity="sha256-ygkqlh3CYSUri3LhQxzdcm0n1EQvH2Y+U5S2idbLtxs=" crossorigin="anonymous">
  
  
  
  
  <link rel="stylesheet" href="//fonts.googleapis.com/css?family=Montserrat:400,700%7cRoboto:400,400italic,700%7cRoboto&#43;Mono">
  
  <link rel="stylesheet" href="/styles.css">
  

  
    <script>
      window.ga=window.ga||function(){(ga.q=ga.q||[]).push(arguments)};ga.l=+new Date;
      ga('create', 'UA-113461893-1', 'auto');
      
      ga('require', 'eventTracker');
      ga('require', 'outboundLinkTracker');
      ga('require', 'urlChangeTracker');
      ga('send', 'pageview');
    </script>
    <script async src="//www.google-analytics.com/analytics.js"></script>
    
    <script async src="https://cdnjs.cloudflare.com/ajax/libs/autotrack/2.4.1/autotrack.js" integrity="sha512-HUmooslVKj4m6OBu0OgzjXXr+QuFYy/k7eLI5jdeEy/F4RSgMn6XRWRGkFi5IFaFgy7uFTkegp3Z0XnJf3Jq+g==" crossorigin="anonymous"></script>
    
  

  
  <link rel="alternate" href="/./index.xml" type="application/rss+xml" title="eye Bioinformatician">
  <link rel="feed" href="/./index.xml" type="application/rss+xml" title="eye Bioinformatician">
  

  <link rel="manifest" href="/site.webmanifest">
  <link rel="icon" type="image/png" href="/img/icon.png">
  <link rel="apple-touch-icon" type="image/png" href="/img/icon-192.png">

  <link rel="canonical" href="/./post/gi2019-day04-morning/">

  <meta property="twitter:card" content="summary_large_image">
  
  <meta property="twitter:site" content="@https://twitter.com/David_McGaughey">
  <meta property="twitter:creator" content="@https://twitter.com/David_McGaughey">
  
  <meta property="og:site_name" content="eye Bioinformatician">
  <meta property="og:url" content="/./post/gi2019-day04-morning/">
  <meta property="og:title" content="2019 11 09 GI2019 Personal and Medical Genomics | eye Bioinformatician">
  <meta property="og:description" content="Day 4 - Session 9 - PERSONAL AND MEDICAL GENOMICS Characterization of prevalence and health consequences of uniparental disomy in four million individuals from the general population Sub-continental ancestry inference based on the gnomAD dataset accurately classifies patients at NCH Patient stratification in the UK’s 100k Genomes Project—Using WGS and machine learning to predict cancer outcomes Inferring clone- and haplotype-specific chromosomal organization in rearranged cancer genomes with multiple sequencing technologies Identification and interpretation of common and rare variants in relation to rare disease phenotype and outcome Somatic mutation status prediction by a splicing-alteration-based machine learning technique Beyond accessibility—ATAC-seq footprinting analysis reveals dynamics of transcription factor binding during preimplantation development   Day 4 - Session 9 - PERSONAL AND MEDICAL GENOMICS  Characterization of prevalence and health consequences of uniparental disomy in four million individuals from the general population Priyanka Nakka, Samuel Pattillo Smith, Anne H.">
  <meta property="og:locale" content="en-us">
  
  <meta property="article:published_time" content="2019-11-09T00:00:00&#43;00:00">
  
  <meta property="article:modified_time" content="2019-11-09T00:00:00&#43;00:00">
  

  
  

  <title>2019 11 09 GI2019 Personal and Medical Genomics | eye Bioinformatician</title>

</head>
<body id="top" data-spy="scroll" data-target="#toc" data-offset="71" >

<nav class="navbar navbar-default navbar-fixed-top" id="navbar-main">
  <div class="container">

    
    <div class="navbar-header">
      
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse"
              data-target=".navbar-collapse" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      
      <a class="navbar-brand" href="/">eye Bioinformatician</a>
    </div>

    
    <div class="collapse navbar-collapse">

      
      
      <ul class="nav navbar-nav navbar-right">
        

        
        
        
        
        
          
        

        <li class="nav-item">
          <a href="/#about">
            
            <span>Home</span>
            
          </a>
        </li>

        
        

        
        <li class="dropdown">
          <a href="#" class="dropdown-toggle" data-toggle="dropdown" aria-haspopup="true">
            
            <span>Posts</span>
            
            <span class="caret"></span>
          </a>
          <ul class="dropdown-menu">
            
            <li class="nav-item">
              <a href="/#posts">
                
                <span>Recent Posts</span>
                
              </a>
            </li>
            
            <li class="nav-item">
              <a href="/post">
                
                <span>All Posts</span>
                
              </a>
            </li>
            
          </ul>
        </li>

        
        

        
        
        
        
        
          
        

        <li class="nav-item">
          <a href="/#projects">
            
            <span>Projects</span>
            
          </a>
        </li>

        
        

        
        
        
        
        
          
        

        <li class="nav-item">
          <a href="/#culture">
            
            <span>Culture</span>
            
          </a>
        </li>

        
        

        
        
        
        
        
          
        

        <li class="nav-item">
          <a href="/#publications_selected">
            
            <span>Publications</span>
            
          </a>
        </li>

        
        
      

      
      </ul>

    </div>
  </div>
</nav>


<article class="article" itemscope itemtype="http://schema.org/Article">

  


  <div class="article-container">
    <h1 itemprop="name">2019 11 09 GI2019 Personal and Medical Genomics</h1>

    

<div class="article-metadata">

  <span class="article-date">
    
    <time datetime="2019-11-09 00:00:00 &#43;0000 UTC" itemprop="datePublished dateModified">
      Nov 9, 2019
    </time>
  </span>
  <span itemscope itemprop="author publisher" itemtype="http://schema.org/Person">
    <meta itemprop="name" content="David McGaughey">
  </span>

  
  <span class="middot-divider"></span>
  <span class="article-reading-time">
    5 min read
  </span>
  

  
  
  <span class="middot-divider"></span>
  <a href="/./post/gi2019-day04-morning/#disqus_thread"></a>
  

  
  
  
  <span class="middot-divider"></span>
  <span class="article-categories">
    <i class="fa fa-folder"></i>
    
    <a href="/./categories/bioinformatics/">bioinformatics</a
    >, 
    
    <a href="/./categories/gi2019/">gi2019</a
    >, 
    
    <a href="/./categories/conferences/">conferences</a
    >, 
    
    <a href="/./categories/talk/">talk</a
    >
    
  </span>
  
  

  
  
<div class="share-box" aria-hidden="true">
  <ul class="share">
    <li>
      <a class="twitter"
         href="https://twitter.com/intent/tweet?text=2019%2011%2009%20GI2019%20Personal%20and%20Medical%20Genomics&amp;url=%2f.%2fpost%2fgi2019-day04-morning%2f"
         target="_blank" rel="noopener">
        <i class="fa fa-twitter"></i>
      </a>
    </li>
    <li>
      <a class="facebook"
         href="https://www.facebook.com/sharer.php?u=%2f.%2fpost%2fgi2019-day04-morning%2f"
         target="_blank" rel="noopener">
        <i class="fa fa-facebook"></i>
      </a>
    </li>
    <li>
      <a class="linkedin"
         href="https://www.linkedin.com/shareArticle?mini=true&amp;url=%2f.%2fpost%2fgi2019-day04-morning%2f&amp;title=2019%2011%2009%20GI2019%20Personal%20and%20Medical%20Genomics"
         target="_blank" rel="noopener">
        <i class="fa fa-linkedin"></i>
      </a>
    </li>
    <li>
      <a class="weibo"
         href="http://service.weibo.com/share/share.php?url=%2f.%2fpost%2fgi2019-day04-morning%2f&amp;title=2019%2011%2009%20GI2019%20Personal%20and%20Medical%20Genomics"
         target="_blank" rel="noopener">
        <i class="fa fa-weibo"></i>
      </a>
    </li>
    <li>
      <a class="email"
         href="mailto:?subject=2019%2011%2009%20GI2019%20Personal%20and%20Medical%20Genomics&amp;body=%2f.%2fpost%2fgi2019-day04-morning%2f">
        <i class="fa fa-envelope"></i>
      </a>
    </li>
  </ul>
</div>


  

</div>


    <div class="article-style" itemprop="articleBody">
      <div id="TOC">
<ul>
<li><a href="#day-4---session-9---personal-and-medical-genomics">Day 4 - Session 9 - PERSONAL AND MEDICAL GENOMICS</a></li>
<li><a href="#characterization-of-prevalence-and-health-consequences-of-uniparental-disomy-in-four-million-individuals-from-the-general-population">Characterization of prevalence and health consequences of uniparental disomy in four million individuals from the general population</a></li>
<li><a href="#sub-continental-ancestry-inference-based-on-the-gnomad-dataset-accurately-classifies-patients-at-nch">Sub-continental ancestry inference based on the gnomAD dataset accurately classifies patients at NCH</a></li>
<li><a href="#patient-stratification-in-the-uks-100k-genomes-projectusing-wgs-and-machine-learning-to-predict-cancer-outcomes">Patient stratification in the UK’s 100k Genomes Project—Using WGS and machine learning to predict cancer outcomes</a></li>
<li><a href="#inferring-clone--and-haplotype-specific-chromosomal-organization-in-rearranged-cancer-genomes-with-multiple-sequencing-technologies">Inferring clone- and haplotype-specific chromosomal organization in rearranged cancer genomes with multiple sequencing technologies</a></li>
<li><a href="#identification-and-interpretation-of-common-and-rare-variants-in-relation-to-rare-disease-phenotype-and-outcome">Identification and interpretation of common and rare variants in relation to rare disease phenotype and outcome</a></li>
<li><a href="#somatic-mutation-status-prediction-by-a-splicing-alteration-based-machine-learning-technique">Somatic mutation status prediction by a splicing-alteration-based machine learning technique</a></li>
<li><a href="#beyond-accessibilityatac-seq-footprinting-analysis-reveals-dynamics-of-transcription-factor-binding-during-preimplantation-development">Beyond accessibility—ATAC-seq footprinting analysis reveals dynamics of transcription factor binding during preimplantation development</a></li>
</ul>
</div>

<div id="day-4---session-9---personal-and-medical-genomics" class="section level1">
<h1>Day 4 - Session 9 - PERSONAL AND MEDICAL GENOMICS</h1>
</div>
<div id="characterization-of-prevalence-and-health-consequences-of-uniparental-disomy-in-four-million-individuals-from-the-general-population" class="section level1">
<h1>Characterization of prevalence and health consequences of uniparental disomy in four million individuals from the general population</h1>
<p>Priyanka Nakka, Samuel Pattillo Smith, Anne H. O’Donnell-Luria, Kimberly F. McManus, 23andMe Research Team, Joanna L. Mountain, <strong>Sohini Ramachandran</strong>, Fah Sathirapongsasuti.</p>
<p><span class="citation">@s_ramach</span></p>
<p>Paper came out two days ago AJHG</p>
<p>“Population-level summaries of ROH increase with divergnce from Agrica”
ROH == Runs of Homozygosity (LONG runs where both chr are identical)</p>
<p>“Class C ROH”</p>
<ul>
<li>don’t know what this is…something about consanguinity?</li>
<li>oh, &gt;0.8MB?</li>
</ul>
<p>Group interested in distribution across human genomes</p>
<p>23andMe partially sort of funded (personel) this study</p>
<p>Also uses 23andMe data</p>
<ul>
<li>obv</li>
<li>not even MacArthur has 4 million genomes</li>
</ul>
<p>Posting IBD karyogram….from reddit (imgur.com/a/hl6Cd)</p>
<p>UPD</p>
<ul>
<li>both homologs of chr come from one parent</li>
<li>meiotic oopsie</li>
</ul>
<p>3,300 reported in clinical literature</p>
<ul>
<li>upd-tl.com/upd.html</li>
<li>more common on chr 6,7,11, and 15 (most common)</li>
</ul>
<p>What is incidence in population (at least 23andMe pop)?</p>
<p>What are rates of maternal/paternal UPD?</p>
<p>Three subtypes of UPD</p>
<ul>
<li>HetUPD, IsoUPD, Partial IsoUPD</li>
</ul>
<p>In 916,712 duos/trios in 23andMe found 199 with UPD</p>
<ul>
<li>Maternal UPD 3X more common than Paternal UPD</li>
<li>different dist across chr than the upd-tl.com data</li>
</ul>
<p>Estimate 1/2000 rate (more common than previously estimated)</p>
<p>Trained log. regression classifier for each chr for each pop to diagnose ROH from UPD (instead of consanguinity)</p>
<ul>
<li>used ROH length</li>
<li>ratio of ROH across chr</li>
<li>other things I missed</li>
</ul>
<p>Found 304 <strong>putative</strong> UPD cases in 1.3 million “singletons”</p>
<ul>
<li>recapitulates chr dist. patterns</li>
<li>found 172 more in UK biobank</li>
<li>found 0 in the 4,000 duos (not a big sample size)</li>
</ul>
</div>
<div id="sub-continental-ancestry-inference-based-on-the-gnomad-dataset-accurately-classifies-patients-at-nch" class="section level1">
<h1>Sub-continental ancestry inference based on the gnomAD dataset accurately classifies patients at NCH</h1>
<p><strong>Andrei Rajkovic</strong>, Defne Ceyhan, Greg Wheeler, Tara Lichtenberg, Ben Kelly, Peter White.</p>
<p>Trying to better infer ancestry with the gnomad projection</p>
<p>ML algorithm (gradient boosted decision tree)</p>
<ul>
<li>trained on 4000 people (vcf input)</li>
<li>accuracy defined as true ancestry in top2 (is that too loose?)</li>
</ul>
<p>90+ ish accuracy (some ancestries don’t do so well)</p>
<p>I guess they are relying on self-reported ancestries? Aren’t these notorious for being not so accurate….?</p>
<ul>
<li>I asked … and yes they are …. which isn’t a problem that can be dealt with so easily</li>
</ul>
</div>
<div id="patient-stratification-in-the-uks-100k-genomes-projectusing-wgs-and-machine-learning-to-predict-cancer-outcomes" class="section level1">
<h1>Patient stratification in the UK’s 100k Genomes Project—Using WGS and machine learning to predict cancer outcomes</h1>
<p><strong>Kate Ridout</strong>, Pauline Robbe, CLL pilot consortium, Anna Schuh.
Presenter affiliation: University of Oxford, Oxford, United Kingdom. 231</p>
<p>Kaplan Meier curves</p>
<ul>
<li>these plots always bum me out</li>
<li>death curve</li>
</ul>
<p>Going through pipeline to call variants / SV / CNV in somatic tissue</p>
<ul>
<li>LOTS of different tools</li>
<li>even chromHMM … not certain what datasets the chromHMM was built off of</li>
<li>anyways just super comprehensive, kitchen sink kind of stuff
<ul>
<li>so many tools I wonder if they are correcting their p values for the bajillion test/tools they are doing….</li>
</ul></li>
</ul>
<p>showing some case studies….but why not combining all these into one framework….oh here we go I think</p>
<ul>
<li>LASSO</li>
<li>non-negative matrix factorization (NMF)
<ul>
<li>combine all features and see patterns, assess against outcomes</li>
<li>n isn’t super high (200 ish)</li>
</ul></li>
<li>several independent-ish models / data / thingies which get merged somehow?
<ul>
<li>don’t quite understand architecture…</li>
</ul></li>
<li>found a mut sig associated with better outcome (useful to de-prioritze for damaging chemo)</li>
</ul>
</div>
<div id="inferring-clone--and-haplotype-specific-chromosomal-organization-in-rearranged-cancer-genomes-with-multiple-sequencing-technologies" class="section level1">
<h1>Inferring clone- and haplotype-specific chromosomal organization in rearranged cancer genomes with multiple sequencing technologies</h1>
<p><strong>Sergey Aganezov</strong>, Fritz Sedlazeck, Sara Goodwin, Gayatri Arun, Isac Lee, Sam Kovaka, Michael Kitsche, Rachel Scherman, Ilya Zban, Vitaly Aksenov, Nikita Alexeev, Robert Wappel, Melissa Kramer, Karen Kostroff, David L. Spector, Winston Timp, Michael S. Schatz, W. Richard McCombie.</p>
<p>Recover linear genome org from cancer genomes</p>
<p>Referring to short-read NGS as 2nd gen and long reads as 3rd gen</p>
<ul>
<li>will this catch on? Probably not…especially if I misunderstood what happened</li>
</ul>
<p>10x Linked Reads + ONT + PacBio</p>
<ul>
<li>check concordance</li>
<li>overlap not great between all three</li>
<li>A TON were found in both ONT + PacBio
<ul>
<li>LOTS of FP for 10x…not a good slide for this tech</li>
<li>so maybe don’t use 10X LR for SV</li>
</ul></li>
</ul>
<p>Overall finding so many more SV with long reads</p>
<p>Get to 90% precision/recall at 25X</p>
<p>Tried to genotype SVs in 1000G - found a few (germline of course). Artifacts? Interesting? Dunno…</p>
<p>Discussing haplotype resolved chr structure technique</p>
<ul>
<li>github.com/izban/contig-covering</li>
</ul>
</div>
<div id="identification-and-interpretation-of-common-and-rare-variants-in-relation-to-rare-disease-phenotype-and-outcome" class="section level1">
<h1>Identification and interpretation of common and rare variants in relation to rare disease phenotype and outcome</h1>
<p>Mana Gajapathy, Brandon Wilk, Arthur Weborg, Angelina Uno- Antonison, Alex Moss, Matt Holt, Donna Brown, Melissa Wilk, Camille Birch, Nadiya Sosonkina, <strong>Elizabeth Worthey</strong></p>
<p>History lesson on how in 2003, “success rate” for rare genetic disease had seq. diagnosis of about 5%. Today is approaching 50%.</p>
<p>But common genetic disease is ~5% <em>today</em></p>
<p>Sanger -&gt; Exome -&gt; WGS</p>
<p>Software / pipelines are curcial in making seq. <strong>useful</strong></p>
<p>Using many orthogonal datasets and phenotype info (HPO?) to improve filtering</p>
<p>This software proprietary / custom for UAB?</p>
<p>varsight:</p>
<ul>
<li>biorxiv.org/content/10.1101/532440v2</li>
<li><a href="https://bmcbioinformatics.biomedcentral.com/articles/10.1186/s12859-019-3026-8" class="uri">https://bmcbioinformatics.biomedcentral.com/articles/10.1186/s12859-019-3026-8</a></li>
</ul>
<p>Patient with paterna isodomic UPD….3!!! different diseases with multiple causual genes! Yikes.</p>
</div>
<div id="somatic-mutation-status-prediction-by-a-splicing-alteration-based-machine-learning-technique" class="section level1">
<h1>Somatic mutation status prediction by a splicing-alteration-based machine learning technique</h1>
<p>Raul N. Mateos, Naoko Iida, Kenichi Chiba, Ai Okada, Yuichi Shiraishi.</p>
<p>NOT HERE</p>
<p>Model selection and permutation testing for association studies within a large direct-to-consumer genetic cohort
<strong>Elena P. Sorokin</strong>, Danny S. Park, Kristin A. Rand, Julie M. Granka, Eurie L. Hong, Kenneth G. Chahine, Catherine A. Ball.</p>
<p>NOT TWEETABLE</p>
</div>
<div id="beyond-accessibilityatac-seq-footprinting-analysis-reveals-dynamics-of-transcription-factor-binding-during-preimplantation-development" class="section level1">
<h1>Beyond accessibility—ATAC-seq footprinting analysis reveals dynamics of transcription factor binding during preimplantation development</h1>
<p>Mette Bentsen, Philipp Goymann, Anastasiia Petrova, Hendrik Schultheis, Kathrin Klee, Jens Preussner, Carsten Kuenne, Mario Looso.</p>
<p>NOT HERE</p>
</div>

    </div>

    


<div class="article-tags">
  
  <a class="btn btn-primary btn-outline" href="/./tags/bioinformatics/">bioinformatics</a>
  
  <a class="btn btn-primary btn-outline" href="/./tags/gi2019/">gi2019</a>
  
  <a class="btn btn-primary btn-outline" href="/./tags/conferences/">conferences</a>
  
  <a class="btn btn-primary btn-outline" href="/./tags/talk/">talk</a>
  
</div>




    
    
    <div class="article-widget">
      <div class="hr-light"></div>
      <h3>Related</h3>
      <ul>
        
        <li><a href="/post/gi2019-day03-morning/">#GI2019 2019 11 08 Morning Session</a></li>
        
        <li><a href="/post/gi2019-day03-afternoon/">#GI2019 2019 11 08 Afternoon Session</a></li>
        
        <li><a href="/post/gi2019-day02/">#GI2019 2019 11 07</a></li>
        
        <li><a href="/post/gi2019-day02-night/">#GI2019 2019 11 07 Night Session</a></li>
        
        <li><a href="/post/gi2019-day01/">#GI2019 2019 11 06</a></li>
        
      </ul>
    </div>
    

    

    
<section id="comments">
  <div id="disqus_thread"></div>
<script>
    var disqus_config = function () {
    
    
    
    };
    (function() {
        if (["localhost", "127.0.0.1"].indexOf(window.location.hostname) != -1) {
            document.getElementById('disqus_thread').innerHTML = 'Disqus comments not available by default when the website is previewed locally.';
            return;
        }
        var d = document, s = d.createElement('script'); s.async = true;
        s.src = '//' + "davemcg-gitub-io" + '.disqus.com/embed.js';
        s.setAttribute('data-timestamp', +new Date());
        (d.head || d.body).appendChild(s);
    })();
</script>
<noscript>Please enable JavaScript to view the <a href="https://disqus.com/?ref_noscript">comments powered by Disqus.</a></noscript>
<a href="https://disqus.com" class="dsq-brlink">comments powered by <span class="logo-disqus">Disqus</span></a>
</section>



  </div>
</article>

<footer class="site-footer">
  <div class="container">
    <p class="powered-by">

      &copy; 2018 &middot; 

      Powered by the
      <a href="https://sourcethemes.com/academic/" target="_blank" rel="noopener">Academic theme</a> for
      <a href="https://gohugo.io" target="_blank" rel="noopener">Hugo</a>.

      <span class="pull-right" aria-hidden="true">
        <a href="#" id="back_to_top">
          <span class="button_icon">
            <i class="fa fa-chevron-up fa-2x"></i>
          </span>
        </a>
      </span>

    </p>
  </div>
</footer>


<div id="modal" class="modal fade" role="dialog">
  <div class="modal-dialog">
    <div class="modal-content">
      <div class="modal-header">
        <button type="button" class="close btn-large" data-dismiss="modal">&times;</button>
        <h4 class="modal-title">Cite</h4>
      </div>
      <div>
        <pre><code class="modal-body tex"></code></pre>
      </div>
      <div class="modal-footer">
        <a class="btn btn-primary btn-outline js-copy-cite" href="#" target="_blank">
          <i class="fa fa-copy"></i> Copy
        </a>
        <a class="btn btn-primary btn-outline js-download-cite" href="#" target="_blank">
          <i class="fa fa-download"></i> Download
        </a>
        <div id="modal-error"></div>
      </div>
    </div>
  </div>
</div>

    

    
    
    <script id="dsq-count-scr" src="//davemcg-gitub-io.disqus.com/count.js" async></script>
    

    

    
    <script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.2.1/jquery.min.js" integrity="sha512-3P8rXCuGJdNZOnUx/03c1jOTnMn3rP63nBip5gOP2qmUh5YAdVAvFZ1E+QLZZbC1rtMrQb+mah3AfYW11RUrWA==" crossorigin="anonymous"></script>
    <script src="https://cdnjs.cloudflare.com/ajax/libs/jquery.imagesloaded/4.1.3/imagesloaded.pkgd.min.js" integrity="sha512-umsR78NN0D23AzgoZ11K7raBD+R6hqKojyBZs1w8WvYlsI+QuKRGBx3LFCwhatzBunCjDuJpDHwxD13sLMbpRA==" crossorigin="anonymous"></script>
    <script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.3.7/js/bootstrap.min.js" integrity="sha512-iztkobsvnjKfAtTNdHkGVjAYTrrtlC7mGp/54c40wowO7LhURYl3gVzzcEqGl/qKXQltJ2HwMrdLcNUdo+N/RQ==" crossorigin="anonymous"></script>
    <script src="https://cdnjs.cloudflare.com/ajax/libs/jquery.isotope/3.0.4/isotope.pkgd.min.js" integrity="sha512-VDBOIlDbuC4VWxGJNmuFRQ0Li0SKkDpmGyuhAG5LTDLd/dJ/S0WMVxriR2Y+CyPL5gzjpN4f/6iqWVBJlht0tQ==" crossorigin="anonymous"></script>
    <script src="https://cdnjs.cloudflare.com/ajax/libs/fancybox/3.2.5/jquery.fancybox.min.js" integrity="sha256-X5PoE3KU5l+JcX+w09p/wHl9AzK333C4hJ2I9S5mD4M=" crossorigin="anonymous"></script>
    
    
    <script src="/js/hugo-academic.js"></script>
    

    
    
      
      
      <script src="//cdnjs.cloudflare.com/ajax/libs/highlight.js/9.12.0/highlight.min.js" integrity="sha256-/BfiIkHlHoVihZdc6TFuj7MmJ0TWcWsMXkeDFwhi0zw=" crossorigin="anonymous"></script>
      

      

      

      <script>hljs.initHighlightingOnLoad();</script>
    

    
    
    <script type="text/x-mathjax-config">
        MathJax.Hub.Config({ tex2jax: { inlineMath: [['$','$'], ['\\(','\\)']] } });
    </script>
    
    <script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.1/MathJax.js?config=TeX-AMS_CHTML" integrity="sha512-tOav5w1OjvsSJzePRtt2uQPFwBoHt1VZcUq8l8nm5284LEKE9FSJBQryzMBzHxY5P0zRdNqEcpLIRVYFNgu1jw==" crossorigin="anonymous"></script>
    
    

  </body>
</html>

