<!DOCTYPE html>
<html lang="en-us">
<head>

  <meta charset="utf-8">
  <meta name="viewport" content="width=device-width, initial-scale=1">
  <meta http-equiv="X-UA-Compatible" content="IE=edge">
  <meta name="theme" content="hugo-academic">
  <meta name="generator" content="Hugo 0.40" />
  <meta name="author" content="David McGaughey">

  
  
  
  
    
      
    
  
  <meta name="description" content="Day 2 - Session 2 - SEQUENCING ALGORITHMS, VARIANT DISCOVERY AND GENOME ASSEMBLY Genomic sketching with HyperLogLog centroFlye—Assembling centromeres with long error-prone reads Genotyping structural variants in pangenome graphs using the vg toolkit Rapidly mapping raw nanopore signal with UNCALLED to enable real-time targeted sequencing The construct and utility of reference pan-genome graphs PRINCESS — A framework for comprehensive detection and phasing of SNPs and structural variants Efficient chromosome-scale haplotype-resolved assembly of human individuals Utilization of an ensemble approach for identification of driver fusions in pediatric cancer   Day 2 - Session 2 - SEQUENCING ALGORITHMS, VARIANT DISCOVERY AND GENOME ASSEMBLY Genome Informatics 2019 at CSHL">

  
  <link rel="alternate" hreflang="en-us" href="/./post/gi2019-day02/">

  


  

  
  
  <meta name="theme-color" content="#0095eb">
  
  
  
  
    
  
  
    
    
      
        <link rel="stylesheet" href="//cdnjs.cloudflare.com/ajax/libs/highlight.js/9.12.0/styles/github.min.css">
      
    
  
  
  <link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.3.7/css/bootstrap.min.css" integrity="sha512-6MXa8B6uaO18Hid6blRMetEIoPqHf7Ux1tnyIQdpt9qI5OACx7C+O3IVTr98vwGnlcg0LOLa02i9Y1HpVhlfiw==" crossorigin="anonymous">
  <link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/academicons/1.8.6/css/academicons.min.css" integrity="sha256-uFVgMKfistnJAfoCUQigIl+JfUaP47GrRKjf6CTPVmw=" crossorigin="anonymous">
  <link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/4.7.0/css/font-awesome.min.css" integrity="sha512-SfTiTlX6kk+qitfevl/7LibUOeJWlt9rbyDn92a1DqWOw9vWG2MFoays0sgObmWazO5BQPiFucnnEAjpAB+/Sw==" crossorigin="anonymous">
  <link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/fancybox/3.2.5/jquery.fancybox.min.css" integrity="sha256-ygkqlh3CYSUri3LhQxzdcm0n1EQvH2Y+U5S2idbLtxs=" crossorigin="anonymous">
  
  
  
  
  <link rel="stylesheet" href="//fonts.googleapis.com/css?family=Montserrat:400,700%7cRoboto:400,400italic,700%7cRoboto&#43;Mono">
  
  <link rel="stylesheet" href="/styles.css">
  

  
    <script>
      window.ga=window.ga||function(){(ga.q=ga.q||[]).push(arguments)};ga.l=+new Date;
      ga('create', 'UA-113461893-1', 'auto');
      
      ga('require', 'eventTracker');
      ga('require', 'outboundLinkTracker');
      ga('require', 'urlChangeTracker');
      ga('send', 'pageview');
    </script>
    <script async src="//www.google-analytics.com/analytics.js"></script>
    
    <script async src="https://cdnjs.cloudflare.com/ajax/libs/autotrack/2.4.1/autotrack.js" integrity="sha512-HUmooslVKj4m6OBu0OgzjXXr+QuFYy/k7eLI5jdeEy/F4RSgMn6XRWRGkFi5IFaFgy7uFTkegp3Z0XnJf3Jq+g==" crossorigin="anonymous"></script>
    
  

  
  <link rel="alternate" href="/./index.xml" type="application/rss+xml" title="eye Bioinformatician">
  <link rel="feed" href="/./index.xml" type="application/rss+xml" title="eye Bioinformatician">
  

  <link rel="manifest" href="/site.webmanifest">
  <link rel="icon" type="image/png" href="/img/icon.png">
  <link rel="apple-touch-icon" type="image/png" href="/img/icon-192.png">

  <link rel="canonical" href="/./post/gi2019-day02/">

  <meta property="twitter:card" content="summary_large_image">
  
  <meta property="twitter:site" content="@https://twitter.com/David_McGaughey">
  <meta property="twitter:creator" content="@https://twitter.com/David_McGaughey">
  
  <meta property="og:site_name" content="eye Bioinformatician">
  <meta property="og:url" content="/./post/gi2019-day02/">
  <meta property="og:title" content="#GI2019 2019 11 07 | eye Bioinformatician">
  <meta property="og:description" content="Day 2 - Session 2 - SEQUENCING ALGORITHMS, VARIANT DISCOVERY AND GENOME ASSEMBLY Genomic sketching with HyperLogLog centroFlye—Assembling centromeres with long error-prone reads Genotyping structural variants in pangenome graphs using the vg toolkit Rapidly mapping raw nanopore signal with UNCALLED to enable real-time targeted sequencing The construct and utility of reference pan-genome graphs PRINCESS — A framework for comprehensive detection and phasing of SNPs and structural variants Efficient chromosome-scale haplotype-resolved assembly of human individuals Utilization of an ensemble approach for identification of driver fusions in pediatric cancer   Day 2 - Session 2 - SEQUENCING ALGORITHMS, VARIANT DISCOVERY AND GENOME ASSEMBLY Genome Informatics 2019 at CSHL">
  <meta property="og:locale" content="en-us">
  
  <meta property="article:published_time" content="2019-11-07T00:00:00&#43;00:00">
  
  <meta property="article:modified_time" content="2019-11-07T00:00:00&#43;00:00">
  

  
  

  <title>#GI2019 2019 11 07 | eye Bioinformatician</title>

</head>
<body id="top" data-spy="scroll" data-target="#toc" data-offset="71" >

<nav class="navbar navbar-default navbar-fixed-top" id="navbar-main">
  <div class="container">

    
    <div class="navbar-header">
      
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse"
              data-target=".navbar-collapse" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      
      <a class="navbar-brand" href="/">eye Bioinformatician</a>
    </div>

    
    <div class="collapse navbar-collapse">

      
      
      <ul class="nav navbar-nav navbar-right">
        

        
        
        
        
        
          
        

        <li class="nav-item">
          <a href="/#about">
            
            <span>Home</span>
            
          </a>
        </li>

        
        

        
        <li class="dropdown">
          <a href="#" class="dropdown-toggle" data-toggle="dropdown" aria-haspopup="true">
            
            <span>Posts</span>
            
            <span class="caret"></span>
          </a>
          <ul class="dropdown-menu">
            
            <li class="nav-item">
              <a href="/#posts">
                
                <span>Recent Posts</span>
                
              </a>
            </li>
            
            <li class="nav-item">
              <a href="/post">
                
                <span>All Posts</span>
                
              </a>
            </li>
            
          </ul>
        </li>

        
        

        
        
        
        
        
          
        

        <li class="nav-item">
          <a href="/#projects">
            
            <span>Projects</span>
            
          </a>
        </li>

        
        

        
        
        
        
        
          
        

        <li class="nav-item">
          <a href="/#culture">
            
            <span>Culture</span>
            
          </a>
        </li>

        
        

        
        
        
        
        
          
        

        <li class="nav-item">
          <a href="/#publications_selected">
            
            <span>Publications</span>
            
          </a>
        </li>

        
        
      

      
      </ul>

    </div>
  </div>
</nav>


<article class="article" itemscope itemtype="http://schema.org/Article">

  


  <div class="article-container">
    <h1 itemprop="name">#GI2019 2019 11 07</h1>

    

<div class="article-metadata">

  <span class="article-date">
    
    <time datetime="2019-11-07 00:00:00 &#43;0000 UTC" itemprop="datePublished dateModified">
      Nov 7, 2019
    </time>
  </span>
  <span itemscope itemprop="author publisher" itemtype="http://schema.org/Person">
    <meta itemprop="name" content="David McGaughey">
  </span>

  
  <span class="middot-divider"></span>
  <span class="article-reading-time">
    8 min read
  </span>
  

  
  
  <span class="middot-divider"></span>
  <a href="/./post/gi2019-day02/#disqus_thread"></a>
  

  
  
  
  <span class="middot-divider"></span>
  <span class="article-categories">
    <i class="fa fa-folder"></i>
    
    <a href="/./categories/bioinformatics/">bioinformatics</a
    >, 
    
    <a href="/./categories/gi2019/">gi2019</a
    >, 
    
    <a href="/./categories/conferences/">conferences</a
    >, 
    
    <a href="/./categories/talk/">talk</a
    >
    
  </span>
  
  

  
  
<div class="share-box" aria-hidden="true">
  <ul class="share">
    <li>
      <a class="twitter"
         href="https://twitter.com/intent/tweet?text=%23GI2019%202019%2011%2007&amp;url=%2f.%2fpost%2fgi2019-day02%2f"
         target="_blank" rel="noopener">
        <i class="fa fa-twitter"></i>
      </a>
    </li>
    <li>
      <a class="facebook"
         href="https://www.facebook.com/sharer.php?u=%2f.%2fpost%2fgi2019-day02%2f"
         target="_blank" rel="noopener">
        <i class="fa fa-facebook"></i>
      </a>
    </li>
    <li>
      <a class="linkedin"
         href="https://www.linkedin.com/shareArticle?mini=true&amp;url=%2f.%2fpost%2fgi2019-day02%2f&amp;title=%23GI2019%202019%2011%2007"
         target="_blank" rel="noopener">
        <i class="fa fa-linkedin"></i>
      </a>
    </li>
    <li>
      <a class="weibo"
         href="http://service.weibo.com/share/share.php?url=%2f.%2fpost%2fgi2019-day02%2f&amp;title=%23GI2019%202019%2011%2007"
         target="_blank" rel="noopener">
        <i class="fa fa-weibo"></i>
      </a>
    </li>
    <li>
      <a class="email"
         href="mailto:?subject=%23GI2019%202019%2011%2007&amp;body=%2f.%2fpost%2fgi2019-day02%2f">
        <i class="fa fa-envelope"></i>
      </a>
    </li>
  </ul>
</div>


  

</div>


    <div class="article-style" itemprop="articleBody">
      <div id="TOC">
<ul>
<li><a href="#day-2---session-2---sequencing-algorithms-variant-discovery-and-genome-assembly">Day 2 - Session 2 - SEQUENCING ALGORITHMS, VARIANT DISCOVERY AND GENOME ASSEMBLY</a></li>
<li><a href="#genomic-sketching-with-hyperloglog">Genomic sketching with HyperLogLog</a></li>
<li><a href="#centroflyeassembling-centromeres-with-long-error-prone-reads">centroFlye—Assembling centromeres with long error-prone reads</a></li>
<li><a href="#genotyping-structural-variants-in-pangenome-graphs-using-the-vg-toolkit">Genotyping structural variants in pangenome graphs using the vg toolkit</a></li>
<li><a href="#rapidly-mapping-raw-nanopore-signal-with-uncalled-to-enable-real-time-targeted-sequencing">Rapidly mapping raw nanopore signal with UNCALLED to enable real-time targeted sequencing</a></li>
<li><a href="#the-construct-and-utility-of-reference-pan-genome-graphs">The construct and utility of reference pan-genome graphs</a></li>
<li><a href="#princess-a-framework-for-comprehensive-detection-and-phasing-of-snps-and-structural-variants">PRINCESS — A framework for comprehensive detection and phasing of SNPs and structural variants</a></li>
<li><a href="#efficient-chromosome-scale-haplotype-resolved-assembly-of-human-individuals">Efficient chromosome-scale haplotype-resolved assembly of human individuals</a></li>
<li><a href="#utilization-of-an-ensemble-approach-for-identification-of-driver-fusions-in-pediatric-cancer">Utilization of an ensemble approach for identification of driver fusions in pediatric cancer</a></li>
</ul>
</div>

<div id="day-2---session-2---sequencing-algorithms-variant-discovery-and-genome-assembly" class="section level1">
<h1>Day 2 - Session 2 - SEQUENCING ALGORITHMS, VARIANT DISCOVERY AND GENOME ASSEMBLY</h1>
<p>Genome Informatics 2019 at CSHL</p>
<p><strong>Bold</strong> is the speaker</p>
</div>
<div id="genomic-sketching-with-hyperloglog" class="section level1">
<h1>Genomic sketching with HyperLogLog</h1>
<p>Daniel N. Baker, <strong>Ben Langmead.</strong></p>
<p>Slides at…crap missed it. Email Ben I guess?</p>
<p>Sketching</p>
<ul>
<li>algorithm to collapse genomes into summary info</li>
<li>fasta -&gt; kmers (“shingle”) -&gt; sample (smartly)</li>
<li>showing set relationships</li>
<li>the “minimum” of a set of sketches gives you useful info on the range you would expect</li>
</ul>
<p>Unions and Intersections</p>
<ul>
<li>space of possible coincidences requires multiple samples to get useful info</li>
<li>bottom 3?</li>
<li>or minimum in 3 partitions?</li>
</ul>
<p>MinHash uses bottom-k approach (Ondov / Phillipy)</p>
<p>Tremendous talk - I feel like I understand what is happening. Invite Ben to talk at your place…I’d imagine you’d get a great talk.</p>
<p>OK, so take <em>log log minimum</em></p>
<p>Daniel Baker wrote:</p>
<ul>
<li>bit.ly/dash_pre</li>
<li>github.com/dnbaker/dashing</li>
<li>github.com/dnbaker/sketch</li>
</ul>
<p>HyperLogLog *HLL)</p>
<ul>
<li>k partition</li>
<li>log<sub>2</sub>n</li>
<li>exponent</li>
<li>average, bias correction</li>
</ul>
<p>Highly vectorized (for speed! Massively parallel calculation)</p>
<ul>
<li>only on intel?</li>
</ul>
<p>Comparing to MinHash</p>
<ul>
<li>HLL deals with lopsided sets better (bottom k not ideal, better minimum k partition)</li>
<li>Uses WAY less memory and wall clock</li>
</ul>
<p>Future stuffs</p>
<ul>
<li>multi-k (kmer lengths)</li>
<li>weighted jaccard</li>
</ul>
</div>
<div id="centroflyeassembling-centromeres-with-long-error-prone-reads" class="section level1">
<h1>centroFlye—Assembling centromeres with long error-prone reads</h1>
<p><strong>Andrey Bzikadze</strong>, Pavel A. Pevzner.</p>
<p><span class="citation">@AndreyBzikadze</span></p>
<p>biorxiv preprint (missed link)</p>
<p>brings up t2t consortium (telomere to telomere)</p>
<p>github.com/nanopore-wgs-consortium/CHM13</p>
<p>mentions karen miga’s chrX t2t biorxiv preprint</p>
<p>centromere:</p>
<ul>
<li>assembler’s favoriate region
<ul>
<li>probably not, but keeps them gainfully employed for now</li>
</ul></li>
<li>highly repetitive….etc etc.</li>
<li>3% of genome - and because are unassembled, poor idea how they influence human diseasse</li>
</ul>
<p>centroFlye</p>
<ul>
<li>classify reads (prefix…something…suffix… what is going on? Andrey already moved on)
<ul>
<li>guessing something to do with begining / middle / end of repeat?</li>
</ul></li>
<li>find rare k-mers that can be used to anchor assembly</li>
<li>but rare k-mers are usually <strong>errors</strong>
<ul>
<li>if the <strong>distance between two rare k-mers is conserved</strong> then they are likely real</li>
<li>neat idea!</li>
<li>but I’m not certain what conservation is…</li>
</ul></li>
<li>throw everything away except for rare k-mers</li>
</ul>
<p>centroFlye used to improve chrX centromere assembly</p>
<ul>
<li>greater length</li>
<li>but how do you more robustly define success?</li>
<li>one metric is to count shared (or discrepant) k-mers between assembly and a read</li>
<li>discorance (a,b) = sharedRead(assembly A) - sharedRead(assembly B)</li>
</ul>
</div>
<div id="genotyping-structural-variants-in-pangenome-graphs-using-the-vg-toolkit" class="section level1">
<h1>Genotyping structural variants in pangenome graphs using the vg toolkit</h1>
<p><strong>Jean Monlong</strong>, Glenn Hickey, David Heller, Jonas Andreas Sibbesen, Jouni Siren, Jordan Eizenga, Eric T. Dawson, Erik Garrison, Adam Novak, Benedict Paten.</p>
<p>Yo time for graphs!</p>
<p>Intro on value over linear genomes</p>
<ul>
<li>highly polymorphic regions should have better geontyping</li>
<li>points out graphs not for discovery but for genotyping</li>
</ul>
<p>Increasing <em>n</em> of SV catalogs with long read data</p>
<ul>
<li>HGSVC (3 samples)</li>
<li>SVPOP (15)</li>
<li>GiaB (1)</li>
</ul>
<p>Probably lots of missing SV from 50-500bp (cmparing HGSVC against gnomadSV which is short-read based)</p>
<p>github.com/vgteam/vg</p>
<p>vg toolkit can make graphs, map reads, and call variants</p>
<p>“can we genotype SVs from short-read datasets with vg toolkit”</p>
<p>Build graph with long-read data, genotype SVs against it with short-reads</p>
<p>R package to evaluate calls</p>
<ul>
<li>github.com/jmonlong/sveval</li>
</ul>
<p>with simulated data</p>
<ul>
<li>vg vs paragraph vs bayestyper vs delly vs SVTyper (last two are non-graph tolls)</li>
<li>basestyper does best (not their tool)</li>
<li>f1 score</li>
</ul>
<p>with real data from HGSVC</p>
<ul>
<li>all do worse</li>
</ul>
<p>simple repeat regions are hard to genotype</p>
<ul>
<li>line, sine, alu,</li>
<li>for both insertion/deletion</li>
</ul>
<p>hard to work with vcf…representation confusion with equivalent representation and oversimplification</p>
<ul>
<li>just use the assemblies?!</li>
<li><p>testing with yeast right now</p></li>
<li><p>vg performance best/near best</p></li>
</ul>
</div>
<div id="rapidly-mapping-raw-nanopore-signal-with-uncalled-to-enable-real-time-targeted-sequencing" class="section level1">
<h1>Rapidly mapping raw nanopore signal with UNCALLED to enable real-time targeted sequencing</h1>
<p><strong>Sam Kovaka</strong>, Yunfan Fan, Winston Timp, Michael C. Schatz.</p>
<p>Don’t want unwanted DNA sequence</p>
<ul>
<li>especially for low throughput sequencers (like ONT)</li>
<li>targeted seq techniques not ideal for ONT
<ul>
<li>length not enough</li>
<li>erase DNA info (methylation, etc.)</li>
</ul></li>
</ul>
<p>ReadUntil</p>
<ul>
<li>selectively start/stop</li>
<li>davemcg: uh, does this actually work yet?
<ul>
<li>was announced a year ago or more??</li>
</ul></li>
</ul>
<p>UNCALLED</p>
<ul>
<li>utility for nanopore current alignment to large expanses of DNA</li>
<li>novel streaming algorithm which maps raw nanopore signale in real-time</li>
<li>works with raw nanopore output (electrical signal)</li>
<li>discussing kmer matching to nanopore output
<ul>
<li>it looks…hard</li>
</ul></li>
<li>insane slide discussing algorith/implementation
<ul>
<li>i am not equipped to summarize</li>
</ul></li>
</ul>
<p>ReadUntil
- enrichment (ejcet a read if it does not map)
- depletion (the converse)</p>
<p>Longer reads get more enrichment</p>
<ul>
<li>saves on seq cost and shorter reads hard to map</li>
</ul>
<p>Testing with bacteria</p>
<ul>
<li>4.5x enrichment of on-target</li>
<li>0.4x off-target</li>
</ul>
<p>Doing a human cancer “panel” for SV</p>
<ul>
<li>28 genes</li>
<li>overall 3X enrichment</li>
<li>and can assemble the genes with much higher success rate</li>
</ul>
<p>Future</p>
<ul>
<li>want to improve yield</li>
<li>is ssDNA “knotting” and blocking ejection?</li>
<li>better API from ONT?</li>
</ul>
<p>github.com/skovaka/UNCALLED</p>
</div>
<div id="the-construct-and-utility-of-reference-pan-genome-graphs" class="section level1">
<h1>The construct and utility of reference pan-genome graphs</h1>
<p><strong>Heng Li.</strong></p>
<p>pangenome = collection of genomes</p>
<ul>
<li>graph (collapse similar seq)</li>
<li>or</li>
<li>compressed full-text index</li>
</ul>
<p>10 years ago (review with Nils Homer): “alignment against multiple genomes will become increasingly important”</p>
<p>“hasn’t happened yet”</p>
<p>vcf doesn’t handle graphs</p>
<p>GFP format (assembly format)</p>
<ul>
<li>davemcg note: looks like a directed(?) acyclic graph</li>
<li>not good because if you split a segment, the <strong>coordinate changes</strong></li>
</ul>
<p>Proposal:</p>
<ul>
<li>reference GFA (GFA <strong>with tags that have coord info and version</strong>)</li>
<li>start with GRCh38, incrementally add other genomes</li>
<li>blacklist and decoy seq for linear tools</li>
<li>updates to preserve coords</li>
</ul>
<p>Incrementally add new assemblies</p>
<ul>
<li>add assembly, make graph</li>
<li>then add another assembly, make new graph</li>
</ul>
<p>Discussion of alignment linear seq to graph…which I don’t quite follow</p>
<ul>
<li>but ideal approach too slow</li>
<li>so approximate with k shortest paths</li>
</ul>
<p>minigraph</p>
<ul>
<li>based on minimap2</li>
<li>limitations!
<ul>
<li>doesn’t work with dense graphs (too many k paths?)</li>
</ul></li>
<li>1.5 hours over 24 CPUs with human graph of 20 haplotypes
<ul>
<li>36k bubbles</li>
<li>94% of GRCh37 is invariable</li>
</ul></li>
<li><a href="https://github.com/lh3/minigraph" class="uri">https://github.com/lh3/minigraph</a> (didn’t give link but it’s on his github)</li>
</ul>
<p>multi-allelic regions and minisatellites are hard to assemble and genotype</p>
<p>applications</p>
<ul>
<li>blacklist regions (SV, etc)</li>
</ul>
</div>
<div id="princess-a-framework-for-comprehensive-detection-and-phasing-of-snps-and-structural-variants" class="section level1">
<h1>PRINCESS — A framework for comprehensive detection and phasing of SNPs and structural variants</h1>
<p><strong>Medhat Mahmoud</strong>, Winston Timp, Fritz J. Sedlazeck.</p>
<p>Burn. Winston Timp not on the slide.</p>
<p><span class="citation">@MedhatHelmy7</span></p>
<p>Points out there are diff tech to detect SNV, SV, Phasing, Methylation</p>
<p>Would prefer just use one platform</p>
<ul>
<li>ONT?</li>
</ul>
<p>PRINCESS</p>
<ul>
<li><strong>framework</strong> to integrate tools to analyze Long Reads</li>
<li>mapping: minimap2 and/or NGMLR</li>
<li>SNVs: Clair (deep NN) - first NN mention?</li>
<li>SVs: Sniffles</li>
<li>Phasing: WhatsHap, Princess-subtools</li>
<li>can also use trio info (SNPs)</li>
<li>Nanopolish for Methyl C</li>
<li>outputs….statistics!</li>
<li>runs on workstation or hpc or cloud</li>
<li>outputs phase SNV, SV with optional MethylC</li>
</ul>
<p>Benchmarking with GiaB HG002</p>
<ul>
<li>PacBio CCS, PacBio CLR, ONT</li>
<li>LOOOOOONG tail for ONT (read length dist)</li>
<li>remove &lt;500bp in length</li>
<li>try diff coverage (10x, 25x, 50x, 95x)</li>
<li>with one tech (70-80% sensitivity, 85-93% precision for SNP)
<ul>
<li>PacBio CCS &gt; CLR &gt; ONT</li>
</ul></li>
<li>SV calling
<ul>
<li>CCS &gt; CLR == ONT (80% sensitivity, 85-95% precision)</li>
</ul></li>
<li>Phasing
<ul>
<li>Use parental SNPs</li>
<li>N50 better with ONT (loooong reads)</li>
<li>but better accuracy with PacBio</li>
</ul></li>
</ul>
<p>Now real data</p>
<ul>
<li>Common Disease Genomics</li>
<li>focus on cardiovascular and … something else missed it</li>
<li>44k+ short read WGS -&gt; do some with long read -&gt; merge to make comprehensive genome -&gt; which can be used to improve catalogues</li>
<li>SVcollector (github.com/fritzsedlazeck/SVCollector)</li>
<li>then Princess (github.com/MeHelmy/princess)</li>
<li>then Paragraph (github.com/Illumina/paragraph)
<ul>
<li>uh, what is paragraph</li>
<li>from github: &quot; A graph-based structural variant genotyper for short-read sequence data.&quot;</li>
<li>oh, that makes sense</li>
<li>so use 44k short reads on graph?</li>
</ul></li>
</ul>
</div>
<div id="efficient-chromosome-scale-haplotype-resolved-assembly-of-human-individuals" class="section level1">
<h1>Efficient chromosome-scale haplotype-resolved assembly of human individuals</h1>
<p><strong>Shilpa Garg</strong>, Arkarachai Fungtammasan, Anthony Schmitt, Andrew
Carroll, Paul Pelusol, Emily Hatas, Fritz Sedlazeck, Justin Zook, Mike
Chou, John Aach, Jason Chin, Heng Li, George Church.</p>
<p>Discussing PacBio HiFi tech</p>
<ul>
<li>circularize DNA</li>
<li>15k long</li>
<li>99% with 5 passes</li>
</ul>
<p>Now Hi-C discussion/intro</p>
<ul>
<li>wild how common Hi-C is. I guess I’m getting old - in grad school this stuff was just IMPOSSIBLE to do</li>
</ul>
<p>now genome assembly discussion</p>
<ul>
<li>graph geneeration …. getting longer reads gives you fewer branches on your graph (contiguous assembly)</li>
</ul>
<p>PacBio alone not enough for full assembly/phasing</p>
<ul>
<li>but can use trios with Koren/Phillipy approach (at least works in goats)
<ul>
<li>trio canu</li>
</ul></li>
<li>but want full haplotypes <strong>without</strong> trios</li>
</ul>
<p>workflow</p>
<ul>
<li>PacBio CCS</li>
<li>make contigs</li>
<li>use Hi-C to scaffold</li>
<li>peregrine -&gt; hirrise -&gt; deepvariant -&gt; whatshap + hapcut2 =&gt; whatshap -&gt; peregrine</li>
<li>walltime: <strong>one day</strong></li>
</ul>
<p>benchmark with PGP1, HG002, NA12878</p>
<ul>
<li>contig n50, NGA50, phasing hamming error, phasing switch error</li>
<li>compare against trio canu</li>
<li>generally better than trio canu across all metrics</li>
<li>consistent across diff genomes</li>
</ul>
<p>checking in HLA region</p>
<ul>
<li>can build HLA in two contigs</li>
</ul>
<p>Link!</p>
<ul>
<li>github.com/shilpagarg/WHdenovo</li>
</ul>
</div>
<div id="utilization-of-an-ensemble-approach-for-identification-of-driver-fusions-in-pediatric-cancer" class="section level1">
<h1>Utilization of an ensemble approach for identification of driver fusions in pediatric cancer</h1>
<p><strong>Stephanine LaHaye</strong>, Kyle Voytovich, James Fitch, Natalie Bir, Sean D. McGrath, Anthony Miller, Amy Wetzel, Vincent Magrini, Catherine E. Cottrell, Elaine R. Mardis, Richard K. Wilson.</p>
<p>Last talk of the session!</p>
<p>ID fusions in cancer (pediatric)</p>
<p>Interesting….pediatric cancer have fwer mutations (which I guess makes sense)</p>
<ul>
<li>SN Grobner et al Nature 2018</li>
<li>fusions are helpful to classify cancers</li>
<li>difficult via RNA-seq</li>
</ul>
<p>Used 4 (6?) different callers for an ensembl approach</p>
<ul>
<li>oh god a multi venn diagram</li>
<li>USE AN UpSET plot!!!!</li>
<li>anyways, not much agreement (but low n is a feature if they are true)</li>
<li>not clear how they actually <strong>know</strong> most are false positive</li>
<li>specificity != sensitivity</li>
</ul>
<p>LOL Runtimes</p>
<ul>
<li>vary from 30 minutes to 75 hours</li>
<li>dragen the fast one</li>
</ul>
<p>Synthetic fusion gene cDNA for benchmarking</p>
<ul>
<li>more dilution == harder to detect</li>
<li>1:50 dilution == OK, 1:250 == not so good</li>
</ul>
<p>Running this crazy big complicated pipeline in AWS serveeless config</p>
<ul>
<li>looks like they are paying AWS a lot of $</li>
</ul>
</div>

    </div>

    


<div class="article-tags">
  
  <a class="btn btn-primary btn-outline" href="/./tags/bioinformatics/">bioinformatics</a>
  
  <a class="btn btn-primary btn-outline" href="/./tags/gi2019/">gi2019</a>
  
  <a class="btn btn-primary btn-outline" href="/./tags/conferences/">conferences</a>
  
  <a class="btn btn-primary btn-outline" href="/./tags/talk/">talk</a>
  
</div>




    
    
    <div class="article-widget">
      <div class="hr-light"></div>
      <h3>Related</h3>
      <ul>
        
        <li><a href="/post/gi2019-day02-night/">#GI2019 2019 11 07 Night Session</a></li>
        
        <li><a href="/post/gi2019-day02-night/">#GI2019 2019 11 08 Morning Session</a></li>
        
        <li><a href="/post/gi2019-day01/">#GI2019 2019 11 06</a></li>
        
        <li><a href="/post/one-developer-portal-genesis/">One Developer Portal: eyeIntegration Genesis</a></li>
        
        <li><a href="/post/one-developer-portal-optimization/">One Developer Portal: eyeIntegration Web Optimization</a></li>
        
      </ul>
    </div>
    

    

    
<section id="comments">
  <div id="disqus_thread"></div>
<script>
    var disqus_config = function () {
    
    
    
    };
    (function() {
        if (["localhost", "127.0.0.1"].indexOf(window.location.hostname) != -1) {
            document.getElementById('disqus_thread').innerHTML = 'Disqus comments not available by default when the website is previewed locally.';
            return;
        }
        var d = document, s = d.createElement('script'); s.async = true;
        s.src = '//' + "davemcg-gitub-io" + '.disqus.com/embed.js';
        s.setAttribute('data-timestamp', +new Date());
        (d.head || d.body).appendChild(s);
    })();
</script>
<noscript>Please enable JavaScript to view the <a href="https://disqus.com/?ref_noscript">comments powered by Disqus.</a></noscript>
<a href="https://disqus.com" class="dsq-brlink">comments powered by <span class="logo-disqus">Disqus</span></a>
</section>



  </div>
</article>

<footer class="site-footer">
  <div class="container">
    <p class="powered-by">

      &copy; 2018 &middot; 

      Powered by the
      <a href="https://sourcethemes.com/academic/" target="_blank" rel="noopener">Academic theme</a> for
      <a href="https://gohugo.io" target="_blank" rel="noopener">Hugo</a>.

      <span class="pull-right" aria-hidden="true">
        <a href="#" id="back_to_top">
          <span class="button_icon">
            <i class="fa fa-chevron-up fa-2x"></i>
          </span>
        </a>
      </span>

    </p>
  </div>
</footer>


<div id="modal" class="modal fade" role="dialog">
  <div class="modal-dialog">
    <div class="modal-content">
      <div class="modal-header">
        <button type="button" class="close btn-large" data-dismiss="modal">&times;</button>
        <h4 class="modal-title">Cite</h4>
      </div>
      <div>
        <pre><code class="modal-body tex"></code></pre>
      </div>
      <div class="modal-footer">
        <a class="btn btn-primary btn-outline js-copy-cite" href="#" target="_blank">
          <i class="fa fa-copy"></i> Copy
        </a>
        <a class="btn btn-primary btn-outline js-download-cite" href="#" target="_blank">
          <i class="fa fa-download"></i> Download
        </a>
        <div id="modal-error"></div>
      </div>
    </div>
  </div>
</div>

    

    
    
    <script id="dsq-count-scr" src="//davemcg-gitub-io.disqus.com/count.js" async></script>
    

    

    
    <script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.2.1/jquery.min.js" integrity="sha512-3P8rXCuGJdNZOnUx/03c1jOTnMn3rP63nBip5gOP2qmUh5YAdVAvFZ1E+QLZZbC1rtMrQb+mah3AfYW11RUrWA==" crossorigin="anonymous"></script>
    <script src="https://cdnjs.cloudflare.com/ajax/libs/jquery.imagesloaded/4.1.3/imagesloaded.pkgd.min.js" integrity="sha512-umsR78NN0D23AzgoZ11K7raBD+R6hqKojyBZs1w8WvYlsI+QuKRGBx3LFCwhatzBunCjDuJpDHwxD13sLMbpRA==" crossorigin="anonymous"></script>
    <script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.3.7/js/bootstrap.min.js" integrity="sha512-iztkobsvnjKfAtTNdHkGVjAYTrrtlC7mGp/54c40wowO7LhURYl3gVzzcEqGl/qKXQltJ2HwMrdLcNUdo+N/RQ==" crossorigin="anonymous"></script>
    <script src="https://cdnjs.cloudflare.com/ajax/libs/jquery.isotope/3.0.4/isotope.pkgd.min.js" integrity="sha512-VDBOIlDbuC4VWxGJNmuFRQ0Li0SKkDpmGyuhAG5LTDLd/dJ/S0WMVxriR2Y+CyPL5gzjpN4f/6iqWVBJlht0tQ==" crossorigin="anonymous"></script>
    <script src="https://cdnjs.cloudflare.com/ajax/libs/fancybox/3.2.5/jquery.fancybox.min.js" integrity="sha256-X5PoE3KU5l+JcX+w09p/wHl9AzK333C4hJ2I9S5mD4M=" crossorigin="anonymous"></script>
    
    
    <script src="/js/hugo-academic.js"></script>
    

    
    
      
      
      <script src="//cdnjs.cloudflare.com/ajax/libs/highlight.js/9.12.0/highlight.min.js" integrity="sha256-/BfiIkHlHoVihZdc6TFuj7MmJ0TWcWsMXkeDFwhi0zw=" crossorigin="anonymous"></script>
      

      

      

      <script>hljs.initHighlightingOnLoad();</script>
    

    
    
    <script type="text/x-mathjax-config">
        MathJax.Hub.Config({ tex2jax: { inlineMath: [['$','$'], ['\\(','\\)']] } });
    </script>
    
    <script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.1/MathJax.js?config=TeX-AMS_CHTML" integrity="sha512-tOav5w1OjvsSJzePRtt2uQPFwBoHt1VZcUq8l8nm5284LEKE9FSJBQryzMBzHxY5P0zRdNqEcpLIRVYFNgu1jw==" crossorigin="anonymous"></script>
    
    

  </body>
</html>

