<!DOCTYPE html>
<html lang="en-us">
<head>

  <meta charset="utf-8">
  <meta name="viewport" content="width=device-width, initial-scale=1">
  <meta http-equiv="X-UA-Compatible" content="IE=edge">
  <meta name="theme" content="hugo-academic">
  <meta name="generator" content="Hugo 0.40" />
  <meta name="author" content="David McGaughey">

  
  
  
  
    
      
    
  
  <meta name="description" content="Intro 2018-09-17 Sarah Teichmann (Teichmann Lab) Girgio Gonnella (Stefan Kurtz) Luke Zappia (Oshlack?) Laura Huerta (Papatheodorou) Casey Greene (Greene) Sergei Yakneen   Intro Very sparse and poorly written notes covering #GI2018.
Typos everywhere. Things may change dramatically over time as I scan back through notes.
I’ve tried to respect #notwitter. Will be updated periodically.
Speaker (Lab | Group)
BOLDED is voice
 2018-09-17  Sarah Teichmann (Teichmann Lab) Cell Atlas Technologies and the Maternal-Fetal Interface">

  
  <link rel="alternate" hreflang="en-us" href="/./post/gi2018-day-one/">

  


  

  
  
  <meta name="theme-color" content="#0095eb">
  
  
  
  
    
  
  
    
    
      
        <link rel="stylesheet" href="//cdnjs.cloudflare.com/ajax/libs/highlight.js/9.12.0/styles/github.min.css">
      
    
  
  
  <link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.3.7/css/bootstrap.min.css" integrity="sha512-6MXa8B6uaO18Hid6blRMetEIoPqHf7Ux1tnyIQdpt9qI5OACx7C+O3IVTr98vwGnlcg0LOLa02i9Y1HpVhlfiw==" crossorigin="anonymous">
  <link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/academicons/1.8.6/css/academicons.min.css" integrity="sha256-uFVgMKfistnJAfoCUQigIl+JfUaP47GrRKjf6CTPVmw=" crossorigin="anonymous">
  <link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/4.7.0/css/font-awesome.min.css" integrity="sha512-SfTiTlX6kk+qitfevl/7LibUOeJWlt9rbyDn92a1DqWOw9vWG2MFoays0sgObmWazO5BQPiFucnnEAjpAB+/Sw==" crossorigin="anonymous">
  <link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/fancybox/3.2.5/jquery.fancybox.min.css" integrity="sha256-ygkqlh3CYSUri3LhQxzdcm0n1EQvH2Y+U5S2idbLtxs=" crossorigin="anonymous">
  
  
  
  
  <link rel="stylesheet" href="//fonts.googleapis.com/css?family=Montserrat:400,700%7cRoboto:400,400italic,700%7cRoboto&#43;Mono">
  
  <link rel="stylesheet" href="/styles.css">
  

  
    <script>
      window.ga=window.ga||function(){(ga.q=ga.q||[]).push(arguments)};ga.l=+new Date;
      ga('create', 'UA-113461893-1', 'auto');
      
      ga('require', 'eventTracker');
      ga('require', 'outboundLinkTracker');
      ga('require', 'urlChangeTracker');
      ga('send', 'pageview');
    </script>
    <script async src="//www.google-analytics.com/analytics.js"></script>
    
    <script async src="https://cdnjs.cloudflare.com/ajax/libs/autotrack/2.4.1/autotrack.js" integrity="sha512-HUmooslVKj4m6OBu0OgzjXXr+QuFYy/k7eLI5jdeEy/F4RSgMn6XRWRGkFi5IFaFgy7uFTkegp3Z0XnJf3Jq+g==" crossorigin="anonymous"></script>
    
  

  
  <link rel="alternate" href="/./index.xml" type="application/rss+xml" title="eye Bioinformatician">
  <link rel="feed" href="/./index.xml" type="application/rss+xml" title="eye Bioinformatician">
  

  <link rel="manifest" href="/site.webmanifest">
  <link rel="icon" type="image/png" href="/img/icon.png">
  <link rel="apple-touch-icon" type="image/png" href="/img/icon-192.png">

  <link rel="canonical" href="/./post/gi2018-day-one/">

  <meta property="twitter:card" content="summary_large_image">
  
  <meta property="twitter:site" content="@https://twitter.com/David_McGaughey">
  <meta property="twitter:creator" content="@https://twitter.com/David_McGaughey">
  
  <meta property="og:site_name" content="eye Bioinformatician">
  <meta property="og:url" content="/./post/gi2018-day-one/">
  <meta property="og:title" content="#GI2018 - Day One | eye Bioinformatician">
  <meta property="og:description" content="Intro 2018-09-17 Sarah Teichmann (Teichmann Lab) Girgio Gonnella (Stefan Kurtz) Luke Zappia (Oshlack?) Laura Huerta (Papatheodorou) Casey Greene (Greene) Sergei Yakneen   Intro Very sparse and poorly written notes covering #GI2018.
Typos everywhere. Things may change dramatically over time as I scan back through notes.
I’ve tried to respect #notwitter. Will be updated periodically.
Speaker (Lab | Group)
BOLDED is voice
 2018-09-17  Sarah Teichmann (Teichmann Lab) Cell Atlas Technologies and the Maternal-Fetal Interface">
  <meta property="og:locale" content="en-us">
  
  <meta property="article:published_time" content="2018-09-17T00:00:00&#43;00:00">
  
  <meta property="article:modified_time" content="2018-09-17T00:00:00&#43;00:00">
  

  
  

  <title>#GI2018 - Day One | eye Bioinformatician</title>

</head>
<body id="top" data-spy="scroll" data-target="#toc" data-offset="71" >

<nav class="navbar navbar-default navbar-fixed-top" id="navbar-main">
  <div class="container">

    
    <div class="navbar-header">
      
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse"
              data-target=".navbar-collapse" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      
      <a class="navbar-brand" href="/">eye Bioinformatician</a>
    </div>

    
    <div class="collapse navbar-collapse">

      
      
      <ul class="nav navbar-nav navbar-right">
        

        
        
        
        
        
          
        

        <li class="nav-item">
          <a href="/#about">
            
            <span>Home</span>
            
          </a>
        </li>

        
        

        
        <li class="dropdown">
          <a href="#" class="dropdown-toggle" data-toggle="dropdown" aria-haspopup="true">
            
            <span>Posts</span>
            
            <span class="caret"></span>
          </a>
          <ul class="dropdown-menu">
            
            <li class="nav-item">
              <a href="/#posts">
                
                <span>Recent Posts</span>
                
              </a>
            </li>
            
            <li class="nav-item">
              <a href="/post">
                
                <span>All Posts</span>
                
              </a>
            </li>
            
          </ul>
        </li>

        
        

        
        
        
        
        
          
        

        <li class="nav-item">
          <a href="/#projects">
            
            <span>Projects</span>
            
          </a>
        </li>

        
        

        
        
        
        
        
          
        

        <li class="nav-item">
          <a href="/#culture">
            
            <span>Culture</span>
            
          </a>
        </li>

        
        

        
        
        
        
        
          
        

        <li class="nav-item">
          <a href="/#publications_selected">
            
            <span>Publications</span>
            
          </a>
        </li>

        
        
      

      
      </ul>

    </div>
  </div>
</nav>


<article class="article" itemscope itemtype="http://schema.org/Article">

  


  <div class="article-container">
    <h1 itemprop="name">#GI2018 - Day One</h1>

    

<div class="article-metadata">

  <span class="article-date">
    
    <time datetime="2018-09-17 00:00:00 &#43;0000 UTC" itemprop="datePublished dateModified">
      Sep 17, 2018
    </time>
  </span>
  <span itemscope itemprop="author publisher" itemtype="http://schema.org/Person">
    <meta itemprop="name" content="David McGaughey">
  </span>

  
  <span class="middot-divider"></span>
  <span class="article-reading-time">
    5 min read
  </span>
  

  
  
  <span class="middot-divider"></span>
  <a href="/./post/gi2018-day-one/#disqus_thread"></a>
  

  
  
  
  <span class="middot-divider"></span>
  <span class="article-categories">
    <i class="fa fa-folder"></i>
    
    <a href="/./categories/gi2018/">GI2018</a
    >, 
    
    <a href="/./categories/bioinformatics/">bioinformatics</a
    >
    
  </span>
  
  

  
  
<div class="share-box" aria-hidden="true">
  <ul class="share">
    <li>
      <a class="twitter"
         href="https://twitter.com/intent/tweet?text=%23GI2018%20-%20Day%20One&amp;url=%2f.%2fpost%2fgi2018-day-one%2f"
         target="_blank" rel="noopener">
        <i class="fa fa-twitter"></i>
      </a>
    </li>
    <li>
      <a class="facebook"
         href="https://www.facebook.com/sharer.php?u=%2f.%2fpost%2fgi2018-day-one%2f"
         target="_blank" rel="noopener">
        <i class="fa fa-facebook"></i>
      </a>
    </li>
    <li>
      <a class="linkedin"
         href="https://www.linkedin.com/shareArticle?mini=true&amp;url=%2f.%2fpost%2fgi2018-day-one%2f&amp;title=%23GI2018%20-%20Day%20One"
         target="_blank" rel="noopener">
        <i class="fa fa-linkedin"></i>
      </a>
    </li>
    <li>
      <a class="weibo"
         href="http://service.weibo.com/share/share.php?url=%2f.%2fpost%2fgi2018-day-one%2f&amp;title=%23GI2018%20-%20Day%20One"
         target="_blank" rel="noopener">
        <i class="fa fa-weibo"></i>
      </a>
    </li>
    <li>
      <a class="email"
         href="mailto:?subject=%23GI2018%20-%20Day%20One&amp;body=%2f.%2fpost%2fgi2018-day-one%2f">
        <i class="fa fa-envelope"></i>
      </a>
    </li>
  </ul>
</div>


  

</div>


    <div class="article-style" itemprop="articleBody">
      <div id="TOC">
<ul>
<li><a href="#intro">Intro</a></li>
<li><a href="#section">2018-09-17</a></li>
<li><a href="#sarah-teichmann-teichmann-lab">Sarah Teichmann (Teichmann Lab)</a></li>
<li><a href="#girgio-gonnella-stefan-kurtz">Girgio Gonnella (Stefan Kurtz)</a></li>
<li><a href="#luke-zappia-oshlack">Luke Zappia (Oshlack?)</a></li>
<li><a href="#laura-huerta-papatheodorou">Laura Huerta (Papatheodorou)</a></li>
<li><a href="#casey-greene-greene">Casey Greene (Greene)</a></li>
<li><a href="#sergei-yakneen">Sergei Yakneen</a></li>
</ul>
</div>

<div id="intro" class="section level1">
<h1>Intro</h1>
<p>Very sparse and poorly written notes covering <a href="https://twitter.com/hashtag/GI2018?src=hash">#GI2018</a>.</p>
<p>Typos everywhere. Things may change dramatically over time as I scan back through notes.</p>
<p>I’ve tried to respect #notwitter. Will be updated periodically.</p>
<p>Speaker (Lab | Group)</p>
<p><strong>BOLDED is voice</strong></p>
</div>
<div id="section" class="section level1">
<h1>2018-09-17</h1>
</div>
<div id="sarah-teichmann-teichmann-lab" class="section level1">
<h1>Sarah Teichmann (Teichmann Lab)</h1>
<p>Cell Atlas Technologies and the Maternal-Fetal Interface</p>
<p><a href="humancellatlas.org">Human Cell Atlas plug</a></p>
<ul>
<li>600 scientists</li>
<li>scRNAseq + spatial methods</li>
</ul>
<p>scRNA works OK</p>
<ul>
<li>70%+ pearson cor (among genes?)</li>
<li>Detection limit pretty good after 1 million reads</li>
<li><a href="https://www.biorxiv.org/content/early/2016/09/08/073692">Svensson and Teichmann preprint</a></li>
</ul>
<p>Spatial Tech</p>
<ul>
<li>i.e. gene expression at voxel resolution in a non-destroyed tissue
<ul>
<li>1 voxel ~ 20 cells</li>
</ul></li>
<li>Hope to ID spatially variable genes
<ul>
<li>[github.com/TeichLab/SpatialDE]</li>
<li>Svensson and Teichmann</li>
</ul></li>
</ul>
<p>Braga and Stegle: Merge Spatial TX with scRNA</p>
<p>Future stuffs: Histology merged with scRNA</p>
<p>Back to HCA</p>
<ul>
<li>Many tissues, many scientists, many countries</li>
<li>curious if any integration with Chan-Zuckerberg</li>
</ul>
<p>Moving onto to Maternal-fetal interface</p>
<ul>
<li>tricky area</li>
<li>immune component</li>
<li>two organisms</li>
<li><em>tumor like</em></li>
<li>scRNA + WGS resolved fetal/maternal adn cell tyes</li>
</ul>
</div>
<div id="girgio-gonnella-stefan-kurtz" class="section level1">
<h1>Girgio Gonnella (Stefan Kurtz)</h1>
<p>Flexible and interactive visualization of GFA sequence graphs</p>
<p>Graphical Fragment Assembly</p>
<ul>
<li>format for representing sequence graphs</li>
<li>contig output not as informative as graphs</li>
</ul>
<p>GFA1 proposal format from Heng Li releaseed in 2016-09</p>
<p>GFA2 2017-01 more general</p>
<p>Today more assemblers are using</p>
<p>GFA2 format:</p>
<ul>
<li>header</li>
<li>sequences</li>
<li>relationships</li>
<li>RGFA (Ruby)</li>
<li>Gfapy (python)</li>
</ul>
<p>GfaViz</p>
<ul>
<li>visualization of GFA2 (and GFA1)</li>
<li>C++, QT, OGDF</li>
<li>GUI and CLI</li>
<li>Two layouts, many options to customize views</li>
</ul>
<p>Scaffolding graphs</p>
<ul>
<li>dealing with pos gaps (missing seq) and neg gaps (contig overlap by repeats)</li>
<li>Show how it looks with Bandage, no gap info</li>
<li>much busier with GfaViz
<ul>
<li>but now can see gap info between the pieces</li>
</ul></li>
</ul>
<p>Long reads</p>
<ul>
<li>local alignments are messy (noisy data)</li>
<li>GFA2 has <em>internal</em> alignments</li>
</ul>
<p>Release is <em>later this year</em></p>
<ul>
<li>ggonnella on github</li>
<li>email: <a href="mailto:gonnella@zbh.uni-hamburg.de">gonnella@zbh.uni-hamburg.de</a></li>
</ul>
</div>
<div id="luke-zappia-oshlack" class="section level1">
<h1>Luke Zappia (Oshlack?)</h1>
<p>Using clustering trees to visualize scRNA-seq data</p>
<p>(caps my own) <strong>SO MANY CLUSTERING TOOLS</strong></p>
<p><strong>T-SNE IS NOT CLUSTERING</strong></p>
<p>How do we decide how many clusters?</p>
<p>When deciding k (cluster num) you can think of a graph and with edge weights to assess interesting groups</p>
<p>Cells on edge / num of cells in high res cluster</p>
<p>Above equation good way to think about usefulness of cluster groups</p>
<p>As we increase k, we can see how the graph changes and can get a sense of whether k should be changed</p>
<p>Real data</p>
<ul>
<li>used Seurat to cluster</li>
<li>see a branch with is distinct and doesn’t interact with anything else</li>
<li>also see a stable region</li>
<li>see low proportion edges if you really increase the k</li>
</ul>
<p><strong>I can’t doodle on the computer, which would be helpful here but this stuff looks really helpful</strong></p>
<p><strong>Using edge porportion (above eq.) and cluster relatinoships you get a decent sense of whether the clustering makes sense</strong></p>
<p><strong>WHAT SO COOL</strong> (overlap of cluster trees on t-SNE)</p>
<p>Q: could this be used to magic pick a k? A: Everyone asks this….please don’t….no idea what would happen</p>
</div>
<div id="laura-huerta-papatheodorou" class="section level1">
<h1>Laura Huerta (Papatheodorou)</h1>
<p>Data curation integration and visualization</p>
<p><strong>Oh boy, a viz talk that I’m going to cover with words</strong></p>
<p>Expression Atlas</p>
<ul>
<li>open science resource for holding expression data</li>
<li><strong>super useful resource, unfortunately in my experience pain to get data into R with their R packages</strong>
<ul>
<li><strong>I find it easier to just find the tsv output link and then slurp that into R</strong></li>
</ul></li>
<li><strong>big value in consistent computational workflows</strong></li>
<li><code>&gt;3,300</code> datasets, from all the big consortia (GTEx, FANTOM, etc)</li>
</ul>
<p>clusterSeq package</p>
<ul>
<li><strong>missed what this does but looks like a good thing to check out…</strong></li>
<li>from bioconductor page: <em>Identification of clusters of co-expressed genes based on their expression across multiple (replicated) biological samples.</em></li>
</ul>
<p>Can embed Expression Atlas data on other pages (ensembl does this)</p>
<p>scRNA-seq data also - again public data, consistent processing (extension of iRAP) - can see whether gene is considered a <em>marker</em> gene (<strong>not sure how this is picked</strong>) - have a t-SNE(?) interactive plot view and shows your favorite gene colored</p>
<p>Future:</p>
<ul>
<li>merge bulk and scRNA data</li>
<li>….something else I missed about metadata?</li>
</ul>
<p>Q: How is data updated with genome build change? A: In lock step, can view old versions</p>
</div>
<div id="casey-greene-greene" class="section level1">
<h1>Casey Greene (Greene)</h1>
<p>Can “big data” help us tackle rare diseases?</p>
<p>~3.7 million assays / datasets</p>
<p>~3.8 billion USD</p>
<p>Really tough to compare <em>across</em> datasets with the “modular” approach</p>
<p>PLIER: decompose dataset into latent variables by genes and sample:</p>
<ul>
<li>Mao et al biorxiv 2017</li>
</ul>
<p>Latent variables in individual datasets not helpful for dataset comparison - but could you run PLIER on datasetS and see whether it “works” - with SLE (lupus)….yes - with only 7 datasets, can find vars that are NOT dataset specific</p>
<p>Can you learn patterns from large datasets then <em>transfer</em> to individual dataset / problem?</p>
<ul>
<li><a href="bit.ly/MultiPLIER">MultiPLIER</a></li>
<li>ran with recount2 data on SLE (looking at ANC-associated)</li>
<li>top loadings known info</li>
<li>see how loading score drops in other (non SLE) tissues</li>
<li>slides: <a href="https://www.dropbox.com/sh/j0ysaojqmac29qv/AADBHy-3y8qzWHr34F7M39QIa?dl=0" class="uri">https://www.dropbox.com/sh/j0ysaojqmac29qv/AADBHy-3y8qzWHr34F7M39QIa?dl=0</a></li>
</ul>
<p>Pitch for <a href="http://researchparasite.com" class="uri">http://researchparasite.com</a></p>
</div>
<div id="sergei-yakneen" class="section level1">
<h1>Sergei Yakneen</h1>
<p>Butler: a framework for large-scale scientific analysis on the cloud</p>
<p>[github.com/llevar/butler]</p>
<p>PCAWG</p>
<ul>
<li>pan cancer analysis of whole genomes</li>
<li>2,834 donors, 70,313 files, 729 TB</li>
</ul>
<p>Show graph scaling off as pipelines choke as more samples get added</p>
<p>If progress was linear, <em>years</em> of compute (?) time could be saved</p>
<p>Key needs:</p>
<ul>
<li>provisioning</li>
<li>config management</li>
<li>workflow</li>
<li>operations management</li>
</ul>
<p>Tried to use “off the shelf” infrastructure / processes to build Butler</p>
<p>Workflows are … CWL?</p>
<p><strong>This seems targeted towards a pretty small audience - people running huge compute platforms?</strong></p>
<p>** I guess you could use this for a smaller projects and deploy to cloud, but I would think that the overhead of learning this would likely overwhelm the potential savings relative to using Snakemake / Docker / Conda. **</p>
</div>

    </div>

    


<div class="article-tags">
  
  <a class="btn btn-primary btn-outline" href="/./tags/conference/">conference</a>
  
  <a class="btn btn-primary btn-outline" href="/./tags/gi2018/">GI2018</a>
  
  <a class="btn btn-primary btn-outline" href="/./tags/bioinformatics/">bioinformatics</a>
  
  <a class="btn btn-primary btn-outline" href="/./tags/talks/">talks</a>
  
</div>




    
    
    <div class="article-widget">
      <div class="hr-light"></div>
      <h3>Related</h3>
      <ul>
        
        <li><a href="/post/bog18-notes/">#BoG18: Talk Notes</a></li>
        
        <li><a href="/post/quick-guide-to-gene-name-conversion/">Quick Guide to Gene Name Conversion</a></li>
        
        <li><a href="/post/simple-heatmaps-with-complexheatmaps/">Let&#39;s Plot 6: Simple guide to heatmaps with ComplexHeatmaps</a></li>
        
        <li><a href="/post/easy-bam-downsampling/">Easy bam downsampling</a></li>
        
        <li><a href="/post/are-you-in-genomics-stop-using-roc-use-pr/">Are you in genomics and building models? Stop using ROC - use PR</a></li>
        
      </ul>
    </div>
    

    

    
<section id="comments">
  <div id="disqus_thread"></div>
<script>
    var disqus_config = function () {
    
    
    
    };
    (function() {
        if (["localhost", "127.0.0.1"].indexOf(window.location.hostname) != -1) {
            document.getElementById('disqus_thread').innerHTML = 'Disqus comments not available by default when the website is previewed locally.';
            return;
        }
        var d = document, s = d.createElement('script'); s.async = true;
        s.src = '//' + "davemcg-gitub-io" + '.disqus.com/embed.js';
        s.setAttribute('data-timestamp', +new Date());
        (d.head || d.body).appendChild(s);
    })();
</script>
<noscript>Please enable JavaScript to view the <a href="https://disqus.com/?ref_noscript">comments powered by Disqus.</a></noscript>
<a href="https://disqus.com" class="dsq-brlink">comments powered by <span class="logo-disqus">Disqus</span></a>
</section>



  </div>
</article>

<footer class="site-footer">
  <div class="container">
    <p class="powered-by">

      &copy; 2018 &middot; 

      Powered by the
      <a href="https://sourcethemes.com/academic/" target="_blank" rel="noopener">Academic theme</a> for
      <a href="https://gohugo.io" target="_blank" rel="noopener">Hugo</a>.

      <span class="pull-right" aria-hidden="true">
        <a href="#" id="back_to_top">
          <span class="button_icon">
            <i class="fa fa-chevron-up fa-2x"></i>
          </span>
        </a>
      </span>

    </p>
  </div>
</footer>


<div id="modal" class="modal fade" role="dialog">
  <div class="modal-dialog">
    <div class="modal-content">
      <div class="modal-header">
        <button type="button" class="close btn-large" data-dismiss="modal">&times;</button>
        <h4 class="modal-title">Cite</h4>
      </div>
      <div>
        <pre><code class="modal-body tex"></code></pre>
      </div>
      <div class="modal-footer">
        <a class="btn btn-primary btn-outline js-copy-cite" href="#" target="_blank">
          <i class="fa fa-copy"></i> Copy
        </a>
        <a class="btn btn-primary btn-outline js-download-cite" href="#" target="_blank">
          <i class="fa fa-download"></i> Download
        </a>
        <div id="modal-error"></div>
      </div>
    </div>
  </div>
</div>

    

    
    
    <script id="dsq-count-scr" src="//davemcg-gitub-io.disqus.com/count.js" async></script>
    

    

    
    <script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.2.1/jquery.min.js" integrity="sha512-3P8rXCuGJdNZOnUx/03c1jOTnMn3rP63nBip5gOP2qmUh5YAdVAvFZ1E+QLZZbC1rtMrQb+mah3AfYW11RUrWA==" crossorigin="anonymous"></script>
    <script src="https://cdnjs.cloudflare.com/ajax/libs/jquery.imagesloaded/4.1.3/imagesloaded.pkgd.min.js" integrity="sha512-umsR78NN0D23AzgoZ11K7raBD+R6hqKojyBZs1w8WvYlsI+QuKRGBx3LFCwhatzBunCjDuJpDHwxD13sLMbpRA==" crossorigin="anonymous"></script>
    <script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.3.7/js/bootstrap.min.js" integrity="sha512-iztkobsvnjKfAtTNdHkGVjAYTrrtlC7mGp/54c40wowO7LhURYl3gVzzcEqGl/qKXQltJ2HwMrdLcNUdo+N/RQ==" crossorigin="anonymous"></script>
    <script src="https://cdnjs.cloudflare.com/ajax/libs/jquery.isotope/3.0.4/isotope.pkgd.min.js" integrity="sha512-VDBOIlDbuC4VWxGJNmuFRQ0Li0SKkDpmGyuhAG5LTDLd/dJ/S0WMVxriR2Y+CyPL5gzjpN4f/6iqWVBJlht0tQ==" crossorigin="anonymous"></script>
    <script src="https://cdnjs.cloudflare.com/ajax/libs/fancybox/3.2.5/jquery.fancybox.min.js" integrity="sha256-X5PoE3KU5l+JcX+w09p/wHl9AzK333C4hJ2I9S5mD4M=" crossorigin="anonymous"></script>
    
    
    <script src="/js/hugo-academic.js"></script>
    

    
    
      
      
      <script src="//cdnjs.cloudflare.com/ajax/libs/highlight.js/9.12.0/highlight.min.js" integrity="sha256-/BfiIkHlHoVihZdc6TFuj7MmJ0TWcWsMXkeDFwhi0zw=" crossorigin="anonymous"></script>
      

      

      

      <script>hljs.initHighlightingOnLoad();</script>
    

    
    
    <script type="text/x-mathjax-config">
        MathJax.Hub.Config({ tex2jax: { inlineMath: [['$','$'], ['\\(','\\)']] } });
    </script>
    
    <script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.1/MathJax.js?config=TeX-AMS_CHTML" integrity="sha512-tOav5w1OjvsSJzePRtt2uQPFwBoHt1VZcUq8l8nm5284LEKE9FSJBQryzMBzHxY5P0zRdNqEcpLIRVYFNgu1jw==" crossorigin="anonymous"></script>
    
    

  </body>
</html>

