<!DOCTYPE html>
<html lang="en-us">
<head>

  <meta charset="utf-8">
  <meta name="viewport" content="width=device-width, initial-scale=1">
  <meta http-equiv="X-UA-Compatible" content="IE=edge">
  <meta name="theme" content="hugo-academic">
  <meta name="generator" content="Hugo 0.35" />
  <meta name="author" content="David McGaughey">

  
  
  
  
    
      
    
  
  <meta name="description" content="Day Two Personal and Medical Genomics Katie Pollard Keynote Sri Kosuri (Kosuri) Patrick Brennan (Nationwide Children’s) Kaitlin Samocha (Hurles) Tracy Ballinger (Semple) Lucia Spangenberg (Naya)  Comparative, Evolutionary, Metagenomics Ellen Leffler (Kwiatkowski) Luca Penso-Dolfin (Di Palma) Mario Caccamo (NIAB) Carla Cummins (Flicek)    Day Two  Personal and Medical Genomics Very sparse and poorly written notes covering #GI2018.
Typos everywhere. Things may change dramatically over time as I scan back through notes.">

  
  <link rel="alternate" hreflang="en-us" href="/./post/gi2018-day-two/">

  


  

  
  
  <meta name="theme-color" content="#0095eb">
  
  
  
  
    
  
  
    
    
      
        <link rel="stylesheet" href="//cdnjs.cloudflare.com/ajax/libs/highlight.js/9.12.0/styles/github.min.css">
      
    
  
  
  <link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.3.7/css/bootstrap.min.css" integrity="sha512-6MXa8B6uaO18Hid6blRMetEIoPqHf7Ux1tnyIQdpt9qI5OACx7C+O3IVTr98vwGnlcg0LOLa02i9Y1HpVhlfiw==" crossorigin="anonymous">
  <link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/academicons/1.8.6/css/academicons.min.css" integrity="sha256-uFVgMKfistnJAfoCUQigIl+JfUaP47GrRKjf6CTPVmw=" crossorigin="anonymous">
  <link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/4.7.0/css/font-awesome.min.css" integrity="sha512-SfTiTlX6kk+qitfevl/7LibUOeJWlt9rbyDn92a1DqWOw9vWG2MFoays0sgObmWazO5BQPiFucnnEAjpAB+/Sw==" crossorigin="anonymous">
  <link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/fancybox/3.2.5/jquery.fancybox.min.css" integrity="sha256-ygkqlh3CYSUri3LhQxzdcm0n1EQvH2Y+U5S2idbLtxs=" crossorigin="anonymous">
  
  
  
  
  <link rel="stylesheet" href="//fonts.googleapis.com/css?family=Montserrat:400,700%7cRoboto:400,400italic,700%7cRoboto&#43;Mono">
  
  <link rel="stylesheet" href="/styles.css">
  

  
    <script>
      window.ga=window.ga||function(){(ga.q=ga.q||[]).push(arguments)};ga.l=+new Date;
      ga('create', 'UA-113461893-1', 'auto');
      
      ga('require', 'eventTracker');
      ga('require', 'outboundLinkTracker');
      ga('require', 'urlChangeTracker');
      ga('send', 'pageview');
    </script>
    <script async src="//www.google-analytics.com/analytics.js"></script>
    
    <script async src="https://cdnjs.cloudflare.com/ajax/libs/autotrack/2.4.1/autotrack.js" integrity="sha512-HUmooslVKj4m6OBu0OgzjXXr+QuFYy/k7eLI5jdeEy/F4RSgMn6XRWRGkFi5IFaFgy7uFTkegp3Z0XnJf3Jq+g==" crossorigin="anonymous"></script>
    
  

  
  <link rel="alternate" href="/./index.xml" type="application/rss+xml" title="eye Bioinformatician">
  <link rel="feed" href="/./index.xml" type="application/rss+xml" title="eye Bioinformatician">
  

  <link rel="manifest" href="/site.webmanifest">
  <link rel="icon" type="image/png" href="/img/icon.png">
  <link rel="apple-touch-icon" type="image/png" href="/img/icon-192.png">

  <link rel="canonical" href="/./post/gi2018-day-two/">

  <meta property="twitter:card" content="summary_large_image">
  
  <meta property="twitter:site" content="@https://twitter.com/David_McGaughey">
  <meta property="twitter:creator" content="@https://twitter.com/David_McGaughey">
  
  <meta property="og:site_name" content="eye Bioinformatician">
  <meta property="og:url" content="/./post/gi2018-day-two/">
  <meta property="og:title" content="#GI2018 - Day Two | eye Bioinformatician">
  <meta property="og:description" content="Day Two Personal and Medical Genomics Katie Pollard Keynote Sri Kosuri (Kosuri) Patrick Brennan (Nationwide Children’s) Kaitlin Samocha (Hurles) Tracy Ballinger (Semple) Lucia Spangenberg (Naya)  Comparative, Evolutionary, Metagenomics Ellen Leffler (Kwiatkowski) Luca Penso-Dolfin (Di Palma) Mario Caccamo (NIAB) Carla Cummins (Flicek)    Day Two  Personal and Medical Genomics Very sparse and poorly written notes covering #GI2018.
Typos everywhere. Things may change dramatically over time as I scan back through notes.">
  <meta property="og:locale" content="en-us">
  
  <meta property="article:published_time" content="2018-09-18T00:00:00&#43;00:00">
  
  <meta property="article:modified_time" content="2018-09-18T00:00:00&#43;00:00">
  

  
  

  <title>#GI2018 - Day Two | eye Bioinformatician</title>

</head>
<body id="top" data-spy="scroll" data-target="#toc" data-offset="71" >

<nav class="navbar navbar-default navbar-fixed-top" id="navbar-main">
  <div class="container">

    
    <div class="navbar-header">
      
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse"
              data-target=".navbar-collapse" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      
      <a class="navbar-brand" href="/">eye Bioinformatician</a>
    </div>

    
    <div class="collapse navbar-collapse">

      
      
      <ul class="nav navbar-nav navbar-right">
        

        
        
        
        
        
          
        

        <li class="nav-item">
          <a href="/#about">
            
            <span>Home</span>
            
          </a>
        </li>

        
        

        
        <li class="dropdown">
          <a href="#" class="dropdown-toggle" data-toggle="dropdown" aria-haspopup="true">
            
            <span>Posts</span>
            
            <span class="caret"></span>
          </a>
          <ul class="dropdown-menu">
            
            <li class="nav-item">
              <a href="/#posts">
                
                <span>Recent Posts</span>
                
              </a>
            </li>
            
            <li class="nav-item">
              <a href="/post">
                
                <span>All Posts</span>
                
              </a>
            </li>
            
          </ul>
        </li>

        
        

        
        
        
        
        
          
        

        <li class="nav-item">
          <a href="/#projects">
            
            <span>Projects</span>
            
          </a>
        </li>

        
        

        
        
        
        
        
          
        

        <li class="nav-item">
          <a href="/#culture">
            
            <span>Culture</span>
            
          </a>
        </li>

        
        

        
        
        
        
        
          
        

        <li class="nav-item">
          <a href="/#publications_selected">
            
            <span>Publications</span>
            
          </a>
        </li>

        
        
      

      
      </ul>

    </div>
  </div>
</nav>


<article class="article" itemscope itemtype="http://schema.org/Article">

  


  <div class="article-container">
    <h1 itemprop="name">#GI2018 - Day Two</h1>

    

<div class="article-metadata">

  <span class="article-date">
    
    <time datetime="2018-09-18 00:00:00 &#43;0000 UTC" itemprop="datePublished dateModified">
      Sep 18, 2018
    </time>
  </span>
  <span itemscope itemprop="author publisher" itemtype="http://schema.org/Person">
    <meta itemprop="name" content="David McGaughey">
  </span>

  
  <span class="middot-divider"></span>
  <span class="article-reading-time">
    8 min read
  </span>
  

  
  
  <span class="middot-divider"></span>
  <a href="/./post/gi2018-day-two/#disqus_thread"></a>
  

  
  
  
  <span class="middot-divider"></span>
  <span class="article-categories">
    <i class="fa fa-folder"></i>
    
    <a href="/./categories/gi2018/">GI2018</a
    >, 
    
    <a href="/./categories/bioinformatics/">bioinformatics</a
    >
    
  </span>
  
  

  
  
<div class="share-box" aria-hidden="true">
  <ul class="share">
    <li>
      <a class="twitter"
         href="https://twitter.com/intent/tweet?text=%23GI2018%20-%20Day%20Two&amp;url=%2f.%2fpost%2fgi2018-day-two%2f"
         target="_blank" rel="noopener">
        <i class="fa fa-twitter"></i>
      </a>
    </li>
    <li>
      <a class="facebook"
         href="https://www.facebook.com/sharer.php?u=%2f.%2fpost%2fgi2018-day-two%2f"
         target="_blank" rel="noopener">
        <i class="fa fa-facebook"></i>
      </a>
    </li>
    <li>
      <a class="linkedin"
         href="https://www.linkedin.com/shareArticle?mini=true&amp;url=%2f.%2fpost%2fgi2018-day-two%2f&amp;title=%23GI2018%20-%20Day%20Two"
         target="_blank" rel="noopener">
        <i class="fa fa-linkedin"></i>
      </a>
    </li>
    <li>
      <a class="weibo"
         href="http://service.weibo.com/share/share.php?url=%2f.%2fpost%2fgi2018-day-two%2f&amp;title=%23GI2018%20-%20Day%20Two"
         target="_blank" rel="noopener">
        <i class="fa fa-weibo"></i>
      </a>
    </li>
    <li>
      <a class="email"
         href="mailto:?subject=%23GI2018%20-%20Day%20Two&amp;body=%2f.%2fpost%2fgi2018-day-two%2f">
        <i class="fa fa-envelope"></i>
      </a>
    </li>
  </ul>
</div>


  

</div>


    <div class="article-style" itemprop="articleBody">
      <div id="TOC">
<ul>
<li><a href="#day-two">Day Two</a></li>
<li><a href="#personal-and-medical-genomics">Personal and Medical Genomics</a><ul>
<li><a href="#katie-pollard-keynote">Katie Pollard Keynote</a></li>
<li><a href="#sri-kosuri-kosuri">Sri Kosuri (Kosuri)</a></li>
<li><a href="#patrick-brennan-nationwide-childrens">Patrick Brennan (Nationwide Children’s)</a></li>
<li><a href="#kaitlin-samocha-hurles">Kaitlin Samocha (Hurles)</a></li>
<li><a href="#tracy-ballinger-semple">Tracy Ballinger (Semple)</a></li>
<li><a href="#lucia-spangenberg-naya">Lucia Spangenberg (Naya)</a></li>
</ul></li>
<li><a href="#comparative-evolutionary-metagenomics">Comparative, Evolutionary, Metagenomics</a><ul>
<li><a href="#ellen-leffler-kwiatkowski">Ellen Leffler (Kwiatkowski)</a></li>
<li><a href="#luca-penso-dolfin-di-palma">Luca Penso-Dolfin (Di Palma)</a></li>
<li><a href="#mario-caccamo-niab">Mario Caccamo (NIAB)</a></li>
<li><a href="#carla-cummins-flicek">Carla Cummins (Flicek)</a></li>
</ul></li>
</ul>
</div>

<div id="day-two" class="section level1">
<h1>Day Two</h1>
</div>
<div id="personal-and-medical-genomics" class="section level1">
<h1>Personal and Medical Genomics</h1>
<p>Very sparse and poorly written notes covering <a href="https://twitter.com/hashtag/GI2018?src=hash">#GI2018</a>.</p>
<p>Typos everywhere. Things may change dramatically over time as I scan back through notes.</p>
<p>I’ve tried to respect #notwitter. Will be updated periodically.</p>
<p>Speaker (Lab | Group)</p>
<div id="katie-pollard-keynote" class="section level2">
<h2>Katie Pollard Keynote</h2>
<p>A population genetic view of human chromatin organization</p>
<p>Noncoding human mutation and disease</p>
<ul>
<li>which are causal?</li>
<li>how do they alter gene expression?</li>
<li>what genes / pathways do they affect?</li>
</ul>
<p>core idea:</p>
<ul>
<li>motif disruption (i.e. mutation alter binding site)</li>
</ul>
<p>but, how about we think about genome as structure?</p>
<ul>
<li>if structure (3D folding) is functional, any variation that modifies that would be deleterious</li>
</ul>
<p>chromatin structure and pop genetics</p>
<p>discussion of TADs / HiC / BE (boundary elements)</p>
<p>compare rare variants in patients, variants in healthy humans, fixed diffs between primates</p>
<ul>
<li>deletions depleted at boundary elements (strong negative selection)</li>
<li>also holds up across non-primates</li>
<li>less depletion in disease cohort (autism) variants</li>
</ul>
<p>“suggestion broad role for enhancer hijacking in disease”</p>
<p>noncoding variation assessment should be TAD aware</p>
<p>try to predict how mutations change TAD/BE</p>
<ul>
<li>kind of working (not as “sharp” as the real data)</li>
</ul>
<p>chromatin and genetic structure (LD) both have similar struture</p>
<ul>
<li>are they correlated?</li>
<li>not the same sizes (LD blocks shorter)</li>
<li>not correlated at 5kb to 2Mb</li>
</ul>
<p>chromatin interactions (HiC) more enriched for (compared to closest gene or LD):</p>
<ul>
<li>eQTL</li>
<li>GO enrichment terms</li>
<li>“most distal noncoding variants do not target closest expressed gene”</li>
</ul>
<p>sequence motifs fail to explain key aspects of protein-DNA binding</p>
<ul>
<li>30% of top 2000 ChIP-seq peaks have no sequence motif</li>
<li>many DNA binding proteins have very similar motifs but have disparate binding</li>
<li>seq motifs don’t capture DNA shape well (major / minor groove)</li>
</ul>
<p>hypothesis: “DNA binding proteins recognize DNA shape beyond seq. motifs”</p>
<ul>
<li>ShapeMF (<a href="https://www.bioconductor.org/packages/release/bioc/html/DNAshapeR.html">DNAshape R package</a>)</li>
<li>compare to classic motif with ENCODE ChIP-seq and HT-SELEX</li>
</ul>
<p>shape motifs match up between ENCODE and HT-SELEX</p>
<p>shape motifs can occur without sequence motifs</p>
<ul>
<li>most have both shape and motif patterns</li>
<li>some shape only and some motif only</li>
</ul>
<p>Machine learning in genomics comments (Pollard kindly taking time out her talk to discuss geneeral ML issues)</p>
<ul>
<li>try to model enhancer interaction with konwn enhancers, expressed genes, Hi-C data (TargetFinder)</li>
<li>common issues:
<ul>
<li>balanced training data (genomic data <a href="http://davemcg.github.io/post/are-you-in-genomics-stop-using-roc-use-pr/">usually isn’t</a>)</li>
<li>select features from whole dataset (feature selection should be <em>inside</em> cross validation)</li>
<li>use biologically appropriate examples</li>
<li>use the right statistic (PR, ROC, etc?)</li>
<li>most ML models assume observations are <em>independent</em> and <em>identically distributed</em></li>
<li>genomic data is inherently <strong>not</strong> independent and identically distributed</li>
<li>not trivial….different genome regions are inherently different (model doesn’t transfer)</li>
</ul></li>
</ul>
</div>
<div id="sri-kosuri-kosuri" class="section level2">
<h2>Sri Kosuri (Kosuri)</h2>
<p>Functional Impacts Of Rare Genetic Variation On Exome Recognition</p>
<p>Experimental assays (but huge)</p>
<p>multiplexed reporter assays (100k+ experiments) + DNA synthesis to do reverse genomics</p>
<p><a href="https://www.biorxiv.org/content/early/2018/03/10/199927" class="uri">https://www.biorxiv.org/content/early/2018/03/10/199927</a></p>
<p>Average genome has 4-5 million common variants and 50-100K rare variants (&lt;0.5% AF)</p>
<p>how do rare variants exert large effect changes?</p>
<ul>
<li>you see them in GTEx (under and over expression outliers)</li>
</ul>
<p>Massively parallel exon splicing reporter</p>
<ul>
<li>which mutations cause exon splicing issues</li>
<li>assayed 28k across 2,200 exon</li>
<li>mostly at splice junctions</li>
<li>but you still see plenty in exons and further away</li>
<li><em>in silico</em> predictions are pretty bad (splicing specific models OK)</li>
<li>3.8% (1,505 / 27,900) led to large loss of exon recognition</li>
<li>can’t predict them…. (with existing tools)</li>
<li>nice to see a whole slide on why this isn’t the most perfect experiment ever
<ul>
<li>short (100nt) exons used (biased)</li>
</ul></li>
</ul>
</div>
<div id="patrick-brennan-nationwide-childrens" class="section level2">
<h2>Patrick Brennan (Nationwide Children’s)</h2>
<p>Integration of whole genome, whole exome, and transcriptome pipeline for comprehensive genomic profiling of 60 pediatric cancer subjects</p>
<p>250x WES, 60X WGS (tumor + normal) RNA-seq (tumor only)</p>
<p>7 diff fusion detection algorithms</p>
<ul>
<li>ugh, 7 way venn</li>
<li>they focus on the few which are consistently called (or rather ranked on overlap and evidence support)</li>
</ul>
<p>RNA-seq</p>
<ul>
<li>DESeq, Salmon, then Ingenuity for pathway analysis (TPM as input)</li>
</ul>
<p>Differential comparison (RNA)</p>
<ul>
<li>but to what???</li>
<li>oh the ‘treehouse’ data (not sure what this is)</li>
</ul>
<p>Case study matching indel in gene with RNA expression difference, which helped to suggest tyrosine kinase inhibitor treatment</p>
<p>Case study IDing fusion gene between MET and RBPMS (chr 7 &lt;-&gt; chr 8)</p>
<p>Keep using ‘Churchill’ term for their software….annoying they don’t credit the tools that actually power it (GATK?)</p>
</div>
<div id="kaitlin-samocha-hurles" class="section level2">
<h2>Kaitlin Samocha (Hurles)</h2>
<p>Evaluating the role of rare variation in children with developmental disorders</p>
<p>DDD</p>
<ul>
<li>represent UK well</li>
<li>diagnostic rate of 35%</li>
</ul>
<p>4,300 trios have 94 genes with high gene burden</p>
<ul>
<li>gene test with poisson tests outputting p values for LoF, missense, synonymous</li>
</ul>
<p>New approach</p>
<ul>
<li>weight different types of mutations separately</li>
<li>gene score composed of weighted scores</li>
<li>multiple hypothesis correction (use prior knowledge)</li>
</ul>
<p>where do weights come from?</p>
<ul>
<li>observations from DDD cohort, comparing DDD enrichment by class (LoF, etc) to CADD</li>
</ul>
<p>new method</p>
<ul>
<li>finds more genes relative to poisson (old) method</li>
<li>applying to 10k trios</li>
</ul>
<p>inherited variation</p>
<ul>
<li>do cases have more rare damaging variants</li>
<li>cadd &gt; 21</li>
<li>observed once or more ExAC</li>
<li>effect size of ~1.7 for probands with affect parents</li>
<li>~1.2 for other probands</li>
</ul>
<p>causative enriched in high pLI genes</p>
<p>is this from cryptic recessive diagnoses?</p>
<ul>
<li>probably not</li>
<li>high pLI genes not enriched for recessive disease genes</li>
</ul>
<p>digenic inheritance?</p>
<ul>
<li>parents each have damaging mutations (but dif from each other)</li>
<li>probably not (no strong enrichment in undiagnosed vs diagnosed)</li>
</ul>
</div>
<div id="tracy-ballinger-semple" class="section level2">
<h2>Tracy Ballinger (Semple)</h2>
<p>Modeling Double Strand Breaks to Interrogate Structural Variation in Cancer</p>
<p>SV (structural variation) prevalent across cancer</p>
<p>Nonrandom distribution</p>
<ul>
<li>chromatin structure, sequence composition, and nuclear organization influence SV</li>
</ul>
<p>built Random Forest model that correlates well with real data</p>
<ul>
<li>Replication seq, open chromatin (DNase), pol2db, most important features</li>
<li>predicts well across different cell types</li>
</ul>
<p>Compare SV prediction to cancer breakpoints</p>
<ul>
<li>ID regions with more or less SV breakpoints than expected</li>
<li>also “passenger” regions which are not enriched</li>
</ul>
<p>cancer ‘coldspots’:</p>
<ul>
<li>enriched for active chromatin and deplted in fragile sites</li>
</ul>
</div>
<div id="lucia-spangenberg-naya" class="section level2">
<h2>Lucia Spangenberg (Naya)</h2>
<p>Retrieving Charras genomic tracts from the Uruguayan admixed population</p>
<p>Charras all killed by state 187 years ago</p>
<p>80 Uruguayan genomes (Just 10 for this project?)</p>
<p>Q1: Still have Charrua in Uruguayan genomes?</p>
<p>Calculated local ancestry (rel to EUR or AFR)</p>
<p>PCA to see clustering of Native Uruguayan. Also had 20 genomes from Simons</p>
<p>Want to calculate Fst Tree with other ancient/original ameican populations</p>
<ul>
<li>clusters with Guarani and Chane (expected!)</li>
</ul>
<p>Build protogenome by merging the 10 WGS they did</p>
<ul>
<li>99% covered</li>
</ul>
</div>
</div>
<div id="comparative-evolutionary-metagenomics" class="section level1">
<h1>Comparative, Evolutionary, Metagenomics</h1>
<div id="ellen-leffler-kwiatkowski" class="section level2">
<h2>Ellen Leffler (Kwiatkowski)</h2>
<p>Paired sequencing of host and parasite genomes in severe malaria cases</p>
<p>191 million cases / year!</p>
<p>plasmodium falciparum (most commmon in Africa) and vivax (more common in not Africa)</p>
<p>MalariaGEN</p>
<ul>
<li>GWAS for severe malaria cases (falciparum), matched population controls</li>
<li>found new structural var in GYPA/B/E</li>
</ul>
<p>Big Questions:</p>
<ul>
<li>parasite variation?</li>
<li>genotype &lt;-&gt; parasite susceptibilty?</li>
<li>co-evolution?</li>
<li>individual disease risk?</li>
</ul>
<p>can we sequence parasite DNA from human blood samples?</p>
<ul>
<li>1 - 5 million parasites per ul blood</li>
<li>napkin math gives 80X+ coverage for over 10,000 parasites / ul</li>
</ul>
<p>align WGS to human and parasite genome</p>
<ul>
<li>mean 197X coverage</li>
<li>eyeball median of 100X?</li>
</ul>
<p>selective amplification of plasmodium falciparum gneome which gives ~50X enrichment</p>
<p>1.5 million SNPs total (parasite) 10 million SNPs total (human)</p>
<p>host-parasite association (looking at merozoite SNPs, which interacts with human)</p>
<ul>
<li>get one nice looking peak in parasite gene MSP4 / MSP2 and human GCTN2 and GBGT1</li>
</ul>
</div>
<div id="luca-penso-dolfin-di-palma" class="section level2">
<h2>Luca Penso-Dolfin (Di Palma)</h2>
<p>The role of structural variants in the adaptive radiation of African Cichlids</p>
<p>African cichclids are canonical example of adaptive radition</p>
<p>Also, are freshwater fish, if you didn’t know</p>
<p>Hundreds of species, variation in jaw morphology, color, vision, behavior</p>
<p>Most studies focus on gene evolution</p>
<p>Looking at structural variation across three great East African lakes</p>
<p>M. zebra, P. nyererei, A. burtoni, N. brichardi (already sequenced)</p>
<p>One outgroup (O. niloticus)</p>
<p>Talking about SV identification using paired end reads and looking for internal distance and split reads</p>
<ul>
<li><strong>Is he rolling his own SV caller???</strong></li>
</ul>
<p>FIRST NO_TWITTER!</p>
<ul>
<li>SV pipeline….which uses existing SV tools….not certain why NO_TWITTER…but I won’t give any details</li>
</ul>
<p>duplication net gain of 9.2 events per million years across the four species</p>
<p>lot fewer inversions</p>
<ul>
<li>about 0.5 events per million years</li>
</ul>
<p>SECOND NO_TWITTER</p>
<ul>
<li>cool enrichment of some genes / GO terms in inversions</li>
</ul>
</div>
<div id="mario-caccamo-niab" class="section level2">
<h2>Mario Caccamo (NIAB)</h2>
<p>Understanding the genetic components controlling apomixis</p>
<p><strong>WHAT IS APOMIXIS I HAVE NO IDEA</strong></p>
<p><strong>SOMETHING WITH CROPS?</strong></p>
<p>Plants often have polyploid genomes</p>
<p>Wheat is hexaploid …. and 17Gb (reminder human is 3Gb)</p>
<p>There are 60k wheat lines. (<strong>Whoah</strong>)</p>
<p>Apomixis!</p>
<ul>
<li>clonal reproduction through seeds</li>
<li>asexual reproduction</li>
</ul>
<p>Weeping lovegrass</p>
<ul>
<li>grass from southern africa</li>
<li>polyploid</li>
<li>transcriptome from sexual and asexual
<ul>
<li>4 candidate genes</li>
</ul></li>
</ul>
<p>Make diploid -&gt; pacbio + dovetail -&gt; assembly</p>
<p><strong>OH NO CRAZY VENN DIAGRAM USE AN UpSet YO</strong></p>
<p>Then sequenced tetraploid and the asexual types and mapped back to assembly</p>
<p>4 candidate genes missing in 2x assembly</p>
<ul>
<li>hypothesis is that they get dropped / incompatible with sexual state</li>
<li>no mention of the genes</li>
<li>functional characteriziation work in arabidopsis and rice in progress</li>
</ul>
</div>
<div id="carla-cummins-flicek" class="section level2">
<h2>Carla Cummins (Flicek)</h2>
<p>Comparative analysis of hundreds of vertebrate genomes in Ensembl</p>
<p>What is Ensembl?</p>
<ul>
<li>genome stuffs (<strong>my summary</strong>)</li>
<li><a href="ensembl.org" class="uri">ensembl.org</a></li>
</ul>
<p>Data won’t stop coming</p>
<ul>
<li>Ensembl release 94</li>
<li>154 vertebrate genomes</li>
<li>clade based analysis</li>
<li>huge jump: 41 teleosts this release</li>
</ul>
<p>Bottlenecks:</p>
<ul>
<li>all vs all methods scale quadratically
<ul>
<li>so try to remove redundancies</li>
<li>work more within in clades</li>
</ul></li>
<li>read/write speeds
<ul>
<li>genome in FastA, indexed and read with <a href="https://github.com/Ensembl/Bio-DB-HTS">Bio::DB::HTS (htslib)</a></li>
</ul></li>
<li>multiple sequence alignment
<ul>
<li>do on server (5x faster)</li>
</ul></li>
<li>humans
<ul>
<li>automate manual tasks</li>
<li>infer species tree with Mash and NJ</li>
</ul></li>
</ul>
<p>workflows - rule based (XML) - infer parameters</p>
<p>get more algorithms on linear growth - ProgressCactus (?) - <a href="https://github.com/glennhickey/progressiveCactus">https://github.com/glennhickey/progressiveCactus ?</a></p>
</div>
</div>

    </div>

    


<div class="article-tags">
  
  <a class="btn btn-primary btn-outline" href="/./tags/conference/">conference</a>
  
  <a class="btn btn-primary btn-outline" href="/./tags/gi2018/">GI2018</a>
  
  <a class="btn btn-primary btn-outline" href="/./tags/bioinformatics/">bioinformatics</a>
  
  <a class="btn btn-primary btn-outline" href="/./tags/talks/">talks</a>
  
</div>




    
    
    <div class="article-widget">
      <div class="hr-light"></div>
      <h3>Related</h3>
      <ul>
        
        <li><a href="/post/gi2018-day-one/">#GI2018 - Day One</a></li>
        
        <li><a href="/post/bog18-notes/">#BoG18: Talk Notes</a></li>
        
        <li><a href="/post/quick-guide-to-gene-name-conversion/">Quick Guide to Gene Name Conversion</a></li>
        
        <li><a href="/post/simple-heatmaps-with-complexheatmaps/">Let&#39;s Plot 6: Simple guide to heatmaps with ComplexHeatmaps</a></li>
        
        <li><a href="/post/easy-bam-downsampling/">Easy bam downsampling</a></li>
        
      </ul>
    </div>
    

    

    
<section id="comments">
  <div id="disqus_thread"></div>
<script>
    var disqus_config = function () {
    
    
    
    };
    (function() {
        if (["localhost", "127.0.0.1"].indexOf(window.location.hostname) != -1) {
            document.getElementById('disqus_thread').innerHTML = 'Disqus comments not available by default when the website is previewed locally.';
            return;
        }
        var d = document, s = d.createElement('script'); s.async = true;
        s.src = '//' + "davemcg-gitub-io" + '.disqus.com/embed.js';
        s.setAttribute('data-timestamp', +new Date());
        (d.head || d.body).appendChild(s);
    })();
</script>
<noscript>Please enable JavaScript to view the <a href="https://disqus.com/?ref_noscript">comments powered by Disqus.</a></noscript>
<a href="https://disqus.com" class="dsq-brlink">comments powered by <span class="logo-disqus">Disqus</span></a>
</section>



  </div>
</article>

<footer class="site-footer">
  <div class="container">
    <p class="powered-by">

      &copy; 2018 &middot; 

      Powered by the
      <a href="https://sourcethemes.com/academic/" target="_blank" rel="noopener">Academic theme</a> for
      <a href="https://gohugo.io" target="_blank" rel="noopener">Hugo</a>.

      <span class="pull-right" aria-hidden="true">
        <a href="#" id="back_to_top">
          <span class="button_icon">
            <i class="fa fa-chevron-up fa-2x"></i>
          </span>
        </a>
      </span>

    </p>
  </div>
</footer>


<div id="modal" class="modal fade" role="dialog">
  <div class="modal-dialog">
    <div class="modal-content">
      <div class="modal-header">
        <button type="button" class="close btn-large" data-dismiss="modal">&times;</button>
        <h4 class="modal-title">Cite</h4>
      </div>
      <div>
        <pre><code class="modal-body tex"></code></pre>
      </div>
      <div class="modal-footer">
        <a class="btn btn-primary btn-outline js-copy-cite" href="#" target="_blank">
          <i class="fa fa-copy"></i> Copy
        </a>
        <a class="btn btn-primary btn-outline js-download-cite" href="#" target="_blank">
          <i class="fa fa-download"></i> Download
        </a>
        <div id="modal-error"></div>
      </div>
    </div>
  </div>
</div>

    

    
    
    <script id="dsq-count-scr" src="//davemcg-gitub-io.disqus.com/count.js" async></script>
    

    

    
    <script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.2.1/jquery.min.js" integrity="sha512-3P8rXCuGJdNZOnUx/03c1jOTnMn3rP63nBip5gOP2qmUh5YAdVAvFZ1E+QLZZbC1rtMrQb+mah3AfYW11RUrWA==" crossorigin="anonymous"></script>
    <script src="https://cdnjs.cloudflare.com/ajax/libs/jquery.imagesloaded/4.1.3/imagesloaded.pkgd.min.js" integrity="sha512-umsR78NN0D23AzgoZ11K7raBD+R6hqKojyBZs1w8WvYlsI+QuKRGBx3LFCwhatzBunCjDuJpDHwxD13sLMbpRA==" crossorigin="anonymous"></script>
    <script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.3.7/js/bootstrap.min.js" integrity="sha512-iztkobsvnjKfAtTNdHkGVjAYTrrtlC7mGp/54c40wowO7LhURYl3gVzzcEqGl/qKXQltJ2HwMrdLcNUdo+N/RQ==" crossorigin="anonymous"></script>
    <script src="https://cdnjs.cloudflare.com/ajax/libs/jquery.isotope/3.0.4/isotope.pkgd.min.js" integrity="sha512-VDBOIlDbuC4VWxGJNmuFRQ0Li0SKkDpmGyuhAG5LTDLd/dJ/S0WMVxriR2Y+CyPL5gzjpN4f/6iqWVBJlht0tQ==" crossorigin="anonymous"></script>
    <script src="https://cdnjs.cloudflare.com/ajax/libs/fancybox/3.2.5/jquery.fancybox.min.js" integrity="sha256-X5PoE3KU5l+JcX+w09p/wHl9AzK333C4hJ2I9S5mD4M=" crossorigin="anonymous"></script>
    
    
    <script src="/js/hugo-academic.js"></script>
    

    
    
      
      
      <script src="//cdnjs.cloudflare.com/ajax/libs/highlight.js/9.12.0/highlight.min.js" integrity="sha256-/BfiIkHlHoVihZdc6TFuj7MmJ0TWcWsMXkeDFwhi0zw=" crossorigin="anonymous"></script>
      

      

      

      <script>hljs.initHighlightingOnLoad();</script>
    

    
    
    <script type="text/x-mathjax-config">
        MathJax.Hub.Config({ tex2jax: { inlineMath: [['$','$'], ['\\(','\\)']] } });
    </script>
    
    <script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.1/MathJax.js?config=TeX-AMS_CHTML" integrity="sha512-tOav5w1OjvsSJzePRtt2uQPFwBoHt1VZcUq8l8nm5284LEKE9FSJBQryzMBzHxY5P0zRdNqEcpLIRVYFNgu1jw==" crossorigin="anonymous"></script>
    
    

  </body>
</html>

