<!DOCTYPE html>
<html lang="en-us">
<head>

  <meta charset="utf-8">
  <meta name="viewport" content="width=device-width, initial-scale=1">
  <meta http-equiv="X-UA-Compatible" content="IE=edge">
  <meta name="theme" content="hugo-academic">
  <meta name="generator" content="Hugo 0.40" />
  <meta name="author" content="David McGaughey">

  
  
  
  
    
      
    
  
  <meta name="description" content="Day 3 - Session 7 - MICROBIAL AND METAGENOMICS Strain-level metagenomic assignment and compositional estimation for long reads with MetaMaps metaFlye—Scalable long-read metagenome assembly using repeat graphs Detecting microbial transmission and engraftment after faecal microbiota transplants using long-read metagenomics and reticulatus Entropy of a bacterial stress response is a generalizable predictor for fitness and antibiotic sensitivity The use of kmer counts to train random forests to predict country of origin for bacterial pathogen sequencing data Real-time assembly using Nanopore sequencing data for microbial communities Exploring the role of ribosomal gene repeats in the context of regeneration Genomic epidemiology of West Nile virus in California   Day 3 - Session 7 - MICROBIAL AND METAGENOMICS Genome Informatics 2019 at CSHL">

  
  <link rel="alternate" hreflang="en-us" href="/./post/gi2019-day03-morning/">

  


  

  
  
  <meta name="theme-color" content="#0095eb">
  
  
  
  
    
  
  
    
    
      
        <link rel="stylesheet" href="//cdnjs.cloudflare.com/ajax/libs/highlight.js/9.12.0/styles/github.min.css">
      
    
  
  
  <link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.3.7/css/bootstrap.min.css" integrity="sha512-6MXa8B6uaO18Hid6blRMetEIoPqHf7Ux1tnyIQdpt9qI5OACx7C+O3IVTr98vwGnlcg0LOLa02i9Y1HpVhlfiw==" crossorigin="anonymous">
  <link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/academicons/1.8.6/css/academicons.min.css" integrity="sha256-uFVgMKfistnJAfoCUQigIl+JfUaP47GrRKjf6CTPVmw=" crossorigin="anonymous">
  <link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/4.7.0/css/font-awesome.min.css" integrity="sha512-SfTiTlX6kk+qitfevl/7LibUOeJWlt9rbyDn92a1DqWOw9vWG2MFoays0sgObmWazO5BQPiFucnnEAjpAB+/Sw==" crossorigin="anonymous">
  <link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/fancybox/3.2.5/jquery.fancybox.min.css" integrity="sha256-ygkqlh3CYSUri3LhQxzdcm0n1EQvH2Y+U5S2idbLtxs=" crossorigin="anonymous">
  
  
  
  
  <link rel="stylesheet" href="//fonts.googleapis.com/css?family=Montserrat:400,700%7cRoboto:400,400italic,700%7cRoboto&#43;Mono">
  
  <link rel="stylesheet" href="/styles.css">
  

  
    <script>
      window.ga=window.ga||function(){(ga.q=ga.q||[]).push(arguments)};ga.l=+new Date;
      ga('create', 'UA-113461893-1', 'auto');
      
      ga('require', 'eventTracker');
      ga('require', 'outboundLinkTracker');
      ga('require', 'urlChangeTracker');
      ga('send', 'pageview');
    </script>
    <script async src="//www.google-analytics.com/analytics.js"></script>
    
    <script async src="https://cdnjs.cloudflare.com/ajax/libs/autotrack/2.4.1/autotrack.js" integrity="sha512-HUmooslVKj4m6OBu0OgzjXXr+QuFYy/k7eLI5jdeEy/F4RSgMn6XRWRGkFi5IFaFgy7uFTkegp3Z0XnJf3Jq+g==" crossorigin="anonymous"></script>
    
  

  
  <link rel="alternate" href="/./index.xml" type="application/rss+xml" title="eye Bioinformatician">
  <link rel="feed" href="/./index.xml" type="application/rss+xml" title="eye Bioinformatician">
  

  <link rel="manifest" href="/site.webmanifest">
  <link rel="icon" type="image/png" href="/img/icon.png">
  <link rel="apple-touch-icon" type="image/png" href="/img/icon-192.png">

  <link rel="canonical" href="/./post/gi2019-day03-morning/">

  <meta property="twitter:card" content="summary_large_image">
  
  <meta property="twitter:site" content="@https://twitter.com/David_McGaughey">
  <meta property="twitter:creator" content="@https://twitter.com/David_McGaughey">
  
  <meta property="og:site_name" content="eye Bioinformatician">
  <meta property="og:url" content="/./post/gi2019-day03-morning/">
  <meta property="og:title" content="#GI2019 2019 11 08 Morning Session | eye Bioinformatician">
  <meta property="og:description" content="Day 3 - Session 7 - MICROBIAL AND METAGENOMICS Strain-level metagenomic assignment and compositional estimation for long reads with MetaMaps metaFlye—Scalable long-read metagenome assembly using repeat graphs Detecting microbial transmission and engraftment after faecal microbiota transplants using long-read metagenomics and reticulatus Entropy of a bacterial stress response is a generalizable predictor for fitness and antibiotic sensitivity The use of kmer counts to train random forests to predict country of origin for bacterial pathogen sequencing data Real-time assembly using Nanopore sequencing data for microbial communities Exploring the role of ribosomal gene repeats in the context of regeneration Genomic epidemiology of West Nile virus in California   Day 3 - Session 7 - MICROBIAL AND METAGENOMICS Genome Informatics 2019 at CSHL">
  <meta property="og:locale" content="en-us">
  
  <meta property="article:published_time" content="2019-11-08T00:00:00&#43;00:00">
  
  <meta property="article:modified_time" content="2019-11-08T00:00:00&#43;00:00">
  

  
  

  <title>#GI2019 2019 11 08 Morning Session | eye Bioinformatician</title>

</head>
<body id="top" data-spy="scroll" data-target="#toc" data-offset="71" >

<nav class="navbar navbar-default navbar-fixed-top" id="navbar-main">
  <div class="container">

    
    <div class="navbar-header">
      
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse"
              data-target=".navbar-collapse" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      
      <a class="navbar-brand" href="/">eye Bioinformatician</a>
    </div>

    
    <div class="collapse navbar-collapse">

      
      
      <ul class="nav navbar-nav navbar-right">
        

        
        
        
        
        
          
        

        <li class="nav-item">
          <a href="/#about">
            
            <span>Home</span>
            
          </a>
        </li>

        
        

        
        <li class="dropdown">
          <a href="#" class="dropdown-toggle" data-toggle="dropdown" aria-haspopup="true">
            
            <span>Posts</span>
            
            <span class="caret"></span>
          </a>
          <ul class="dropdown-menu">
            
            <li class="nav-item">
              <a href="/#posts">
                
                <span>Recent Posts</span>
                
              </a>
            </li>
            
            <li class="nav-item">
              <a href="/post">
                
                <span>All Posts</span>
                
              </a>
            </li>
            
          </ul>
        </li>

        
        

        
        
        
        
        
          
        

        <li class="nav-item">
          <a href="/#projects">
            
            <span>Projects</span>
            
          </a>
        </li>

        
        

        
        
        
        
        
          
        

        <li class="nav-item">
          <a href="/#culture">
            
            <span>Culture</span>
            
          </a>
        </li>

        
        

        
        
        
        
        
          
        

        <li class="nav-item">
          <a href="/#publications_selected">
            
            <span>Publications</span>
            
          </a>
        </li>

        
        
      

      
      </ul>

    </div>
  </div>
</nav>


<article class="article" itemscope itemtype="http://schema.org/Article">

  


  <div class="article-container">
    <h1 itemprop="name">#GI2019 2019 11 08 Morning Session</h1>

    

<div class="article-metadata">

  <span class="article-date">
    
    <time datetime="2019-11-08 00:00:00 &#43;0000 UTC" itemprop="datePublished dateModified">
      Nov 8, 2019
    </time>
  </span>
  <span itemscope itemprop="author publisher" itemtype="http://schema.org/Person">
    <meta itemprop="name" content="David McGaughey">
  </span>

  
  <span class="middot-divider"></span>
  <span class="article-reading-time">
    6 min read
  </span>
  

  
  
  <span class="middot-divider"></span>
  <a href="/./post/gi2019-day03-morning/#disqus_thread"></a>
  

  
  
  
  <span class="middot-divider"></span>
  <span class="article-categories">
    <i class="fa fa-folder"></i>
    
    <a href="/./categories/bioinformatics/">bioinformatics</a
    >, 
    
    <a href="/./categories/gi2019/">gi2019</a
    >, 
    
    <a href="/./categories/conferences/">conferences</a
    >, 
    
    <a href="/./categories/talk/">talk</a
    >
    
  </span>
  
  

  
  
<div class="share-box" aria-hidden="true">
  <ul class="share">
    <li>
      <a class="twitter"
         href="https://twitter.com/intent/tweet?text=%23GI2019%202019%2011%2008%20Morning%20Session&amp;url=%2f.%2fpost%2fgi2019-day03-morning%2f"
         target="_blank" rel="noopener">
        <i class="fa fa-twitter"></i>
      </a>
    </li>
    <li>
      <a class="facebook"
         href="https://www.facebook.com/sharer.php?u=%2f.%2fpost%2fgi2019-day03-morning%2f"
         target="_blank" rel="noopener">
        <i class="fa fa-facebook"></i>
      </a>
    </li>
    <li>
      <a class="linkedin"
         href="https://www.linkedin.com/shareArticle?mini=true&amp;url=%2f.%2fpost%2fgi2019-day03-morning%2f&amp;title=%23GI2019%202019%2011%2008%20Morning%20Session"
         target="_blank" rel="noopener">
        <i class="fa fa-linkedin"></i>
      </a>
    </li>
    <li>
      <a class="weibo"
         href="http://service.weibo.com/share/share.php?url=%2f.%2fpost%2fgi2019-day03-morning%2f&amp;title=%23GI2019%202019%2011%2008%20Morning%20Session"
         target="_blank" rel="noopener">
        <i class="fa fa-weibo"></i>
      </a>
    </li>
    <li>
      <a class="email"
         href="mailto:?subject=%23GI2019%202019%2011%2008%20Morning%20Session&amp;body=%2f.%2fpost%2fgi2019-day03-morning%2f">
        <i class="fa fa-envelope"></i>
      </a>
    </li>
  </ul>
</div>


  

</div>


    <div class="article-style" itemprop="articleBody">
      <div id="TOC">
<ul>
<li><a href="#day-3---session-7---microbial-and-metagenomics">Day 3 - Session 7 - MICROBIAL AND METAGENOMICS</a></li>
<li><a href="#strain-level-metagenomic-assignment-and-compositional-estimation-for-long-reads-with-metamaps">Strain-level metagenomic assignment and compositional estimation for long reads with MetaMaps</a></li>
<li><a href="#metaflyescalable-long-read-metagenome-assembly-using-repeat-graphs">metaFlye—Scalable long-read metagenome assembly using repeat graphs</a></li>
<li><a href="#detecting-microbial-transmission-and-engraftment-after-faecal-microbiota-transplants-using-long-read-metagenomics-and-reticulatus">Detecting microbial transmission and engraftment after faecal microbiota transplants using long-read metagenomics and reticulatus</a></li>
<li><a href="#entropy-of-a-bacterial-stress-response-is-a-generalizable-predictor-for-fitness-and-antibiotic-sensitivity">Entropy of a bacterial stress response is a generalizable predictor for fitness and antibiotic sensitivity</a></li>
<li><a href="#the-use-of-kmer-counts-to-train-random-forests-to-predict-country-of-origin-for-bacterial-pathogen-sequencing-data">The use of kmer counts to train random forests to predict country of origin for bacterial pathogen sequencing data</a></li>
<li><a href="#real-time-assembly-using-nanopore-sequencing-data-for-microbial-communities">Real-time assembly using Nanopore sequencing data for microbial communities</a></li>
<li><a href="#exploring-the-role-of-ribosomal-gene-repeats-in-the-context-of-regeneration">Exploring the role of ribosomal gene repeats in the context of regeneration</a></li>
<li><a href="#genomic-epidemiology-of-west-nile-virus-in-california">Genomic epidemiology of West Nile virus in California</a></li>
</ul>
</div>

<div id="day-3---session-7---microbial-and-metagenomics" class="section level1">
<h1>Day 3 - Session 7 - MICROBIAL AND METAGENOMICS</h1>
<p>Genome Informatics 2019 at CSHL</p>
<p><strong>Bold</strong> is the speaker</p>
<p><strong>If you dislike/disagree with my notes/sentiment and you are the speaker/PI then contact me. I very much could be mis-understanding some important points.</strong></p>
<p>My notes will likely reflect my background knowledge of microbes. Which is nil.</p>
</div>
<div id="strain-level-metagenomic-assignment-and-compositional-estimation-for-long-reads-with-metamaps" class="section level1">
<h1>Strain-level metagenomic assignment and compositional estimation for long reads with MetaMaps</h1>
<p><strong>Alexander T. Dilthey</strong>, Chirag Jain, Sergey Koren, Adam M. Phillippy.</p>
<p>Discussion of how MinHash works.</p>
<p>kmers, which are hashed, then sorted (take the smallest), compare kmers between query and ref</p>
<p>MinHash ~ Jaccard similarity (point to assess similarity, e.g. “alignment”)</p>
<p>“Mapping qualities”</p>
<ul>
<li>want prob. dist. of all mapping locations for a read</li>
<li>assume
<ul>
<li>read error rate known</li>
</ul></li>
<li>compute jaccard similarity</li>
<li>model as Bernoulli</li>
</ul>
<p>MetaMaps</p>
<ul>
<li>integrated metagenomic classification system</li>
<li>takes lots from MashMap</li>
<li>auto-tunes minimum read length &amp; identity (EM)</li>
</ul>
<p>Benchmark</p>
<ul>
<li>precision better than others (Kraken, Kraken2, Centrifuge, MEGAN-LR)</li>
<li><strong>REQUIRES LONG READS (&gt; 1000bp)</strong></li>
<li>slower than other methods (and takes more mem)
<ul>
<li>but are working on this</li>
</ul></li>
</ul>
<p>Going through some case studies on finding novel things</p>
<ul>
<li>find mismatches from errors in ref databases</li>
<li>location mapping allows you to quantify categories of genes (or I totally misheard/misread)</li>
</ul>
<p>github.com/DiltheyLab/MetaMaps</p>
</div>
<div id="metaflyescalable-long-read-metagenome-assembly-using-repeat-graphs" class="section level1">
<h1>metaFlye—Scalable long-read metagenome assembly using repeat graphs</h1>
<p><strong>Mikhail Kolmogorov</strong>, Mikhail Rayko, Jeffrey Yuan, Evgeny Polevikov, Pavel Pevzner.</p>
<p>Genome Assembly, with graphs</p>
<p>Flye intro:
- glue repeats into the graph</p>
<p>OK, I’m not following this. Sorry Mikhail!</p>
<p>Isn’t this a metagenome session? This is all genome assembly.</p>
<p>Oh, he is shifting now.</p>
<p>How to detect repeats?</p>
<ul>
<li>in genomes can use coverage</li>
<li>not reliable enough in metagenomes</li>
<li>need to use graph</li>
</ul>
<p>Emphasizes look at graph viz</p>
<ul>
<li>want one loop for bacteria chr</li>
<li>reality sucks</li>
<li>bubbles oh my
<ul>
<li>strain hetergeneity?</li>
<li>or repeat -&gt; bubble -&gt; repeat?!</li>
</ul></li>
<li>BUBBLES CAN HAVE BUBBLES</li>
</ul>
<p>Showing cow rumen microbiome</p>
<p>Now Sheep microbiome</p>
<p>Looks like many successes and many crazy things</p>
<p>Emphasizes metaFlye is still very much in dev.</p>
<p>Recommends Bandage (Wick et al) and AGB (Mikheenko &amp; Kolmogorov) for viz of assemblies</p>
<p>Asks us - would you prefer simple consensus or bubbly haplotype resolved assembly?</p>
<p>github.com/fenderglass/Flye</p>
</div>
<div id="detecting-microbial-transmission-and-engraftment-after-faecal-microbiota-transplants-using-long-read-metagenomics-and-reticulatus" class="section level1">
<h1>Detecting microbial transmission and engraftment after faecal microbiota transplants using long-read metagenomics and reticulatus</h1>
<p><strong>Samuel M. Nicholls</strong>, Joshua C. Quick, Tariq H. Iqbal, Nicholas J. Loman.</p>
<p><span class="citation">@samstudio8</span></p>
<p>Intro on why microbiome is important</p>
<ul>
<li>health</li>
<li>exciting as field isn’t fully define for best practices</li>
</ul>
<p>“three peaks challenge”</p>
<ul>
<li>want extraction method that maintains composition (equal representation of strain?) and long reads</li>
</ul>
<p>(everyone is so coy about what is a faecal transplant)</p>
<p>anyways….checking human microbiomes pre/post transplant</p>
<p>get looooong reads</p>
<p>showing n50, contig nums, etc. they look uh, big? (don’t have a great sense about what is acceptable)</p>
<p>diving into communities. oooooo very cool plot (bubble plot?) with families of bacteria. so many complete (?) genomes</p>
<p>need to do assembly polishing, but expensive computation</p>
<p>github.com/samstudio8/reticulatus (snakemake pipeline for polishing)</p>
<p>tech discussion of GPU acceleration of steps in pipeline</p>
<ul>
<li>100 minutes -&gt; 5 minutes with GPU acceleration</li>
<li>10 - 5 hours on promethION</li>
</ul>
<p>File parsers!</p>
<ul>
<li>poor gzip FQ parser</li>
<li>port heng li readfq…save 40 minutes</li>
</ul>
<p>Huge time savings on real data</p>
<p>back to data</p>
<ul>
<li>showing diffs pre/post transplantation
<ul>
<li>novel communities?</li>
</ul></li>
<li>how do they find these? eyeballing? diff testing?</li>
</ul>
</div>
<div id="entropy-of-a-bacterial-stress-response-is-a-generalizable-predictor-for-fitness-and-antibiotic-sensitivity" class="section level1">
<h1>Entropy of a bacterial stress response is a generalizable predictor for fitness and antibiotic sensitivity</h1>
<p><strong>Defne Surujon</strong>, Zeyu Zhu, Juan C. Ortiz-Marquez, José Bento, Tim van Opijnen.</p>
<p>Transciptional signatures can seprate straisn (antibiotic sensitive/resistant)</p>
<p>Tough to do diff testing because genes often just flat out missing in some strains</p>
<p>Want a more <strong>universal</strong> signature</p>
<ul>
<li>across strains/conditions</li>
<li>without relying on previous knowledge of gene function/identity/patterns</li>
</ul>
<p>Try to predict fitness</p>
<ul>
<li><em>in vitro</em> make high fitness for control (what? how? )</li>
</ul>
<p>biorxiv.org/content/10.1101/813709v1</p>
<p>3d PCA of strains, colored by (expression?) of major classes of genes</p>
<ul>
<li>noticed that high fitness genes had far less variance</li>
</ul>
<p>gene expression is <strong>more chaotic</strong> in low fitness strains</p>
<p>can we use entropy? but how do we calculate entropy?</p>
<ul>
<li>assume each gene independent (which def. is not correct in real world….but still works in practice)</li>
<li>calculate variance in gene expression</li>
<li>then sum all var?</li>
</ul>
<p>making networks of gene interactions with some parameter tuning (to figure out edge nums/connectivity) to improve modelling</p>
<p>to save time/money/sanity moved over testing to just one timepoint</p>
<ul>
<li>big diff in distribution width (log2FC density plot) between low/high fitness</li>
</ul>
<p>Does this work for <strong>many</strong> pathogens?</p>
<ul>
<li>seems like it</li>
</ul>
</div>
<div id="the-use-of-kmer-counts-to-train-random-forests-to-predict-country-of-origin-for-bacterial-pathogen-sequencing-data" class="section level1">
<h1>The use of kmer counts to train random forests to predict country of origin for bacterial pathogen sequencing data</h1>
<p><strong>Lauren Cowley</strong>, Jordan Taylor, Claire Jenkins, Tim Dallman.</p>
<p><span class="citation">@laurencowley4</span></p>
<p>Talking about food borne pathogens</p>
<ul>
<li>apparently 50% (!) of UK Salmonella is associated with foreign travel / imported food</li>
</ul>
<p>Discusses pathogen tracking system in UK</p>
<ul>
<li>local doc, epidemiologists</li>
</ul>
<p>Use RF for k-mer counting to figure out country of origin</p>
<p>fsm-lite used (what is this?)</p>
<ul>
<li>kmer counter?</li>
<li>some kmers mark strains</li>
</ul>
<p>big ’ol matrix</p>
<ul>
<li>5 million kmers across 500 strains, for example</li>
</ul>
<p>random forest (RF) intro</p>
<ul>
<li>classifier</li>
<li>non continuous data</li>
<li>highly resistant to overfitting (bc lots ’o trees)</li>
<li>interpretability (kmers / features which are most informative for classifier)</li>
</ul>
<p>train of 2017 data, test on 2018 data</p>
<ul>
<li>71% accuracy for first pass</li>
<li>only good at domestic/foreign</li>
<li>non-england countries poorly represented</li>
<li>decided to aggregate countries by region
<ul>
<li>little better</li>
</ul></li>
<li>neural network gets to 80% accuracy with most common 100 kmers</li>
<li>try with most strains, fewer kmers</li>
<li>will try unitigs instead of kmers</li>
</ul>
</div>
<div id="real-time-assembly-using-nanopore-sequencing-data-for-microbial-communities" class="section level1">
<h1>Real-time assembly using Nanopore sequencing data for microbial communities</h1>
<p><strong>Son H. Nguyen</strong>, Minh D. Cao, Lachlan J. Coin.</p>
</div>
<div id="exploring-the-role-of-ribosomal-gene-repeats-in-the-context-of-regeneration" class="section level1">
<h1>Exploring the role of ribosomal gene repeats in the context of regeneration</h1>
<p><strong>Sofia N. Barreira</strong>, Andreas D. Baxevanis.</p>
<p>Woo NHGRI (did postdoc there)!</p>
<p>rDNA is highly repetitive in humans</p>
<p>big intergenic spacer (GS)</p>
<p>UBF (nucleolar tf) binds across rRNA</p>
<p>but this is human….baxevanis lab studies hydractinia (highly regenerative marine hydrozoan from phylum cnidaria)</p>
<ul>
<li>you can decaptitate, and head regrows in 3 days</li>
</ul>
<p>are there genomic features that contribute to regeneration?</p>
<ul>
<li>got partial 18S and 28S seq from NCBI as well as human 5.8S and aligned to hydractinia genome assembly</li>
<li>enhancer elements in rRNA?</li>
<li>going to do chip-seq</li>
</ul>
<p>what about UBF?</p>
<ul>
<li>tough to ID</li>
<li>looking for emergence across nematodes, ctenophores, cnidaria, arthopods….chordates</li>
<li>doing a proteome assembly</li>
</ul>
</div>
<div id="genomic-epidemiology-of-west-nile-virus-in-california" class="section level1">
<h1>Genomic epidemiology of West Nile virus in California</h1>
<p><strong>Karthik Gangavarapu</strong>, Nathaniel Matteson, on behalf of the WestNile 4K Project.</p>
<p>West Nile</p>
<ul>
<li>birds &lt;-&gt; mosquito -&gt; human (dead end for west nile, yay)</li>
<li>showing infection rate across time and US states</li>
<li>variable infection rates by state</li>
</ul>
<p>sampling across california (CA) across time</p>
<p>seq -&gt; phylogenetic tree</p>
<ul>
<li>5 clades</li>
<li>cool flow moving plots showing movement of strains across CA across time</li>
<li>virus can persist across seasons</li>
</ul>
</div>

    </div>

    


<div class="article-tags">
  
  <a class="btn btn-primary btn-outline" href="/./tags/bioinformatics/">bioinformatics</a>
  
  <a class="btn btn-primary btn-outline" href="/./tags/gi2019/">gi2019</a>
  
  <a class="btn btn-primary btn-outline" href="/./tags/conferences/">conferences</a>
  
  <a class="btn btn-primary btn-outline" href="/./tags/talk/">talk</a>
  
</div>




    
    
    <div class="article-widget">
      <div class="hr-light"></div>
      <h3>Related</h3>
      <ul>
        
        <li><a href="/post/gi2019-day03-afternoon/">#GI2019 2019 11 08 Afternoon Session</a></li>
        
        <li><a href="/post/gi2019-day02/">#GI2019 2019 11 07</a></li>
        
        <li><a href="/post/gi2019-day02-night/">#GI2019 2019 11 07 Night Session</a></li>
        
        <li><a href="/post/gi2019-day01/">#GI2019 2019 11 06</a></li>
        
        <li><a href="/post/one-developer-portal-genesis/">One Developer Portal: eyeIntegration Genesis</a></li>
        
      </ul>
    </div>
    

    

    
<section id="comments">
  <div id="disqus_thread"></div>
<script>
    var disqus_config = function () {
    
    
    
    };
    (function() {
        if (["localhost", "127.0.0.1"].indexOf(window.location.hostname) != -1) {
            document.getElementById('disqus_thread').innerHTML = 'Disqus comments not available by default when the website is previewed locally.';
            return;
        }
        var d = document, s = d.createElement('script'); s.async = true;
        s.src = '//' + "davemcg-gitub-io" + '.disqus.com/embed.js';
        s.setAttribute('data-timestamp', +new Date());
        (d.head || d.body).appendChild(s);
    })();
</script>
<noscript>Please enable JavaScript to view the <a href="https://disqus.com/?ref_noscript">comments powered by Disqus.</a></noscript>
<a href="https://disqus.com" class="dsq-brlink">comments powered by <span class="logo-disqus">Disqus</span></a>
</section>



  </div>
</article>

<footer class="site-footer">
  <div class="container">
    <p class="powered-by">

      &copy; 2018 &middot; 

      Powered by the
      <a href="https://sourcethemes.com/academic/" target="_blank" rel="noopener">Academic theme</a> for
      <a href="https://gohugo.io" target="_blank" rel="noopener">Hugo</a>.

      <span class="pull-right" aria-hidden="true">
        <a href="#" id="back_to_top">
          <span class="button_icon">
            <i class="fa fa-chevron-up fa-2x"></i>
          </span>
        </a>
      </span>

    </p>
  </div>
</footer>


<div id="modal" class="modal fade" role="dialog">
  <div class="modal-dialog">
    <div class="modal-content">
      <div class="modal-header">
        <button type="button" class="close btn-large" data-dismiss="modal">&times;</button>
        <h4 class="modal-title">Cite</h4>
      </div>
      <div>
        <pre><code class="modal-body tex"></code></pre>
      </div>
      <div class="modal-footer">
        <a class="btn btn-primary btn-outline js-copy-cite" href="#" target="_blank">
          <i class="fa fa-copy"></i> Copy
        </a>
        <a class="btn btn-primary btn-outline js-download-cite" href="#" target="_blank">
          <i class="fa fa-download"></i> Download
        </a>
        <div id="modal-error"></div>
      </div>
    </div>
  </div>
</div>

    

    
    
    <script id="dsq-count-scr" src="//davemcg-gitub-io.disqus.com/count.js" async></script>
    

    

    
    <script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.2.1/jquery.min.js" integrity="sha512-3P8rXCuGJdNZOnUx/03c1jOTnMn3rP63nBip5gOP2qmUh5YAdVAvFZ1E+QLZZbC1rtMrQb+mah3AfYW11RUrWA==" crossorigin="anonymous"></script>
    <script src="https://cdnjs.cloudflare.com/ajax/libs/jquery.imagesloaded/4.1.3/imagesloaded.pkgd.min.js" integrity="sha512-umsR78NN0D23AzgoZ11K7raBD+R6hqKojyBZs1w8WvYlsI+QuKRGBx3LFCwhatzBunCjDuJpDHwxD13sLMbpRA==" crossorigin="anonymous"></script>
    <script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.3.7/js/bootstrap.min.js" integrity="sha512-iztkobsvnjKfAtTNdHkGVjAYTrrtlC7mGp/54c40wowO7LhURYl3gVzzcEqGl/qKXQltJ2HwMrdLcNUdo+N/RQ==" crossorigin="anonymous"></script>
    <script src="https://cdnjs.cloudflare.com/ajax/libs/jquery.isotope/3.0.4/isotope.pkgd.min.js" integrity="sha512-VDBOIlDbuC4VWxGJNmuFRQ0Li0SKkDpmGyuhAG5LTDLd/dJ/S0WMVxriR2Y+CyPL5gzjpN4f/6iqWVBJlht0tQ==" crossorigin="anonymous"></script>
    <script src="https://cdnjs.cloudflare.com/ajax/libs/fancybox/3.2.5/jquery.fancybox.min.js" integrity="sha256-X5PoE3KU5l+JcX+w09p/wHl9AzK333C4hJ2I9S5mD4M=" crossorigin="anonymous"></script>
    
    
    <script src="/js/hugo-academic.js"></script>
    

    
    
      
      
      <script src="//cdnjs.cloudflare.com/ajax/libs/highlight.js/9.12.0/highlight.min.js" integrity="sha256-/BfiIkHlHoVihZdc6TFuj7MmJ0TWcWsMXkeDFwhi0zw=" crossorigin="anonymous"></script>
      

      

      

      <script>hljs.initHighlightingOnLoad();</script>
    

    
    
    <script type="text/x-mathjax-config">
        MathJax.Hub.Config({ tex2jax: { inlineMath: [['$','$'], ['\\(','\\)']] } });
    </script>
    
    <script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.1/MathJax.js?config=TeX-AMS_CHTML" integrity="sha512-tOav5w1OjvsSJzePRtt2uQPFwBoHt1VZcUq8l8nm5284LEKE9FSJBQryzMBzHxY5P0zRdNqEcpLIRVYFNgu1jw==" crossorigin="anonymous"></script>
    
    

  </body>
</html>

