<!DOCTYPE html>
<html lang="en-us">
<head>

  <meta charset="utf-8">
  <meta name="viewport" content="width=device-width, initial-scale=1">
  <meta http-equiv="X-UA-Compatible" content="IE=edge">
  <meta name="theme" content="hugo-academic">
  <meta name="generator" content="Hugo 0.40" />
  <meta name="author" content="David McGaughey">

  
  
  
  
    
      
    
  
  <meta name="description" content="Day 3 - Session 8 - EVOLUTION AND PHYLOGENETICS Is pathogen evolution predictable? The role of population genomics Learning the properties of adaptive regions with functional data analysis Creating pan-human and population-specific consensus representations of the reference genome and assessing their effect on functional genomic data analysis Gramene subsites—Pangenome browsers for crops  What do we gain when tolerating loss? The information bottleneck, lossy compression, and detecting horizontal gene transfer A recurrent neural network for inferring sweeps and allele frequency trajectories using gene trees based on the ancestral recombination graph Assembling the Y chromosomes of anopheles mosquitoes   Day 3 - Session 8 - EVOLUTION AND PHYLOGENETICS Genome Informatics 2019 at CSHL">

  
  <link rel="alternate" hreflang="en-us" href="/./post/gi2019-day03-afternoon/">

  


  

  
  
  <meta name="theme-color" content="#0095eb">
  
  
  
  
    
  
  
    
    
      
        <link rel="stylesheet" href="//cdnjs.cloudflare.com/ajax/libs/highlight.js/9.12.0/styles/github.min.css">
      
    
  
  
  <link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.3.7/css/bootstrap.min.css" integrity="sha512-6MXa8B6uaO18Hid6blRMetEIoPqHf7Ux1tnyIQdpt9qI5OACx7C+O3IVTr98vwGnlcg0LOLa02i9Y1HpVhlfiw==" crossorigin="anonymous">
  <link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/academicons/1.8.6/css/academicons.min.css" integrity="sha256-uFVgMKfistnJAfoCUQigIl+JfUaP47GrRKjf6CTPVmw=" crossorigin="anonymous">
  <link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/4.7.0/css/font-awesome.min.css" integrity="sha512-SfTiTlX6kk+qitfevl/7LibUOeJWlt9rbyDn92a1DqWOw9vWG2MFoays0sgObmWazO5BQPiFucnnEAjpAB+/Sw==" crossorigin="anonymous">
  <link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/fancybox/3.2.5/jquery.fancybox.min.css" integrity="sha256-ygkqlh3CYSUri3LhQxzdcm0n1EQvH2Y+U5S2idbLtxs=" crossorigin="anonymous">
  
  
  
  
  <link rel="stylesheet" href="//fonts.googleapis.com/css?family=Montserrat:400,700%7cRoboto:400,400italic,700%7cRoboto&#43;Mono">
  
  <link rel="stylesheet" href="/styles.css">
  

  
    <script>
      window.ga=window.ga||function(){(ga.q=ga.q||[]).push(arguments)};ga.l=+new Date;
      ga('create', 'UA-113461893-1', 'auto');
      
      ga('require', 'eventTracker');
      ga('require', 'outboundLinkTracker');
      ga('require', 'urlChangeTracker');
      ga('send', 'pageview');
    </script>
    <script async src="//www.google-analytics.com/analytics.js"></script>
    
    <script async src="https://cdnjs.cloudflare.com/ajax/libs/autotrack/2.4.1/autotrack.js" integrity="sha512-HUmooslVKj4m6OBu0OgzjXXr+QuFYy/k7eLI5jdeEy/F4RSgMn6XRWRGkFi5IFaFgy7uFTkegp3Z0XnJf3Jq+g==" crossorigin="anonymous"></script>
    
  

  
  <link rel="alternate" href="/./index.xml" type="application/rss+xml" title="eye Bioinformatician">
  <link rel="feed" href="/./index.xml" type="application/rss+xml" title="eye Bioinformatician">
  

  <link rel="manifest" href="/site.webmanifest">
  <link rel="icon" type="image/png" href="/img/icon.png">
  <link rel="apple-touch-icon" type="image/png" href="/img/icon-192.png">

  <link rel="canonical" href="/./post/gi2019-day03-afternoon/">

  <meta property="twitter:card" content="summary_large_image">
  
  <meta property="twitter:site" content="@https://twitter.com/David_McGaughey">
  <meta property="twitter:creator" content="@https://twitter.com/David_McGaughey">
  
  <meta property="og:site_name" content="eye Bioinformatician">
  <meta property="og:url" content="/./post/gi2019-day03-afternoon/">
  <meta property="og:title" content="#GI2019 2019 11 08 Afternoon Session | eye Bioinformatician">
  <meta property="og:description" content="Day 3 - Session 8 - EVOLUTION AND PHYLOGENETICS Is pathogen evolution predictable? The role of population genomics Learning the properties of adaptive regions with functional data analysis Creating pan-human and population-specific consensus representations of the reference genome and assessing their effect on functional genomic data analysis Gramene subsites—Pangenome browsers for crops  What do we gain when tolerating loss? The information bottleneck, lossy compression, and detecting horizontal gene transfer A recurrent neural network for inferring sweeps and allele frequency trajectories using gene trees based on the ancestral recombination graph Assembling the Y chromosomes of anopheles mosquitoes   Day 3 - Session 8 - EVOLUTION AND PHYLOGENETICS Genome Informatics 2019 at CSHL">
  <meta property="og:locale" content="en-us">
  
  <meta property="article:published_time" content="2019-11-08T00:00:00&#43;00:00">
  
  <meta property="article:modified_time" content="2019-11-08T00:00:00&#43;00:00">
  

  
  

  <title>#GI2019 2019 11 08 Afternoon Session | eye Bioinformatician</title>

</head>
<body id="top" data-spy="scroll" data-target="#toc" data-offset="71" >

<nav class="navbar navbar-default navbar-fixed-top" id="navbar-main">
  <div class="container">

    
    <div class="navbar-header">
      
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse"
              data-target=".navbar-collapse" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      
      <a class="navbar-brand" href="/">eye Bioinformatician</a>
    </div>

    
    <div class="collapse navbar-collapse">

      
      
      <ul class="nav navbar-nav navbar-right">
        

        
        
        
        
        
          
        

        <li class="nav-item">
          <a href="/#about">
            
            <span>Home</span>
            
          </a>
        </li>

        
        

        
        <li class="dropdown">
          <a href="#" class="dropdown-toggle" data-toggle="dropdown" aria-haspopup="true">
            
            <span>Posts</span>
            
            <span class="caret"></span>
          </a>
          <ul class="dropdown-menu">
            
            <li class="nav-item">
              <a href="/#posts">
                
                <span>Recent Posts</span>
                
              </a>
            </li>
            
            <li class="nav-item">
              <a href="/post">
                
                <span>All Posts</span>
                
              </a>
            </li>
            
          </ul>
        </li>

        
        

        
        
        
        
        
          
        

        <li class="nav-item">
          <a href="/#projects">
            
            <span>Projects</span>
            
          </a>
        </li>

        
        

        
        
        
        
        
          
        

        <li class="nav-item">
          <a href="/#culture">
            
            <span>Culture</span>
            
          </a>
        </li>

        
        

        
        
        
        
        
          
        

        <li class="nav-item">
          <a href="/#publications_selected">
            
            <span>Publications</span>
            
          </a>
        </li>

        
        
      

      
      </ul>

    </div>
  </div>
</nav>


<article class="article" itemscope itemtype="http://schema.org/Article">

  


  <div class="article-container">
    <h1 itemprop="name">#GI2019 2019 11 08 Afternoon Session</h1>

    

<div class="article-metadata">

  <span class="article-date">
    
    <time datetime="2019-11-08 00:00:00 &#43;0000 UTC" itemprop="datePublished dateModified">
      Nov 8, 2019
    </time>
  </span>
  <span itemscope itemprop="author publisher" itemtype="http://schema.org/Person">
    <meta itemprop="name" content="David McGaughey">
  </span>

  
  <span class="middot-divider"></span>
  <span class="article-reading-time">
    5 min read
  </span>
  

  
  
  <span class="middot-divider"></span>
  <a href="/./post/gi2019-day03-afternoon/#disqus_thread"></a>
  

  
  
  
  <span class="middot-divider"></span>
  <span class="article-categories">
    <i class="fa fa-folder"></i>
    
    <a href="/./categories/bioinformatics/">bioinformatics</a
    >, 
    
    <a href="/./categories/gi2019/">gi2019</a
    >, 
    
    <a href="/./categories/conferences/">conferences</a
    >, 
    
    <a href="/./categories/talk/">talk</a
    >
    
  </span>
  
  

  
  
<div class="share-box" aria-hidden="true">
  <ul class="share">
    <li>
      <a class="twitter"
         href="https://twitter.com/intent/tweet?text=%23GI2019%202019%2011%2008%20Afternoon%20Session&amp;url=%2f.%2fpost%2fgi2019-day03-afternoon%2f"
         target="_blank" rel="noopener">
        <i class="fa fa-twitter"></i>
      </a>
    </li>
    <li>
      <a class="facebook"
         href="https://www.facebook.com/sharer.php?u=%2f.%2fpost%2fgi2019-day03-afternoon%2f"
         target="_blank" rel="noopener">
        <i class="fa fa-facebook"></i>
      </a>
    </li>
    <li>
      <a class="linkedin"
         href="https://www.linkedin.com/shareArticle?mini=true&amp;url=%2f.%2fpost%2fgi2019-day03-afternoon%2f&amp;title=%23GI2019%202019%2011%2008%20Afternoon%20Session"
         target="_blank" rel="noopener">
        <i class="fa fa-linkedin"></i>
      </a>
    </li>
    <li>
      <a class="weibo"
         href="http://service.weibo.com/share/share.php?url=%2f.%2fpost%2fgi2019-day03-afternoon%2f&amp;title=%23GI2019%202019%2011%2008%20Afternoon%20Session"
         target="_blank" rel="noopener">
        <i class="fa fa-weibo"></i>
      </a>
    </li>
    <li>
      <a class="email"
         href="mailto:?subject=%23GI2019%202019%2011%2008%20Afternoon%20Session&amp;body=%2f.%2fpost%2fgi2019-day03-afternoon%2f">
        <i class="fa fa-envelope"></i>
      </a>
    </li>
  </ul>
</div>


  

</div>


    <div class="article-style" itemprop="articleBody">
      <div id="TOC">
<ul>
<li><a href="#day-3---session-8---evolution-and-phylogenetics">Day 3 - Session 8 - EVOLUTION AND PHYLOGENETICS</a></li>
<li><a href="#is-pathogen-evolution-predictable-the-role-of-population-genomics">Is pathogen evolution predictable? The role of population genomics</a></li>
<li><a href="#learning-the-properties-of-adaptive-regions-with-functional-data-analysis">Learning the properties of adaptive regions with functional data analysis</a></li>
<li><a href="#creating-pan-human-and-population-specific-consensus-representations-of-the-reference-genome-and-assessing-their-effect-on-functional-genomic-data-analysis">Creating pan-human and population-specific consensus representations of the reference genome and assessing their effect on functional genomic data analysis</a></li>
<li><a href="#gramene-subsitespangenome-browsers-for-crops">Gramene subsites—Pangenome browsers for crops</a></li>
<li><a href="#section"></a></li>
<li><a href="#what-do-we-gain-when-tolerating-loss-the-information-bottleneck-lossy-compression-and-detecting-horizontal-gene-transfer">What do we gain when tolerating loss? The information bottleneck, lossy compression, and detecting horizontal gene transfer</a></li>
<li><a href="#a-recurrent-neural-network-for-inferring-sweeps-and-allele-frequency-trajectories-using-gene-trees-based-on-the-ancestral-recombination-graph">A recurrent neural network for inferring sweeps and allele frequency trajectories using gene trees based on the ancestral recombination graph</a></li>
<li><a href="#assembling-the-y-chromosomes-of-anopheles-mosquitoes">Assembling the Y chromosomes of anopheles mosquitoes</a></li>
</ul>
</div>

<div id="day-3---session-8---evolution-and-phylogenetics" class="section level1">
<h1>Day 3 - Session 8 - EVOLUTION AND PHYLOGENETICS</h1>
<p>Genome Informatics 2019 at CSHL</p>
<p><strong>Bold</strong> is the speaker</p>
<p><strong>If you dislike/disagree with my notes/sentiment and you are the speaker/PI then contact me. I very much could be mis-understanding some important points.</strong></p>
</div>
<div id="is-pathogen-evolution-predictable-the-role-of-population-genomics" class="section level1">
<h1>Is pathogen evolution predictable? The role of population genomics</h1>
<p><strong>William Hanage.</strong></p>
<p>Is pathogen evolution predictable?</p>
<ul>
<li>Probably not</li>
<li>Cool, talk over, I guess.</li>
</ul>
<p>Oh he is still going</p>
<p>“Accessory genome”</p>
<ul>
<li>three <em>e. coli</em> strains only share 39% of the genes
<ul>
<li>Welsh et al…….2001 (?)</li>
</ul></li>
<li>the “mr. potato head” model</li>
</ul>
<p>Sample 4k+ pneumococcal genomes from 4 dif places</p>
<ul>
<li>Corander Nat Eco Evol 2017</li>
<li>found 73 “sequence clusters” based on population (tree?) structure</li>
<li>also 1700+ accessory genes (COGS)</li>
</ul>
<p><strong>Dif strains are correlated when using accessory genes (was effectively R^2 of 0 with full genome)</strong></p>
<p>Negative Frequency Dependent Selection</p>
<ul>
<li>balancing selection</li>
<li>corrolary are antigens/phage receptors - if too common get ID’ed and wiped out</li>
</ul>
<p>Pneumococcal Conjugate vaccines (PCVs) as an evolutionary experiment</p>
<ul>
<li>we have good vaccines against <em>some</em> strains</li>
<li>7</li>
<li>but there are 90+ serotypes</li>
</ul>
<p>Can we predict what types will persist/succeed after vaccination?</p>
<p>Sampling (seq.) per/post vaccination</p>
<ul>
<li>35 strains found</li>
<li>vaccination wiped out the 7 that were covered</li>
<li>uneven “replacement” rate by other strains
<ul>
<li>not super convinced … is this just noise?</li>
</ul></li>
</ul>
<p>Isolates in the same group tend to have the same set of COGS (getting increasingly concerned I don’t know what COGS stands for…)</p>
<p>Check for prevalence of COGS pre/post vaccination and sum distance from linear expected.</p>
<ul>
<li>directional</li>
</ul>
<p>Can predict fitness … and it roughly matches what you’d expect. Not perfect, but the correlation looks real.</p>
<p>doi.org/10.1101/420315 (&lt;- bioRxiv)</p>
<p>What are the genes?! (&lt;- his formatting)</p>
<ul>
<li>aaaah, he didn’t say.</li>
<li>tease</li>
</ul>
</div>
<div id="learning-the-properties-of-adaptive-regions-with-functional-data-analysis" class="section level1">
<h1>Learning the properties of adaptive regions with functional data analysis</h1>
<p><strong>Mehreen R. Mughal</strong>, Hillary Koch, Jinguo Huang, Francesca Chiaromonte, Michael DeGiorgio.</p>
<p>How do organisms adapt to their environment?</p>
<ul>
<li>pos. selection == (over time) fixation</li>
<li>idea is more beneficial == faster fixation</li>
<li>selective sweep reduces diversity at haplotype
<ul>
<li>(chunk of genome)</li>
<li>so one beneficial variant can “drag” a whole lot of other variants into purity.
<ul>
<li>free-loaders</li>
</ul></li>
</ul></li>
</ul>
<p>SURFDAWave</p>
<ul>
<li>ha</li>
<li>learn spatial dist. from statistics of sweep / neutral</li>
<li>doi.org/10.1101/834010</li>
<li>pick dist to match data</li>
<li>penalized regression
<ul>
<li>with wavelet coefficient</li>
<li>cross-validate params to check fit</li>
</ul></li>
<li>predicts neutral/sweep pretty well in TF/TP/FN/FP</li>
<li>each statistic tested gets own model
<ul>
<li>initial freq</li>
<li>selection coef</li>
<li>time of selection</li>
</ul></li>
</ul>
<p>Now running on SLC45A2</p>
<ul>
<li>related to skin color in europeans</li>
<li>estimated time of selection 2000 years ago with initial freq of 0.04</li>
</ul>
<p>Adaptive introgression</p>
<ul>
<li>selective sweep (?)</li>
<li>beneficial mutation has unique pattern</li>
<li>which SURFDAWave has 52% successful classification
<ul>
<li>kind of bleh</li>
<li>can be improved with more info?</li>
<li>with 2D info 52 -&gt; 62%</li>
</ul></li>
</ul>
</div>
<div id="creating-pan-human-and-population-specific-consensus-representations-of-the-reference-genome-and-assessing-their-effect-on-functional-genomic-data-analysis" class="section level1">
<h1>Creating pan-human and population-specific consensus representations of the reference genome and assessing their effect on functional genomic data analysis</h1>
<p><strong>Benjamin Kaminow</strong>, Sara Ballouz, Jesse Gillis, Alex Dobin.</p>
<p>How much variation do we need to improve ref genome?</p>
<p>We can use <strong>all</strong> known…but graph is complicated.</p>
<p>Consensus genome</p>
<ul>
<li>simple</li>
<li>replace minor alleles with major alleles in other pop</li>
<li>have been used on variant calling</li>
<li><strong>here we assess effects of RNA-seq</strong></li>
</ul>
<p>ConsDB download and parse db files which makes consensus VCF and FASTA</p>
<p>Uses 1000G</p>
<ul>
<li>better pop balance</li>
<li>uh, but doesn’t 1000G have loads of technical errors due to seq. errors?
<ul>
<li>was this ever fixed……?</li>
</ul></li>
<li>got RNA-seq data from one person from 4 diff pops</li>
<li>anyways….millions of replacements of SNPS</li>
<li>once replacements made, then # of diff SNPs decreases from one genome &lt;-&gt; consensus genome</li>
</ul>
<p>check mapping with STARconsensus</p>
<ul>
<li>STAR + VCF file to transform the reference</li>
<li>also does compare pre/post correction<br />
= VERY large drop in mapping error rate…(3.5 to 0.5%)
<ul>
<li>much bigger effect than I would have thought….</li>
</ul></li>
<li>wait I’m confused
<ul>
<li>y axis says “% of reads overlapping variant”</li>
<li>that’s not error rate….</li>
</ul></li>
</ul>
</div>
<div id="gramene-subsitespangenome-browsers-for-crops" class="section level1">
<h1>Gramene subsites—Pangenome browsers for crops</h1>
<p>Marcela K. Tello-Ruiz, Sharon Wei, Joshua Stein, Kapeel Chougule, <strong>Andrew Olson</strong>, Yinping Jiao, Bo Wang, Ivar Meijs, Doreen Ware.</p>
</div>
<div id="section" class="section level1">
<h1></h1>
<p><strong>Janet Kelso.</strong></p>
<p>NOT TWEETABLE</p>
</div>
<div id="what-do-we-gain-when-tolerating-loss-the-information-bottleneck-lossy-compression-and-detecting-horizontal-gene-transfer" class="section level1">
<h1>What do we gain when tolerating loss? The information bottleneck, lossy compression, and detecting horizontal gene transfer</h1>
<p><strong>Apurva Narechania</strong>, Rob Desalle, Barun Mathema, Barry Kreiswirth,
Paul Planet.</p>
<p>Can we do horizontal gene transfer without genes?</p>
<p>Make pangenome</p>
<p>count kmers across genomes</p>
<p>clusters kmers</p>
<p>but how do we cluster well? don’t want to lose too much info…but also want to compress…how to get the proper balance?</p>
<p>claude shannon!</p>
<ul>
<li>transmitter -&gt; noisy channel -&gt; receiver</li>
<li>noisy channel ~ noisy clusters</li>
<li>want middle ground where you can interpret but haven’t lost much info</li>
</ul>
<p>can see how clusters stabliize with n kmers</p>
<p>I’m reminded of luke zappias/oshlack clustree work</p>
<ul>
<li>which is ironic bc this talk is all about avoiding trees</li>
<li>done in 2 hours … which I guess is fast?</li>
</ul>
<p>Plot show inflection between cluster n and norm. mutual information</p>
<p>q: you fixed at 19 kmer, why?
a: looking at other kmer n - balance between closeness of variants and kmer n</p>
</div>
<div id="a-recurrent-neural-network-for-inferring-sweeps-and-allele-frequency-trajectories-using-gene-trees-based-on-the-ancestral-recombination-graph" class="section level1">
<h1>A recurrent neural network for inferring sweeps and allele frequency trajectories using gene trees based on the ancestral recombination graph</h1>
<p><strong>Hussein A</strong>. Hejase, Ziyi Mo, Adam Siepel.</p>
</div>
<div id="assembling-the-y-chromosomes-of-anopheles-mosquitoes" class="section level1">
<h1>Assembling the Y chromosomes of anopheles mosquitoes</h1>
<p><strong>Chujia Chen</strong>, Austin Compton, Yang Wu, Jiangtao Liang, Dustin Miller, Xiaoguang Chen, Igor Sharakhov, Chunhong Mao, Zhijian J. Tu.</p>
</div>

    </div>

    


<div class="article-tags">
  
  <a class="btn btn-primary btn-outline" href="/./tags/bioinformatics/">bioinformatics</a>
  
  <a class="btn btn-primary btn-outline" href="/./tags/gi2019/">gi2019</a>
  
  <a class="btn btn-primary btn-outline" href="/./tags/conferences/">conferences</a>
  
  <a class="btn btn-primary btn-outline" href="/./tags/talk/">talk</a>
  
</div>




    
    
    <div class="article-widget">
      <div class="hr-light"></div>
      <h3>Related</h3>
      <ul>
        
        <li><a href="/post/gi2019-day03-morning/">#GI2019 2019 11 08 Morning Session</a></li>
        
        <li><a href="/post/gi2019-day02-night/">#GI2019 2019 11 07 Night Session</a></li>
        
        <li><a href="/post/gi2019-day02/">#GI2019 2019 11 07</a></li>
        
        <li><a href="/post/gi2019-day01/">#GI2019 2019 11 06</a></li>
        
        <li><a href="/post/one-developer-portal-genesis/">One Developer Portal: eyeIntegration Genesis</a></li>
        
      </ul>
    </div>
    

    

    
<section id="comments">
  <div id="disqus_thread"></div>
<script>
    var disqus_config = function () {
    
    
    
    };
    (function() {
        if (["localhost", "127.0.0.1"].indexOf(window.location.hostname) != -1) {
            document.getElementById('disqus_thread').innerHTML = 'Disqus comments not available by default when the website is previewed locally.';
            return;
        }
        var d = document, s = d.createElement('script'); s.async = true;
        s.src = '//' + "davemcg-gitub-io" + '.disqus.com/embed.js';
        s.setAttribute('data-timestamp', +new Date());
        (d.head || d.body).appendChild(s);
    })();
</script>
<noscript>Please enable JavaScript to view the <a href="https://disqus.com/?ref_noscript">comments powered by Disqus.</a></noscript>
<a href="https://disqus.com" class="dsq-brlink">comments powered by <span class="logo-disqus">Disqus</span></a>
</section>



  </div>
</article>

<footer class="site-footer">
  <div class="container">
    <p class="powered-by">

      &copy; 2018 &middot; 

      Powered by the
      <a href="https://sourcethemes.com/academic/" target="_blank" rel="noopener">Academic theme</a> for
      <a href="https://gohugo.io" target="_blank" rel="noopener">Hugo</a>.

      <span class="pull-right" aria-hidden="true">
        <a href="#" id="back_to_top">
          <span class="button_icon">
            <i class="fa fa-chevron-up fa-2x"></i>
          </span>
        </a>
      </span>

    </p>
  </div>
</footer>


<div id="modal" class="modal fade" role="dialog">
  <div class="modal-dialog">
    <div class="modal-content">
      <div class="modal-header">
        <button type="button" class="close btn-large" data-dismiss="modal">&times;</button>
        <h4 class="modal-title">Cite</h4>
      </div>
      <div>
        <pre><code class="modal-body tex"></code></pre>
      </div>
      <div class="modal-footer">
        <a class="btn btn-primary btn-outline js-copy-cite" href="#" target="_blank">
          <i class="fa fa-copy"></i> Copy
        </a>
        <a class="btn btn-primary btn-outline js-download-cite" href="#" target="_blank">
          <i class="fa fa-download"></i> Download
        </a>
        <div id="modal-error"></div>
      </div>
    </div>
  </div>
</div>

    

    
    
    <script id="dsq-count-scr" src="//davemcg-gitub-io.disqus.com/count.js" async></script>
    

    

    
    <script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.2.1/jquery.min.js" integrity="sha512-3P8rXCuGJdNZOnUx/03c1jOTnMn3rP63nBip5gOP2qmUh5YAdVAvFZ1E+QLZZbC1rtMrQb+mah3AfYW11RUrWA==" crossorigin="anonymous"></script>
    <script src="https://cdnjs.cloudflare.com/ajax/libs/jquery.imagesloaded/4.1.3/imagesloaded.pkgd.min.js" integrity="sha512-umsR78NN0D23AzgoZ11K7raBD+R6hqKojyBZs1w8WvYlsI+QuKRGBx3LFCwhatzBunCjDuJpDHwxD13sLMbpRA==" crossorigin="anonymous"></script>
    <script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.3.7/js/bootstrap.min.js" integrity="sha512-iztkobsvnjKfAtTNdHkGVjAYTrrtlC7mGp/54c40wowO7LhURYl3gVzzcEqGl/qKXQltJ2HwMrdLcNUdo+N/RQ==" crossorigin="anonymous"></script>
    <script src="https://cdnjs.cloudflare.com/ajax/libs/jquery.isotope/3.0.4/isotope.pkgd.min.js" integrity="sha512-VDBOIlDbuC4VWxGJNmuFRQ0Li0SKkDpmGyuhAG5LTDLd/dJ/S0WMVxriR2Y+CyPL5gzjpN4f/6iqWVBJlht0tQ==" crossorigin="anonymous"></script>
    <script src="https://cdnjs.cloudflare.com/ajax/libs/fancybox/3.2.5/jquery.fancybox.min.js" integrity="sha256-X5PoE3KU5l+JcX+w09p/wHl9AzK333C4hJ2I9S5mD4M=" crossorigin="anonymous"></script>
    
    
    <script src="/js/hugo-academic.js"></script>
    

    
    
      
      
      <script src="//cdnjs.cloudflare.com/ajax/libs/highlight.js/9.12.0/highlight.min.js" integrity="sha256-/BfiIkHlHoVihZdc6TFuj7MmJ0TWcWsMXkeDFwhi0zw=" crossorigin="anonymous"></script>
      

      

      

      <script>hljs.initHighlightingOnLoad();</script>
    

    
    
    <script type="text/x-mathjax-config">
        MathJax.Hub.Config({ tex2jax: { inlineMath: [['$','$'], ['\\(','\\)']] } });
    </script>
    
    <script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.1/MathJax.js?config=TeX-AMS_CHTML" integrity="sha512-tOav5w1OjvsSJzePRtt2uQPFwBoHt1VZcUq8l8nm5284LEKE9FSJBQryzMBzHxY5P0zRdNqEcpLIRVYFNgu1jw==" crossorigin="anonymous"></script>
    
    

  </body>
</html>

